<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Eduard Magerusan">

<title>SLE_Transcriptome_Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="SLE_Transcriptome_Analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/quarto.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SLE_Transcriptome_Analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">SLE_Transcriptome_Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Eduard Magerusan </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="reanalysis-of-rna-seq-data-from-multicell-type-gene-coexpression-network-analysis-reveals-coordinated-interferon-response-and-crosscell-type-correlations-in-systemic-lupus-erythematosus" class="level2">
<h2 class="anchored" data-anchor-id="reanalysis-of-rna-seq-data-from-multicell-type-gene-coexpression-network-analysis-reveals-coordinated-interferon-response-and-crosscell-type-correlations-in-systemic-lupus-erythematosus">Reanalysis of RNA-Seq Data from ‘Multi–Cell Type Gene Coexpression Network Analysis Reveals Coordinated Interferon Response and Cross–Cell Type Correlations in Systemic Lupus Erythematosus’</h2>
<p>Systemic lupus erythematosus (SLE) is a complex autoimmune disease characterized by dysregulated immune responses involving multiple cell types. In this study, we perform a reanalysis of publicly available bulk RNA-seq data from GSE149050 to identify differentially expressed genes and enriched pathways associated with SLE pathogenesis.</p>
</section>
<section id="loading-the-dataset" class="level1">
<h1>1. Loading the dataset</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Core Bioconductor</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)        </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)       </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)          </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene information</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)  </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)   </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)       </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano) </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Data wrangling</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)         </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:gridExtra':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:matrixStats':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:GenomicRanges':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:IRanges':

    collapse, desc, intersect, setdiff, slice, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    first, intersect, rename, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dbscan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbscan'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    as.dendrogram</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apeglm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE149050_Bulk_Human_RawCounts.txt"</span>,</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">row.names =</span> <span class="dv">1</span>,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">check.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">quote =</span> <span class="st">""</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">comment.char =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/SraRunTable.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="metadata-preprocessing-and-alignment" class="level1">
<h1>2. Metadata Preprocessing and Alignment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>cell_type_clean <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" cells"</span>, <span class="st">""</span>, metadata<span class="sc">$</span>cell_type)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metadata<span class="sc">$</span>patientid, <span class="st">"_"</span>, metadata<span class="sc">$</span>ifn_status, <span class="st">"_"</span>, metadata<span class="sc">$</span>cell_type_clean)</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the number prefix (e.g., "001_") from counts colnames</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>counts_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^[0-9]+_"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(counts))</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset metadata to only matched samples</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>metadata_matched <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>sample_id <span class="sc">%in%</span> counts_id, ]</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder metadata to match the order of counts</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>match_index <span class="ot">&lt;-</span> <span class="fu">match</span>(counts_id, metadata_matched<span class="sc">$</span>sample_id)</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>metadata_aligned <span class="ot">&lt;-</span> metadata_matched[match_index, ]</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Align rownames</span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_aligned) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(counts)</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Final check</span></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(metadata_aligned) <span class="sc">==</span> <span class="fu">colnames</span>(counts)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
<section id="principal-component-analysis-for-exploratory-data-visualization" class="level1">
<h1>3. Principal Component Analysis for Exploratory Data Visualization</h1>
<p>Principal component analysis was performed to explore patterns in gene expression and assess sample clustering based on experimental conditions and cell types. Multiple PCA plots were generated to identify potential batch effects and biological variability.</p>
<section id="pca-plot-highlighting-interferon-status" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-highlighting-interferon-status">3.1 PCA Plot Highlighting Interferon Status</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>ifn_status, </span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"HC"</span>, <span class="st">"IFNneg"</span>, <span class="st">"IFNpos"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out low-count genes</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize and transform</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"ifn_status"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-highlighting-cell-type-specific-expression-patterns" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-highlighting-cell-type-specific-expression-patterns">3.2 PCA Plot Highlighting Cell Type-Specific Expression Patterns</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>cell_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>cell_type)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"cell_type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-stratified-by-cell-type-and-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-stratified-by-cell-type-and-disease-state">3.3 PCA Plot Stratified by Cell Type and Disease State</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>disease_state)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type <span class="sc">*</span> disease_state)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"cell_type"</span>, <span class="st">"disease_state"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-stratified-by-cell-type-and-interferon-status" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-stratified-by-cell-type-and-interferon-status">3.3 PCA Plot Stratified by Cell Type and Interferon Status</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type <span class="sc">*</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"cell_type"</span>, <span class="st">"ifn_status"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cell-type-specific-pca-plots-stratified-by-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="cell-type-specific-pca-plots-stratified-by-disease-state">3.4 Cell Type-Specific PCA Plots Stratified by Disease State</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over all cell types and generate PCA plots by disease_state</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata_aligned<span class="sc">$</span>cell_type)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"Processing PCA for: "</span>, ct)</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>  meta_sub <span class="ot">&lt;-</span> metadata_aligned[metadata_aligned<span class="sc">$</span>cell_type <span class="sc">==</span> ct, ]</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>  counts_sub <span class="ot">&lt;-</span> counts[, <span class="fu">rownames</span>(meta_sub)]</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts_sub,</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta_sub,</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> disease_state)</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_sub) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_sub[keep, ]</span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_sub, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_sub, <span class="at">intgroup =</span> <span class="st">"disease_state"</span>) <span class="sc">+</span></span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">"PCA for"</span>, ct))</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: T cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: B cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: PMN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cDC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: pDC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cMo</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="detecting-and-removing-outliers-using-local-outlier-factor-lof" class="level1">
<h1>4. Detecting and Removing Outliers using Local Outlier Factor (LOF)</h1>
<p>To improve accuracy in PCA space, we applied the Local Outlier Factor (LOF) algorithm to detect and exclude samples that deviate from the local density of their respective cell types.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>pc <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>pc_scores <span class="ot">&lt;-</span> pc<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a><span class="co"># LOF calculation</span></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>lof_scores <span class="ot">&lt;-</span> <span class="fu">lof</span>(pc_scores, <span class="at">k =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in lof(pc_scores, k = 10): lof: k is now deprecated. use minPts = 11
instead .</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lof_scores) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dds)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize LOF scores</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">sort</span>(lof_scores), <span class="at">type =</span> <span class="st">"h"</span>, <span class="at">main =</span> <span class="st">"Sorted LOF Scores"</span>,</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"LOF Score"</span>, <span class="at">xlab =</span> <span class="st">"Sample Index"</span>)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">quantile</span>(lof_scores, <span class="fl">0.995</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set a threshold</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(lof_scores, <span class="fl">0.99</span>)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">names</span>(lof_scores[lof_scores <span class="sc">&gt;</span> threshold])</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove outlier and replot</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>dds_clean <span class="ot">&lt;-</span> dds[, <span class="sc">!</span>(<span class="fu">colnames</span>(dds) <span class="sc">%in%</span> outliers)]</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>vsd_clean <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_clean, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd_clean, <span class="at">intgroup =</span> <span class="st">"cell_type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cell-type-specific-pca-plots-stratified-by-disease-state-1" class="level1">
<h1>5. Cell Type-Specific PCA Plots Stratified by Disease State</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over all cell types and generate PCA plots by ifn_status</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"Processing PCA for: "</span>, ct)</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset cleaned dds object</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> ct]</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply minimal filtering (if needed)</span></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_sub) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_sub[keep, ]</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use no design formula for PCA</span></span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">counts</span>(dds_sub),</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> <span class="fu">colData</span>(dds_sub),</span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transform and PCA</span></span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_sub, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_sub, <span class="at">intgroup =</span> <span class="st">"ifn_status"</span>) <span class="sc">+</span></span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">"PCA for"</span>, ct))</span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: T cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: B cells
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: PMN
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cDC
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: pDC
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cMo
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Principal component analysis reveals disease-related clustering driven by the interferon response, with IFN-positive samples separating clearly from healthy controls and IFN-negative individuals. This indicates that IFN signaling contributes substantially to the transcriptional variability observed across different immune cell types in systemic lupus erythematosus.</p>
</section>
<section id="differential-expression-analysis" class="level1">
<h1>6. Differential Expression Analysis</h1>
<section id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-hc">6.1 Differential Expression Analysis for cMo Cells (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>cell_type_cMo <span class="ot">&lt;-</span> <span class="st">"cMo"</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cMo cells</span></span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_cMo]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>keep_cMo <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_cMo) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> dds_cMo[keep_cMo, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>cts_cMo <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_cMo)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>coldata_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cMo))</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>coldata_cMo<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cMo<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cMo,</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cMo,</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 673 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>res_cMo <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cMo, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>res_cMo <span class="ot">&lt;-</span> res_cMo[<span class="fu">order</span>(res_cMo<span class="sc">$</span>padj), ]</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 23365 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1049, 4.5%
LFC &lt; 0 (down)     : 666, 2.9%
outliers [1]       : 0, 0%
low counts [2]     : 1812, 7.8%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27     1601.10        8.16377  0.558304   14.6225 2.01987e-48 4.35344e-44
EIF2AK2   1371.42        1.60031  0.111775   14.3172 1.70912e-46 1.84184e-42
ISG15     2959.12        2.50801  0.177266   14.1483 1.91351e-45 1.37473e-41
IFI6      7850.30        2.69260  0.197190   13.6548 1.89012e-42 1.01844e-38
IFITM1    2738.47        2.89249  0.212503   13.6115 3.42048e-42 1.47443e-38
LY6E      6767.12        2.47251  0.186137   13.2833 2.89459e-40 1.03978e-36</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>resLFC_cMo <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cMo,</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-cmo-cells" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-cmo-cells">6.1.1 Exporting results for cMo Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>res_sig_cMo <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cMo, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cMo), <span class="at">file =</span> <span class="st">"results/DEG_significant_cMo_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes">6.1.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cMo, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in cMo"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-hc">6.1.3 Top 10 Differentially Expressed Genes in cMo (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cMo[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cMo<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27     1601.09952        8.16377  0.558304  14.62245 2.01987e-48 4.35344e-44
OTOF       274.27434        6.67924  0.724277   9.22194 2.91779e-20 9.38615e-18
LINC00487    8.53610        5.87209  1.298883   4.52088 6.15841e-06 3.71799e-04
RBP4         5.27853        5.16154  1.160490   4.44772 8.67867e-06 4.98803e-04
MS4A1        4.09053        4.97487  1.686323   2.95013 3.17640e-03 5.35269e-02
KCNMA1       4.00869        4.63947  1.217596   3.81035 1.38771e-04 5.00992e-03
ADAMTS2     89.31312        4.26389  0.670728   6.35710 2.05604e-10 3.01454e-08
IGHG2        2.88288        4.17064  1.197272   3.48346 4.94985e-04 1.33532e-02
STAP1       15.04623        4.14981  0.980391   4.23281 2.30788e-05 1.14877e-03
NCAPGP1      2.96550        3.64041  1.370619   2.65603 7.90664e-03 9.95977e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot">6.1.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo)</span>
<span id="cb128-3"><a href="#cb128-3" aria-hidden="true" tabindex="-1"></a>res_df_cMo<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo)</span>
<span id="cb128-4"><a href="#cb128-4" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> res_df_cMo[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb128-5"><a href="#cb128-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-6"><a href="#cb128-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cMo,</span>
<span id="cb128-7"><a href="#cb128-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cMo<span class="sc">$</span>gene,                </span>
<span id="cb128-8"><a href="#cb128-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb128-9"><a href="#cb128-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb128-10"><a href="#cb128-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb128-11"><a href="#cb128-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb128-12"><a href="#cb128-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in cMo'</span>,</span>
<span id="cb128-13"><a href="#cb128-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb128-14"><a href="#cb128-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb128-15"><a href="#cb128-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb128-16"><a href="#cb128-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb128-17"><a href="#cb128-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb128-18"><a href="#cb128-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb128-19"><a href="#cb128-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb128-20"><a href="#cb128-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb128-21"><a href="#cb128-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb128-22"><a href="#cb128-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb128-23"><a href="#cb128-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb128-24"><a href="#cb128-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb128-25"><a href="#cb128-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1701 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes">6.1.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>vsd_cMo <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_cMo, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_cMo)</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo)</span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> res_df_cMo[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>padj), ]</span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>res_df_cMo<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo)</span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo[<span class="fu">order</span>(res_df_cMo<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cMo)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-19"><a href="#cb130-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb130-20"><a href="#cb130-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb130-21"><a href="#cb130-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb130-22"><a href="#cb130-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb130-23"><a href="#cb130-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb130-24"><a href="#cb130-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb130-25"><a href="#cb130-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb130-26"><a href="#cb130-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in cMo (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-hc">6.2 Differential Expression Analysis for cDC Cells (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>cell_type_cDC <span class="ot">&lt;-</span> <span class="st">"cDC"</span></span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_cDC]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>keep_cDC <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_cDC) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> dds_cDC[keep_cDC, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>cts_cDC <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_cDC)</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>coldata_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cDC))</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>coldata_cDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cDC,</span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cDC,</span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 102 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>res_cDC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>res_cDC <span class="ot">&lt;-</span> res_cDC[<span class="fu">order</span>(res_cDC<span class="sc">$</span>padj), ]</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 18625 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1529, 8.2%
LFC &lt; 0 (down)     : 1310, 7%
outliers [1]       : 0, 0%
low counts [2]     : 2889, 16%
(mean count &lt; 9)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFIT1    688.106        3.77915  0.293633   12.8703 6.61310e-38 1.04064e-33
IFITM1  3325.129        3.44492  0.271635   12.6822 7.42639e-37 5.84308e-33
ISG15   1460.197        3.05677  0.260789   11.7212 9.92123e-32 5.20401e-28
IFI6    2847.881        3.60185  0.311729   11.5544 7.01290e-31 2.75888e-27
IFI27   1272.994        9.01383  0.791256   11.3918 4.59382e-30 1.44577e-26
IFIT3   1268.449        3.89440  0.344651   11.2995 1.31915e-29 3.45970e-26</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>resLFC_cDC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cDC,</span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-cdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-cdc-cells">6.2.1 Exporting results for cDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>res_sig_cDC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cDC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cDC), <span class="at">file =</span> <span class="st">"results/DEG_significant_cDC_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-1" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-1">6.2.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cDC, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in cDC"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-hc">6.2.3 Top 10 Differentially Expressed Genes in cDC (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cDC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cDC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
           baseMean log2FoldChange     lfcSE      stat      pvalue        padj
          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A     101.84935       14.28602  2.848197   5.01581 5.28103e-07 4.77599e-05
IFI27    1272.99387        9.01383  0.791256  11.39180 4.59382e-30 1.44577e-26
OTOF       50.04154        7.67626  1.401762   5.47615 4.34676e-08 5.56102e-06
IFI44L   1998.83463        5.09890  0.483852  10.53815 5.76208e-26 9.06721e-23
APOBEC3A  380.21260        4.97504  0.670576   7.41904 1.17969e-13 3.63991e-11
USP18     198.72203        4.64458  0.427297  10.86968 1.60759e-27 3.16213e-24
ETV7       43.93760        4.55012  0.722553   6.29728 3.02907e-10 6.11096e-08
USP41      31.90146        4.30186  0.476438   9.02922 1.72900e-19 1.29559e-16
TNFAIP6     9.92146        4.24865  0.888660   4.78096 1.74461e-06 1.26512e-04
IFIT3    1268.44887        3.89440  0.344651  11.29953 1.31915e-29 3.45970e-26</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-1" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot-1">6.2.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC)</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>res_df_cDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC)</span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cDC,</span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cDC<span class="sc">$</span>gene,                </span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in cDC'</span>,</span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb155-19"><a href="#cb155-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb155-20"><a href="#cb155-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb155-21"><a href="#cb155-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb155-22"><a href="#cb155-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb155-23"><a href="#cb155-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb155-24"><a href="#cb155-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb155-25"><a href="#cb155-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2821 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-1" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-1">6.2.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>vsd_cDC <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_cDC, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_cDC)</span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb157-8"><a href="#cb157-8" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC)</span>
<span id="cb157-9"><a href="#cb157-9" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>padj), ]</span>
<span id="cb157-10"><a href="#cb157-10" aria-hidden="true" tabindex="-1"></a>res_df_cDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC)</span>
<span id="cb157-11"><a href="#cb157-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-12"><a href="#cb157-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC[<span class="fu">order</span>(res_df_cDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb157-13"><a href="#cb157-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-14"><a href="#cb157-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb157-15"><a href="#cb157-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb157-16"><a href="#cb157-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-17"><a href="#cb157-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cDC)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb157-18"><a href="#cb157-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-19"><a href="#cb157-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb157-20"><a href="#cb157-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb157-21"><a href="#cb157-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb157-22"><a href="#cb157-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb157-23"><a href="#cb157-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb157-24"><a href="#cb157-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb157-25"><a href="#cb157-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb157-26"><a href="#cb157-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in cDC (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-b-cells-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-b-cells-ifnpos-vs-hc">6.3 Differential Expression Analysis for B Cells (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>cell_type_B_Cells <span class="ot">&lt;-</span> <span class="st">"B cells"</span></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_B_Cells]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>keep_B_Cells <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_B_Cells) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> dds_B_Cells[keep_B_Cells, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>cts_B_Cells <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_B_Cells)</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>coldata_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_B_Cells))</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>coldata_B_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_B_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_B_Cells,</span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_B_Cells,</span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 116 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_B_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells <span class="ot">&lt;-</span> res_B_Cells[<span class="fu">order</span>(res_B_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20081 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1422, 7.1%
LFC &lt; 0 (down)     : 730, 3.6%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
CMPK2    168.043        3.81075  0.268312  14.20267 8.81859e-46 1.77086e-41
IFI44L  1479.816        4.58296  0.377188  12.15034 5.71290e-34 5.73604e-30
IFI44    536.649        3.91298  0.339869  11.51320 1.13204e-30 7.57750e-27
IFIT3    488.133        3.60142  0.314413  11.45441 2.23475e-30 1.12190e-26
RSAD2    187.138        3.90328  0.356874  10.93740 7.63618e-28 3.06684e-24
IFIT1    229.237        3.55396  0.361406   9.83371 8.05946e-23 2.69737e-19</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>resLFC_B_Cells <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_B_Cells,</span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-b-cells" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-b-cells">6.3.1 Exporting results for B Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>res_sig_B_Cells <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_B_Cells, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_B_Cells), <span class="at">file =</span> <span class="st">"results/DEG_significant_B_Cells_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-2" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-2">6.3.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_B_Cells, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in B Cells"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-hc">6.3.3 Top 10 Differentially Expressed Genes in B Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_B_Cells[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_B_Cells<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27      254.35770        7.73796  0.983877   7.86476 3.69797e-15 3.37541e-12
SIGLEC1     27.18012        7.02780  1.018920   6.89731 5.29980e-12 3.43308e-09
OTOF         6.11952        6.60454  1.717900   3.84455 1.20776e-04 7.43959e-03
CCL2         5.41444        6.08373  1.222757   4.97542 6.51062e-07 1.63425e-04
LINC00487    4.18743        5.77172  1.000063   5.77135 7.86367e-09 4.04898e-06
FRZB         5.14098       -4.73843  1.209170  -3.91875 8.90102e-05 6.10039e-03
VN1R105P     3.82649        4.61473  1.307863   3.52845 4.18000e-04 1.56569e-02
IFI44L    1479.81593        4.58296  0.377188  12.15034 5.71290e-34 5.73604e-30
GPR84        3.01661        4.21111  1.243101   3.38759 7.05101e-04 2.12600e-02
EMP1         5.84359        4.06424  0.843375   4.81901 1.44270e-06 3.29215e-04</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-2" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot-2">6.3.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells)</span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells)</span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> res_df_B_Cells[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb182-5"><a href="#cb182-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-6"><a href="#cb182-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_B_Cells,</span>
<span id="cb182-7"><a href="#cb182-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_B_Cells<span class="sc">$</span>gene,                </span>
<span id="cb182-8"><a href="#cb182-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb182-9"><a href="#cb182-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb182-10"><a href="#cb182-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb182-11"><a href="#cb182-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb182-12"><a href="#cb182-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in B Cells'</span>,</span>
<span id="cb182-13"><a href="#cb182-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb182-14"><a href="#cb182-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb182-15"><a href="#cb182-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb182-16"><a href="#cb182-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb182-17"><a href="#cb182-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb182-18"><a href="#cb182-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb182-19"><a href="#cb182-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb182-20"><a href="#cb182-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb182-21"><a href="#cb182-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb182-22"><a href="#cb182-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb182-23"><a href="#cb182-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb182-24"><a href="#cb182-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb182-25"><a href="#cb182-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2138 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-2" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-2">6.3.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>vsd_B_Cells <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_B_Cells, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_B_Cells)</span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells)</span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells)</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells[<span class="fu">order</span>(res_df_B_Cells<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_B_Cells)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb184-23"><a href="#cb184-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb184-24"><a href="#cb184-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb184-25"><a href="#cb184-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb184-26"><a href="#cb184-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in B Cells (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-t-cells-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-t-cells-ifnpos-vs-hc">6.4 Differential Expression Analysis for T Cells (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a>cell_type_T_Cells <span class="ot">&lt;-</span> <span class="st">"T cells"</span></span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_T_Cells]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb186-2"><a href="#cb186-2" aria-hidden="true" tabindex="-1"></a>keep_T_Cells <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_T_Cells) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb186-3"><a href="#cb186-3" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> dds_T_Cells[keep_T_Cells, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>cts_T_Cells <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_T_Cells)</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>coldata_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_T_Cells))</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-5"><a href="#cb187-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb187-6"><a href="#cb187-6" aria-hidden="true" tabindex="-1"></a>coldata_T_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_T_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb187-7"><a href="#cb187-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-8"><a href="#cb187-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb187-9"><a href="#cb187-9" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_T_Cells,</span>
<span id="cb187-10"><a href="#cb187-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_T_Cells,</span>
<span id="cb187-11"><a href="#cb187-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb188-2"><a href="#cb188-2" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 50 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_T_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells <span class="ot">&lt;-</span> res_T_Cells[<span class="fu">order</span>(res_T_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20883 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 949, 4.5%
LFC &lt; 0 (down)     : 518, 2.5%
outliers [1]       : 0, 0%
low counts [2]     : 5264, 25%
(mean count &lt; 13)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI6    2116.879        2.52368  0.200713   12.5736 2.95069e-36 4.60868e-32
OAS3     622.486        1.92040  0.165602   11.5965 4.29171e-31 3.35161e-27
IFI44L  2005.728        4.29735  0.376195   11.4232 3.20170e-30 1.66691e-26
IFI44   1137.650        2.71482  0.241037   11.2631 1.99666e-29 7.79645e-26
IFIT3    346.838        3.31500  0.305503   10.8510 1.97325e-27 6.16404e-24
OAS1     712.529        2.66137  0.245767   10.8288 2.51351e-27 6.54309e-24</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>resLFC_T_Cells <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_T_Cells,</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-t-cells" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-t-cells">6.4.1 Exporting results for T Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>res_sig_T_Cells <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_T_Cells, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_T_Cells), <span class="at">file =</span> <span class="st">"results/DEG_significant_T_Cells_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-3" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-3">6.4.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_T_Cells, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in T Cells"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-hc">6.4.3 Top 10 Differentially Expressed Genes in T Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_T_Cells[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_T_Cells<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27   807.3096        7.04081  0.682009  10.32363 5.50986e-25 6.14704e-22
IFI44L 2005.7283        4.29735  0.376195  11.42321 3.20170e-30 1.66691e-26
OTOF     32.4095        3.54321  0.630762   5.61734 1.93922e-08 5.13369e-06
IFIT3   346.8381        3.31500  0.305503  10.85097 1.97325e-27 6.16404e-24
NRIR     15.4155        3.04216  0.373345   8.14839 3.68799e-16 1.74554e-13
USP41    59.5937        2.98922  0.282578  10.57839 3.75354e-26 6.51406e-23
RSAD2   345.9063        2.92321  0.285344  10.24454 1.25225e-24 1.30392e-21
CMPK2   205.6176        2.86353  0.266171  10.75825 5.41901e-27 1.20914e-23
USP18   376.3179        2.78829  0.273649  10.18928 2.21407e-24 2.16135e-21
IFIT1   267.1082        2.78112  0.262896  10.57880 3.73696e-26 6.51406e-23</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-3" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot-3">6.4.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells)</span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells)</span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> res_df_T_Cells[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb209-5"><a href="#cb209-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-6"><a href="#cb209-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_T_Cells,</span>
<span id="cb209-7"><a href="#cb209-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_T_Cells<span class="sc">$</span>gene,                </span>
<span id="cb209-8"><a href="#cb209-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb209-9"><a href="#cb209-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb209-10"><a href="#cb209-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb209-11"><a href="#cb209-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb209-12"><a href="#cb209-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in T Cells'</span>,</span>
<span id="cb209-13"><a href="#cb209-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb209-14"><a href="#cb209-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb209-15"><a href="#cb209-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb209-16"><a href="#cb209-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb209-17"><a href="#cb209-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb209-18"><a href="#cb209-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb209-19"><a href="#cb209-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb209-20"><a href="#cb209-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb209-21"><a href="#cb209-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb209-22"><a href="#cb209-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb209-23"><a href="#cb209-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb209-24"><a href="#cb209-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb209-25"><a href="#cb209-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1453 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-3" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-3">6.4.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb211-2"><a href="#cb211-2" aria-hidden="true" tabindex="-1"></a>vsd_T_Cells <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_T_Cells, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb211-3"><a href="#cb211-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-4"><a href="#cb211-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb211-5"><a href="#cb211-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_T_Cells)</span>
<span id="cb211-6"><a href="#cb211-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-7"><a href="#cb211-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb211-8"><a href="#cb211-8" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells)</span>
<span id="cb211-9"><a href="#cb211-9" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb211-10"><a href="#cb211-10" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells)</span>
<span id="cb211-11"><a href="#cb211-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-12"><a href="#cb211-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells[<span class="fu">order</span>(res_df_T_Cells<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb211-13"><a href="#cb211-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-14"><a href="#cb211-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb211-15"><a href="#cb211-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb211-16"><a href="#cb211-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-17"><a href="#cb211-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_T_Cells)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb211-18"><a href="#cb211-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-19"><a href="#cb211-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb211-20"><a href="#cb211-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb211-21"><a href="#cb211-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb211-22"><a href="#cb211-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb211-23"><a href="#cb211-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb211-24"><a href="#cb211-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb211-25"><a href="#cb211-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb211-26"><a href="#cb211-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in T Cells (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pdc-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-pdc-ifnpos-vs-hc">6.5 Differential Expression Analysis for pDC (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>cell_type_pDC <span class="ot">&lt;-</span> <span class="st">"pDC"</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for pDC cells</span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_pDC]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a>keep_pDC <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_pDC) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb213-3"><a href="#cb213-3" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> dds_pDC[keep_pDC, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a>cts_pDC <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_pDC)</span>
<span id="cb214-3"><a href="#cb214-3" aria-hidden="true" tabindex="-1"></a>coldata_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_pDC))</span>
<span id="cb214-4"><a href="#cb214-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-5"><a href="#cb214-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb214-6"><a href="#cb214-6" aria-hidden="true" tabindex="-1"></a>coldata_pDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_pDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb214-7"><a href="#cb214-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-8"><a href="#cb214-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb214-9"><a href="#cb214-9" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_pDC,</span>
<span id="cb214-10"><a href="#cb214-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_pDC,</span>
<span id="cb214-11"><a href="#cb214-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 78 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a>res_pDC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_pDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a>res_pDC <span class="ot">&lt;-</span> res_pDC[<span class="fu">order</span>(res_pDC<span class="sc">$</span>padj), ]</span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 19119 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 454, 2.4%
LFC &lt; 0 (down)     : 354, 1.9%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 0)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
PARP14   419.5817        3.57201  0.289372   12.3440 5.24719e-35 1.00321e-30
TNFSF10  197.4279        3.06011  0.258398   11.8426 2.35051e-32 2.24697e-28
STAT1   4832.2032        1.95411  0.186362   10.4855 1.00674e-25 6.41594e-22
EIF2AK2 1075.8132        2.18572  0.212010   10.3095 6.38506e-25 2.03460e-21
IFI27    245.1479        7.50725  0.727046   10.3257 5.39294e-25 2.03460e-21
RSAD2     84.3289        3.98667  0.385998   10.3282 5.25159e-25 2.03460e-21</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a>resLFC_pDC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_pDC,</span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb230-4"><a href="#cb230-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-pdc" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-pdc">6.5.1 Exporting results for pDC</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a>res_sig_pDC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_pDC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_pDC), <span class="at">file =</span> <span class="st">"results/DEG_significant_pDC_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-4" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-4">6.5.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_pDC, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in pDC"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-66-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-hc">6.5.3 Top 10 Differentially Expressed Genes in pDC (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_pDC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_pDC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
              baseMean log2FoldChange     lfcSE      stat      pvalue
             &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
IL36A        200.67437       10.01732  1.730896   5.78736 7.15002e-09
IFI27        245.14788        7.50725  0.727046  10.32569 5.39294e-25
C1GALT1P3      5.44997        5.65935  1.894706   2.98693 2.81796e-03
USP18        410.60234        5.12661  0.587139   8.73152 2.51274e-18
USP41         67.68563        4.98772  0.580768   8.58815 8.83797e-18
AC005780.1     4.00141       -4.97312  1.045627  -4.75611 1.97356e-06
KDM5D         23.26211        4.97233  0.914887   5.43491 5.48231e-08
LGALS3BP      69.19806        4.94647  0.531916   9.29935 1.41312e-20
NRIR           9.60052        4.73858  0.782365   6.05674 1.38905e-09
RP11-640L9.2   2.43023        4.64686  1.057428   4.39449 1.11034e-05
                    padj
               &lt;numeric&gt;
IL36A        1.64700e-06
IFI27        2.03460e-21
C1GALT1P3    7.90704e-02
USP18        2.40205e-15
USP41        7.68060e-15
AC005780.1   2.45016e-04
KDM5D        1.04816e-05
LGALS3BP     2.25145e-17
NRIR         3.84889e-07
RP11-640L9.2 1.12321e-03</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-4" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot-4">6.5.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC)</span>
<span id="cb236-3"><a href="#cb236-3" aria-hidden="true" tabindex="-1"></a>res_df_pDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC)</span>
<span id="cb236-4"><a href="#cb236-4" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> res_df_pDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb236-5"><a href="#cb236-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-6"><a href="#cb236-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_pDC,</span>
<span id="cb236-7"><a href="#cb236-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_pDC<span class="sc">$</span>gene,                </span>
<span id="cb236-8"><a href="#cb236-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb236-9"><a href="#cb236-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb236-10"><a href="#cb236-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb236-11"><a href="#cb236-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb236-12"><a href="#cb236-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in pDC'</span>,</span>
<span id="cb236-13"><a href="#cb236-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb236-14"><a href="#cb236-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb236-15"><a href="#cb236-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb236-16"><a href="#cb236-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb236-17"><a href="#cb236-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb236-18"><a href="#cb236-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb236-19"><a href="#cb236-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb236-20"><a href="#cb236-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb236-21"><a href="#cb236-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb236-22"><a href="#cb236-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb236-23"><a href="#cb236-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb236-24"><a href="#cb236-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb236-25"><a href="#cb236-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 787 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-4" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-4">6.5.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>vsd_pDC <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_pDC, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb238-5"><a href="#cb238-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_pDC)</span>
<span id="cb238-6"><a href="#cb238-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-7"><a href="#cb238-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb238-8"><a href="#cb238-8" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC)</span>
<span id="cb238-9"><a href="#cb238-9" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>padj), ]</span>
<span id="cb238-10"><a href="#cb238-10" aria-hidden="true" tabindex="-1"></a>res_df_pDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC)</span>
<span id="cb238-11"><a href="#cb238-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-12"><a href="#cb238-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC[<span class="fu">order</span>(res_df_pDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb238-13"><a href="#cb238-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-14"><a href="#cb238-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb238-15"><a href="#cb238-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb238-16"><a href="#cb238-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-17"><a href="#cb238-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_pDC)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb238-18"><a href="#cb238-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-19"><a href="#cb238-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb238-20"><a href="#cb238-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb238-21"><a href="#cb238-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb238-22"><a href="#cb238-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb238-23"><a href="#cb238-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb238-24"><a href="#cb238-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb238-25"><a href="#cb238-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb238-26"><a href="#cb238-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in pDC (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pmn-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-pmn-ifnpos-vs-hc">6.6 Differential Expression Analysis for PMN (IFNpos vs HC)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a>cell_type_PMN <span class="ot">&lt;-</span> <span class="st">"PMN"</span></span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for PMN cells</span></span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_PMN]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>keep_PMN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_PMN) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> dds_PMN[keep_PMN, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a>cts_PMN <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_PMN)</span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>coldata_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_PMN))</span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>coldata_PMN<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_PMN<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_PMN,</span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_PMN,</span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 230 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb252-2"><a href="#cb252-2" aria-hidden="true" tabindex="-1"></a>res_PMN <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_PMN, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>res_PMN <span class="ot">&lt;-</span> res_PMN[<span class="fu">order</span>(res_PMN<span class="sc">$</span>padj), ]</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 17241 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1333, 7.7%
LFC &lt; 0 (down)     : 1212, 7%
outliers [1]       : 0, 0%
low counts [2]     : 335, 1.9%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
SPATS2L   283.842        4.89608  0.386544   12.6663 9.08962e-37 1.53669e-32
IFIT3   26824.278        3.72780  0.300865   12.3903 2.95049e-35 1.66270e-31
IRF7     2198.851        2.46721  0.199080   12.3930 2.85016e-35 1.66270e-31
PARP14   2981.760        2.48630  0.202578   12.2733 1.26063e-34 5.32805e-31
TMEM123  4779.615        2.22638  0.187922   11.8474 2.22082e-32 7.50904e-29
CHMP5    1993.580        1.74595  0.147942   11.8016 3.83169e-32 1.07964e-28</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb257-2"><a href="#cb257-2" aria-hidden="true" tabindex="-1"></a>resLFC_PMN <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_PMN,</span>
<span id="cb257-3"><a href="#cb257-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb257-4"><a href="#cb257-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-pmn" class="level3">
<h3 class="anchored" data-anchor-id="exporting-results-for-pmn">6.6.1 Exporting results for PMN</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>res_sig_PMN <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_PMN, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb259-2"><a href="#cb259-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_PMN), <span class="at">file =</span> <span class="st">"results/DEG_significant_PMN_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-5" class="level3">
<h3 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-5">6.6.2 MA-plot of shrunken log2 fold changes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_PMN, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in PMN"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-77-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-hc">6.6.3 Top 10 Differentially Expressed Genes in PMN (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_PMN[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_PMN<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
                baseMean log2FoldChange     lfcSE      stat      pvalue
               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
OTOF           130.96602       10.98273  1.010050  10.87345 1.54248e-27
IFI27          317.09943        9.62776  0.909950  10.58055 3.66817e-26
SIGLEC1         61.42598        8.34894  0.898786   9.28913 1.55557e-20
LY6E          9047.49266        6.81384  0.650138  10.48060 1.06067e-25
RP11-288L9.1     7.14567        6.53839  0.900625   7.25984 3.87544e-13
IFI44L        3549.05385        5.96718  0.645829   9.23956 2.47508e-20
USP18          423.57521        5.72042  0.781464   7.32012 2.47744e-13
HESX1            2.83300        5.64466  1.499583   3.76415 1.67114e-04
RSAD2        11379.20506        5.61395  0.543520  10.32887 5.21700e-25
OAS2          1926.24454        5.57061  0.671548   8.29518 1.08418e-16
                    padj
               &lt;numeric&gt;
OTOF         2.17310e-24
IFI27        3.87588e-23
SIGLEC1      5.25969e-18
LY6E         9.43775e-23
RP11-288L9.1 5.32668e-11
IFI44L       7.89504e-18
USP18        3.51963e-11
HESX1        3.31212e-03
RSAD2        3.67494e-22
OAS2         2.32015e-14</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-5" class="level3">
<h3 class="anchored" data-anchor-id="volcano-plot-5">6.6.4 Volcano Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN)</span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a>res_df_PMN<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN)</span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> res_df_pDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb263-5"><a href="#cb263-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-6"><a href="#cb263-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_PMN,</span>
<span id="cb263-7"><a href="#cb263-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_PMN<span class="sc">$</span>gene,                </span>
<span id="cb263-8"><a href="#cb263-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb263-9"><a href="#cb263-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb263-10"><a href="#cb263-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb263-11"><a href="#cb263-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb263-12"><a href="#cb263-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in PMN'</span>,</span>
<span id="cb263-13"><a href="#cb263-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb263-14"><a href="#cb263-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb263-15"><a href="#cb263-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb263-16"><a href="#cb263-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb263-17"><a href="#cb263-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb263-18"><a href="#cb263-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb263-19"><a href="#cb263-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb263-20"><a href="#cb263-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb263-21"><a href="#cb263-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb263-22"><a href="#cb263-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb263-23"><a href="#cb263-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb263-24"><a href="#cb263-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb263-25"><a href="#cb263-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2776 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-79-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-5" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-5">6.6.5 Heatmap of top differentially expressed genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>vsd_PMN <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_PMN, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb265-3"><a href="#cb265-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-4"><a href="#cb265-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb265-5"><a href="#cb265-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_PMN)</span>
<span id="cb265-6"><a href="#cb265-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-7"><a href="#cb265-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb265-8"><a href="#cb265-8" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN)</span>
<span id="cb265-9"><a href="#cb265-9" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>padj), ]</span>
<span id="cb265-10"><a href="#cb265-10" aria-hidden="true" tabindex="-1"></a>res_df_PMN<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN)</span>
<span id="cb265-11"><a href="#cb265-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-12"><a href="#cb265-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC[<span class="fu">order</span>(res_df_pDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb265-13"><a href="#cb265-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-14"><a href="#cb265-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb265-15"><a href="#cb265-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb265-16"><a href="#cb265-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-17"><a href="#cb265-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_PMN)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb265-18"><a href="#cb265-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-19"><a href="#cb265-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb265-20"><a href="#cb265-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb265-21"><a href="#cb265-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb265-22"><a href="#cb265-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb265-23"><a href="#cb265-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb265-24"><a href="#cb265-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb265-25"><a href="#cb265-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb265-26"><a href="#cb265-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in PMN (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="references" class="level1">
<h1>7. References</h1>
<p>Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for sequence count data: removing the noise and preserving large differences. Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/bty895" class="uri">https://doi.org/10.1093/bioinformatics/bty895</a></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>