<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Eduard Magerusan">

<title>SLE_Transcriptome_Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="SLE_Transcriptome_Analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/quarto.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SLE_Transcriptome_Analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SLE_Transcriptome_Analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<link href="SLE_Transcriptome_Analysis_files/libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="SLE_Transcriptome_Analysis_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/typedarray-0.1/typedarray.min.js"></script>
<script src="SLE_Transcriptome_Analysis_files/libs/jquery-3.5.1/jquery.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="SLE_Transcriptome_Analysis_files/libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="SLE_Transcriptome_Analysis_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="SLE_Transcriptome_Analysis_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">SLE_Transcriptome_Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Eduard Magerusan </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="reanalysis-of-rna-seq-data-from-multicell-type-gene-coexpression-network-analysis-reveals-coordinated-interferon-response-and-crosscell-type-correlations-in-systemic-lupus-erythematosus" class="level2">
<h2 class="anchored" data-anchor-id="reanalysis-of-rna-seq-data-from-multicell-type-gene-coexpression-network-analysis-reveals-coordinated-interferon-response-and-crosscell-type-correlations-in-systemic-lupus-erythematosus">Reanalysis of RNA-Seq Data from ‘Multi–Cell Type Gene Coexpression Network Analysis Reveals Coordinated Interferon Response and Cross–Cell Type Correlations in Systemic Lupus Erythematosus’</h2>
<p>Systemic lupus erythematosus is a complex autoimmune disease characterized by dysregulated immune responses involving multiple cell types. In this study, we perform a reanalysis of publicly available bulk RNA-seq data from GSE149050 to identify differentially expressed genes and enriched pathways associated with SLE pathogenesis.</p>
</section>
<section id="loading-the-dataset" class="level1">
<h1>1. Loading the dataset</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Core Bioconductor</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)        </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)       </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)          </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene information</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)  </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)   </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)       </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Data wrangling</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)         </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:gridExtra':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:matrixStats':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:GenomicRanges':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:IRanges':

    collapse, desc, intersect, setdiff, slice, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    first, intersect, rename, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dbscan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbscan'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    as.dendrogram</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apeglm)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Enrichment Analysis</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummArIzeR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: circlize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.16
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ComplexHeatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.22.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: igraph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:circlize':

    degree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: plotly</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'plotly'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    groups</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ComplexHeatmap':

    add_heatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggplot2':

    last_plot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:graphics':

    layout</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: proxy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'proxy'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    as.matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    as.matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    as.dist, dist</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    as.matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stringr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: tidyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'dplyr::as_data_frame' by
'igraph::as_data_frame' when loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'dplyr::groups' by 'igraph::groups' when
loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'dplyr::union' by 'igraph::union' when
loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'circlize::degree' by 'igraph::degree' when
loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'ComplexHeatmap::add_heatmap' by
'plotly::add_heatmap' when loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'igraph::groups' by 'plotly::groups' when
loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'ggplot2::last_plot' by 'plotly::last_plot'
when loading 'SummArIzeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'igraph::crossing' by 'tidyr::crossing' when
loading 'SummArIzeR'</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to enrichR
Checking connections ... </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Enrichr ... Connection is Live!
FlyEnrichr ... Connection is Live!
WormEnrichr ... Connection is Live!
YeastEnrichr ... Connection is Live!
FishEnrichr ... Connection is Live!
OxEnrichr ... Connection is Live!</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hacksig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE149050_Bulk_Human_RawCounts.txt"</span>,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">row.names =</span> <span class="dv">1</span>,</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">check.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">quote =</span> <span class="st">""</span>,</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">comment.char =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/SraRunTable.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="metadata-preprocessing-and-matching" class="level1">
<h1>2. Metadata Preprocessing and Matching</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>cell_type_clean <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" cells"</span>, <span class="st">""</span>, metadata<span class="sc">$</span>cell_type)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metadata<span class="sc">$</span>patientid, <span class="st">"_"</span>, metadata<span class="sc">$</span>ifn_status, <span class="st">"_"</span>, metadata<span class="sc">$</span>cell_type_clean)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the number prefix (e.g., "001_") from counts colnames</span></span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>counts_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^[0-9]+_"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(counts))</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset metadata to only matched samples</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>metadata_matched <span class="ot">&lt;-</span> metadata[metadata<span class="sc">$</span>sample_id <span class="sc">%in%</span> counts_id, ]</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder metadata to match the order of counts</span></span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>match_index <span class="ot">&lt;-</span> <span class="fu">match</span>(counts_id, metadata_matched<span class="sc">$</span>sample_id)</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>metadata_aligned <span class="ot">&lt;-</span> metadata_matched[match_index, ]</span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Align rownames</span></span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_aligned) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(counts)</span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Final check</span></span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(metadata_aligned) <span class="sc">==</span> <span class="fu">colnames</span>(counts)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
</section>
<section id="principal-component-analysis-for-exploratory-data-visualization" class="level1">
<h1>3. Principal Component Analysis for Exploratory Data Visualization</h1>
<p>Principal component analysis was performed to explore patterns in gene expression and assess sample clustering based on experimental conditions and cell types. Multiple PCA plots were generated to identify potential batch effects and biological variability.</p>
<section id="pca-plot-highlighting-interferon-status" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-highlighting-interferon-status">3.1 PCA Plot Highlighting Interferon Status</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>ifn_status, </span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"HC"</span>, <span class="st">"IFNneg"</span>, <span class="st">"IFNpos"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out low-count genes</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize and transform</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"ifn_status"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-highlighting-cell-type-specific-expression-patterns" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-highlighting-cell-type-specific-expression-patterns">3.2 PCA Plot Highlighting Cell Type-Specific Expression Patterns</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>cell_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>cell_type)</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"cell_type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-stratified-by-cell-type-and-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-stratified-by-cell-type-and-disease-state">3.3 PCA Plot Stratified by Cell Type and Disease State</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>metadata_aligned<span class="sc">$</span>disease_state <span class="ot">&lt;-</span> <span class="fu">factor</span>(metadata_aligned<span class="sc">$</span>disease_state)</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type <span class="sc">*</span> disease_state)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"cell_type"</span>, <span class="st">"disease_state"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-plot-stratified-by-cell-type-and-interferon-status" class="level2">
<h2 class="anchored" data-anchor-id="pca-plot-stratified-by-cell-type-and-interferon-status">3.3 PCA Plot Stratified by Cell Type and Interferon Status</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts,</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata_aligned,</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> cell_type <span class="sc">*</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"cell_type"</span>, <span class="st">"ifn_status"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cell-type-specific-pca-plots-stratified-by-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="cell-type-specific-pca-plots-stratified-by-disease-state">3.4 Cell Type-Specific PCA Plots Stratified by Disease State</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over all cell types and generate PCA plots by disease_state</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata_aligned<span class="sc">$</span>cell_type)</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"Processing PCA for: "</span>, ct)</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>  meta_sub <span class="ot">&lt;-</span> metadata_aligned[metadata_aligned<span class="sc">$</span>cell_type <span class="sc">==</span> ct, ]</span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>  counts_sub <span class="ot">&lt;-</span> counts[, <span class="fu">rownames</span>(meta_sub)]</span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts_sub,</span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> meta_sub,</span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> disease_state)</span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_sub) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb123-15"><a href="#cb123-15" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_sub[keep, ]</span>
<span id="cb123-16"><a href="#cb123-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-17"><a href="#cb123-17" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_sub, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb123-18"><a href="#cb123-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-19"><a href="#cb123-19" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_sub, <span class="at">intgroup =</span> <span class="st">"disease_state"</span>) <span class="sc">+</span></span>
<span id="cb123-20"><a href="#cb123-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">"PCA for"</span>, ct))</span>
<span id="cb123-21"><a href="#cb123-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-22"><a href="#cb123-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb123-23"><a href="#cb123-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: T cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: B cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: PMN</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cDC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: pDC</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cMo</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]
  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-12-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="detecting-and-removing-outliers-using-local-outlier-factor-lof" class="level1">
<h1>4. Detecting and Removing Outliers using Local Outlier Factor (LOF)</h1>
<p>To improve accuracy in PCA space, we applied the Local Outlier Factor (LOF) algorithm to detect and exclude samples that deviate from the local density of their respective cell types.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>pc <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>pc_scores <span class="ot">&lt;-</span> pc<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a><span class="co"># LOF calculation</span></span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a>lof_scores <span class="ot">&lt;-</span> <span class="fu">lof</span>(pc_scores, <span class="at">k =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in lof(pc_scores, k = 10): lof: k is now deprecated. use minPts = 11
instead .</code></pre>
</div>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lof_scores) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dds)</span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize LOF scores</span></span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">sort</span>(lof_scores), <span class="at">type =</span> <span class="st">"h"</span>, <span class="at">main =</span> <span class="st">"Sorted LOF Scores"</span>,</span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"LOF Score"</span>, <span class="at">xlab =</span> <span class="st">"Sample Index"</span>)</span>
<span id="cb144-6"><a href="#cb144-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">quantile</span>(lof_scores, <span class="fl">0.995</span>), <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set a threshold</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(lof_scores, <span class="fl">0.99</span>)</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">names</span>(lof_scores[lof_scores <span class="sc">&gt;</span> threshold])</span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove outlier and replot</span></span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a>dds_clean <span class="ot">&lt;-</span> dds[, <span class="sc">!</span>(<span class="fu">colnames</span>(dds) <span class="sc">%in%</span> outliers)]</span>
<span id="cb145-7"><a href="#cb145-7" aria-hidden="true" tabindex="-1"></a>vsd_clean <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_clean, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-8"><a href="#cb145-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-9"><a href="#cb145-9" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA before cleaning</span></span>
<span id="cb145-10"><a href="#cb145-10" aria-hidden="true" tabindex="-1"></a>vsd_before <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-11"><a href="#cb145-11" aria-hidden="true" tabindex="-1"></a>pca_before <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_before, <span class="at">intgroup =</span> <span class="st">"cell_type"</span>) <span class="sc">+</span></span>
<span id="cb145-12"><a href="#cb145-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">ggtitle</span>(<span class="st">"PCA Before Outlier Removal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA after cleaning</span></span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a>vsd_after <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_clean, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>pca_after <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_after, <span class="at">intgroup =</span> <span class="st">"cell_type"</span>) <span class="sc">+</span></span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">ggtitle</span>(<span class="st">"PCA After Outlier Removal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot side by side</span></span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>pca_before <span class="sc">+</span> pca_after</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cell-type-specific-pca-plots-stratified-by-disease-state-1" class="level1">
<h1>5. Cell Type-Specific PCA Plots Stratified by Disease State</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over all cell types and generate PCA plots by ifn_status</span></span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type)</span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"Processing PCA for: "</span>, ct)</span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset cleaned dds object</span></span>
<span id="cb150-8"><a href="#cb150-8" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> ct]</span>
<span id="cb150-9"><a href="#cb150-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-10"><a href="#cb150-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply minimal filtering (if needed)</span></span>
<span id="cb150-11"><a href="#cb150-11" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_sub) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb150-12"><a href="#cb150-12" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> dds_sub[keep, ]</span>
<span id="cb150-13"><a href="#cb150-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-14"><a href="#cb150-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use no design formula for PCA</span></span>
<span id="cb150-15"><a href="#cb150-15" aria-hidden="true" tabindex="-1"></a>  dds_sub <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">counts</span>(dds_sub),</span>
<span id="cb150-16"><a href="#cb150-16" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> <span class="fu">colData</span>(dds_sub),</span>
<span id="cb150-17"><a href="#cb150-17" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb150-18"><a href="#cb150-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-19"><a href="#cb150-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Transform and PCA</span></span>
<span id="cb150-20"><a href="#cb150-20" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_sub, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb150-21"><a href="#cb150-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-22"><a href="#cb150-22" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd_sub, <span class="at">intgroup =</span> <span class="st">"ifn_status"</span>) <span class="sc">+</span></span>
<span id="cb150-23"><a href="#cb150-23" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">"PCA for"</span>, ct))</span>
<span id="cb150-24"><a href="#cb150-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-25"><a href="#cb150-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb150-26"><a href="#cb150-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: T cells</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: B cells
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: PMN
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cDC
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: pDC
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Processing PCA for: cMo
using ntop=500 top features by variance</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-14-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Principal component analysis reveals disease-related clustering driven by the interferon response, with IFN-positive samples separating clearly from healthy controls and IFN-negative individuals. This indicates that IFN signaling contributes substantially to the transcriptional variability observed across different immune cell types in systemic lupus erythematosus.</p>
</section>
<section id="differential-expression-analysis" class="level1">
<h1>6. Differential Expression Analysis</h1>
<section id="differential-expression-analysis-for-ifnpos-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-ifnpos-vs-hc">6.1 Differential Expression Analysis for IFNpos vs HC</h2>
<p>In this section, we perform differential expression analysis to identify genes that are significantly differentially expressed between interferon-positive (IFNpos) patients and healthy controls (HC) across various immune cell types.</p>
<section id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-hc">6.1.1 Differential Expression Analysis for cMo Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>cell_type_cMo <span class="ot">&lt;-</span> <span class="st">"cMo"</span></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cMo cells</span></span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_cMo]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>keep_cMo <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_cMo) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> dds_cMo[keep_cMo, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>cts_cMo <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_cMo)</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>coldata_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cMo))</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>coldata_cMo<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cMo<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cMo,</span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cMo,</span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 673 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>res_cMo <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cMo, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a>res_cMo <span class="ot">&lt;-</span> res_cMo[<span class="fu">order</span>(res_cMo<span class="sc">$</span>padj), ]</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 23365 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1049, 4.5%
LFC &lt; 0 (down)     : 666, 2.9%
outliers [1]       : 0, 0%
low counts [2]     : 1812, 7.8%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27     1601.10        8.16377  0.558304   14.6225 2.01987e-48 4.35344e-44
EIF2AK2   1371.42        1.60031  0.111775   14.3172 1.70912e-46 1.84184e-42
ISG15     2959.12        2.50801  0.177266   14.1483 1.91351e-45 1.37473e-41
IFI6      7850.30        2.69260  0.197190   13.6548 1.89012e-42 1.01844e-38
IFITM1    2738.47        2.89249  0.212503   13.6115 3.42048e-42 1.47443e-38
LY6E      6767.12        2.47251  0.186137   13.2833 2.89459e-40 1.03978e-36</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>resLFC_cMo <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cMo,</span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-cmo-cells" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cmo-cells">6.1.1.1 Exporting results for cMo Cells</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>res_sig_cMo <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cMo, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cMo), <span class="at">file =</span> <span class="st">"results/DEG_significant_cMo_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes">6.1.1.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cMo, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in cMo"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-hc">6.1.1.3 Top 10 Differentially Expressed Genes in cMo (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cMo[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cMo<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27     1601.09952        8.16377  0.558304  14.62245 2.01987e-48 4.35344e-44
OTOF       274.27434        6.67924  0.724277   9.22194 2.91779e-20 9.38615e-18
LINC00487    8.53610        5.87209  1.298883   4.52088 6.15841e-06 3.71799e-04
RBP4         5.27853        5.16154  1.160490   4.44772 8.67867e-06 4.98803e-04
MS4A1        4.09053        4.97487  1.686323   2.95013 3.17640e-03 5.35269e-02
KCNMA1       4.00869        4.63947  1.217596   3.81035 1.38771e-04 5.00992e-03
ADAMTS2     89.31312        4.26389  0.670728   6.35710 2.05604e-10 3.01454e-08
IGHG2        2.88288        4.17064  1.197272   3.48346 4.94985e-04 1.33532e-02
STAP1       15.04623        4.14981  0.980391   4.23281 2.30788e-05 1.14877e-03
NCAPGP1      2.96550        3.64041  1.370619   2.65603 7.90664e-03 9.95977e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot">6.1.1.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo)</span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>res_df_cMo<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo)</span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> res_df_cMo[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb182-5"><a href="#cb182-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-6"><a href="#cb182-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cMo,</span>
<span id="cb182-7"><a href="#cb182-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cMo<span class="sc">$</span>gene,                </span>
<span id="cb182-8"><a href="#cb182-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb182-9"><a href="#cb182-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb182-10"><a href="#cb182-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb182-11"><a href="#cb182-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb182-12"><a href="#cb182-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in cMo'</span>,</span>
<span id="cb182-13"><a href="#cb182-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb182-14"><a href="#cb182-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb182-15"><a href="#cb182-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb182-16"><a href="#cb182-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb182-17"><a href="#cb182-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb182-18"><a href="#cb182-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb182-19"><a href="#cb182-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb182-20"><a href="#cb182-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb182-21"><a href="#cb182-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb182-22"><a href="#cb182-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb182-23"><a href="#cb182-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb182-24"><a href="#cb182-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb182-25"><a href="#cb182-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1701 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes">6.1.1.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>vsd_cMo <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_cMo, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_cMo)</span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo)</span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a>res_df_cMo <span class="ot">&lt;-</span> res_df_cMo[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo<span class="sc">$</span>padj), ]</span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a>res_df_cMo<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo)</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo[<span class="fu">order</span>(res_df_cMo<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cMo)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb184-23"><a href="#cb184-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb184-24"><a href="#cb184-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb184-25"><a href="#cb184-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb184-26"><a href="#cb184-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in cMo (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-hc">6.1.2 Differential Expression Analysis for cDC Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a>cell_type_cDC <span class="ot">&lt;-</span> <span class="st">"cDC"</span></span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_cDC]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb186-2"><a href="#cb186-2" aria-hidden="true" tabindex="-1"></a>keep_cDC <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_cDC) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb186-3"><a href="#cb186-3" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> dds_cDC[keep_cDC, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>cts_cDC <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_cDC)</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>coldata_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cDC))</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-5"><a href="#cb187-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb187-6"><a href="#cb187-6" aria-hidden="true" tabindex="-1"></a>coldata_cDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb187-7"><a href="#cb187-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-8"><a href="#cb187-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb187-9"><a href="#cb187-9" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cDC,</span>
<span id="cb187-10"><a href="#cb187-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cDC,</span>
<span id="cb187-11"><a href="#cb187-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb188-2"><a href="#cb188-2" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 102 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a>res_cDC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>res_cDC <span class="ot">&lt;-</span> res_cDC[<span class="fu">order</span>(res_cDC<span class="sc">$</span>padj), ]</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 18625 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1529, 8.2%
LFC &lt; 0 (down)     : 1310, 7%
outliers [1]       : 0, 0%
low counts [2]     : 2889, 16%
(mean count &lt; 9)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFIT1    688.106        3.77915  0.293633   12.8703 6.61310e-38 1.04064e-33
IFITM1  3325.129        3.44492  0.271635   12.6822 7.42639e-37 5.84308e-33
ISG15   1460.197        3.05677  0.260789   11.7212 9.92123e-32 5.20401e-28
IFI6    2847.881        3.60185  0.311729   11.5544 7.01290e-31 2.75888e-27
IFI27   1272.994        9.01383  0.791256   11.3918 4.59382e-30 1.44577e-26
IFIT3   1268.449        3.89440  0.344651   11.2995 1.31915e-29 3.45970e-26</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>resLFC_cDC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cDC,</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-cdc-cells" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cdc-cells">6.1.2.1 Exporting results for cDC Cells</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>res_sig_cDC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cDC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cDC), <span class="at">file =</span> <span class="st">"results/DEG_significant_cDC_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-1" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-1">6.1.2.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cDC, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in cDC"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-hc">6.1.2.3 Top 10 Differentially Expressed Genes in cDC (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cDC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cDC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
           baseMean log2FoldChange     lfcSE      stat      pvalue        padj
          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A     101.84935       14.28602  2.848197   5.01581 5.28103e-07 4.77599e-05
IFI27    1272.99387        9.01383  0.791256  11.39180 4.59382e-30 1.44577e-26
OTOF       50.04154        7.67626  1.401762   5.47615 4.34676e-08 5.56102e-06
IFI44L   1998.83463        5.09890  0.483852  10.53815 5.76208e-26 9.06721e-23
APOBEC3A  380.21260        4.97504  0.670576   7.41904 1.17969e-13 3.63991e-11
USP18     198.72203        4.64458  0.427297  10.86968 1.60759e-27 3.16213e-24
ETV7       43.93760        4.55012  0.722553   6.29728 3.02907e-10 6.11096e-08
USP41      31.90146        4.30186  0.476438   9.02922 1.72900e-19 1.29559e-16
TNFAIP6     9.92146        4.24865  0.888660   4.78096 1.74461e-06 1.26512e-04
IFIT3    1268.44887        3.89440  0.344651  11.29953 1.31915e-29 3.45970e-26</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-1" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-1">6.1.2.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC)</span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a>res_df_cDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC)</span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb209-5"><a href="#cb209-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-6"><a href="#cb209-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cDC,</span>
<span id="cb209-7"><a href="#cb209-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cDC<span class="sc">$</span>gene,                </span>
<span id="cb209-8"><a href="#cb209-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb209-9"><a href="#cb209-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb209-10"><a href="#cb209-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb209-11"><a href="#cb209-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb209-12"><a href="#cb209-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in cDC'</span>,</span>
<span id="cb209-13"><a href="#cb209-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb209-14"><a href="#cb209-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb209-15"><a href="#cb209-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb209-16"><a href="#cb209-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb209-17"><a href="#cb209-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb209-18"><a href="#cb209-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb209-19"><a href="#cb209-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb209-20"><a href="#cb209-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb209-21"><a href="#cb209-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb209-22"><a href="#cb209-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb209-23"><a href="#cb209-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb209-24"><a href="#cb209-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb209-25"><a href="#cb209-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2821 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-1" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-1">6.1.2.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb211-2"><a href="#cb211-2" aria-hidden="true" tabindex="-1"></a>vsd_cDC <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_cDC, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb211-3"><a href="#cb211-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-4"><a href="#cb211-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb211-5"><a href="#cb211-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_cDC)</span>
<span id="cb211-6"><a href="#cb211-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-7"><a href="#cb211-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb211-8"><a href="#cb211-8" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC)</span>
<span id="cb211-9"><a href="#cb211-9" aria-hidden="true" tabindex="-1"></a>res_df_cDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC<span class="sc">$</span>padj), ]</span>
<span id="cb211-10"><a href="#cb211-10" aria-hidden="true" tabindex="-1"></a>res_df_cDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC)</span>
<span id="cb211-11"><a href="#cb211-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-12"><a href="#cb211-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC[<span class="fu">order</span>(res_df_cDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb211-13"><a href="#cb211-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-14"><a href="#cb211-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb211-15"><a href="#cb211-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb211-16"><a href="#cb211-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-17"><a href="#cb211-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_cDC)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb211-18"><a href="#cb211-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb211-19"><a href="#cb211-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb211-20"><a href="#cb211-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb211-21"><a href="#cb211-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb211-22"><a href="#cb211-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb211-23"><a href="#cb211-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb211-24"><a href="#cb211-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb211-25"><a href="#cb211-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb211-26"><a href="#cb211-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in cDC (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-b-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-b-cells-ifnpos-vs-hc">6.1.3 Differential Expression Analysis for B Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>cell_type_B_Cells <span class="ot">&lt;-</span> <span class="st">"B cells"</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_B_Cells]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a>keep_B_Cells <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_B_Cells) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb213-3"><a href="#cb213-3" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> dds_B_Cells[keep_B_Cells, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a>cts_B_Cells <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_B_Cells)</span>
<span id="cb214-3"><a href="#cb214-3" aria-hidden="true" tabindex="-1"></a>coldata_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_B_Cells))</span>
<span id="cb214-4"><a href="#cb214-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-5"><a href="#cb214-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb214-6"><a href="#cb214-6" aria-hidden="true" tabindex="-1"></a>coldata_B_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_B_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb214-7"><a href="#cb214-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-8"><a href="#cb214-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb214-9"><a href="#cb214-9" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_B_Cells,</span>
<span id="cb214-10"><a href="#cb214-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_B_Cells,</span>
<span id="cb214-11"><a href="#cb214-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 116 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_B_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells <span class="ot">&lt;-</span> res_B_Cells[<span class="fu">order</span>(res_B_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20081 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1422, 7.1%
LFC &lt; 0 (down)     : 730, 3.6%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
CMPK2    168.043        3.81075  0.268312  14.20267 8.81859e-46 1.77086e-41
IFI44L  1479.816        4.58296  0.377188  12.15034 5.71290e-34 5.73604e-30
IFI44    536.649        3.91298  0.339869  11.51320 1.13204e-30 7.57750e-27
IFIT3    488.133        3.60142  0.314413  11.45441 2.23475e-30 1.12190e-26
RSAD2    187.138        3.90328  0.356874  10.93740 7.63618e-28 3.06684e-24
IFIT1    229.237        3.55396  0.361406   9.83371 8.05946e-23 2.69737e-19</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a>resLFC_B_Cells <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_B_Cells,</span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb230-4"><a href="#cb230-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-b-cells" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-b-cells">6.1.3.1 Exporting results for B Cells</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a>res_sig_B_Cells <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_B_Cells, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_B_Cells), <span class="at">file =</span> <span class="st">"results/DEG_significant_B_Cells_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-2" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-2">6.1.3.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_B_Cells, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in B Cells"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-hc">6.1.3.3 Top 10 Differentially Expressed Genes in B Cells (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_B_Cells[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_B_Cells<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27      254.35770        7.73796  0.983877   7.86476 3.69797e-15 3.37541e-12
SIGLEC1     27.18012        7.02780  1.018920   6.89731 5.29980e-12 3.43308e-09
OTOF         6.11952        6.60454  1.717900   3.84455 1.20776e-04 7.43959e-03
CCL2         5.41444        6.08373  1.222757   4.97542 6.51062e-07 1.63425e-04
LINC00487    4.18743        5.77172  1.000063   5.77135 7.86367e-09 4.04898e-06
FRZB         5.14098       -4.73843  1.209170  -3.91875 8.90102e-05 6.10039e-03
VN1R105P     3.82649        4.61473  1.307863   3.52845 4.18000e-04 1.56569e-02
IFI44L    1479.81593        4.58296  0.377188  12.15034 5.71290e-34 5.73604e-30
GPR84        3.01661        4.21111  1.243101   3.38759 7.05101e-04 2.12600e-02
EMP1         5.84359        4.06424  0.843375   4.81901 1.44270e-06 3.29215e-04</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-2" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-2">6.1.3.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells)</span>
<span id="cb236-3"><a href="#cb236-3" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells)</span>
<span id="cb236-4"><a href="#cb236-4" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> res_df_B_Cells[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb236-5"><a href="#cb236-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-6"><a href="#cb236-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_B_Cells,</span>
<span id="cb236-7"><a href="#cb236-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_B_Cells<span class="sc">$</span>gene,                </span>
<span id="cb236-8"><a href="#cb236-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb236-9"><a href="#cb236-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb236-10"><a href="#cb236-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb236-11"><a href="#cb236-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb236-12"><a href="#cb236-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in B Cells'</span>,</span>
<span id="cb236-13"><a href="#cb236-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb236-14"><a href="#cb236-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb236-15"><a href="#cb236-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb236-16"><a href="#cb236-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb236-17"><a href="#cb236-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb236-18"><a href="#cb236-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb236-19"><a href="#cb236-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb236-20"><a href="#cb236-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb236-21"><a href="#cb236-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb236-22"><a href="#cb236-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb236-23"><a href="#cb236-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb236-24"><a href="#cb236-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb236-25"><a href="#cb236-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2138 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-2" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-2">6.1.3.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>vsd_B_Cells <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_B_Cells, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb238-5"><a href="#cb238-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_B_Cells)</span>
<span id="cb238-6"><a href="#cb238-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-7"><a href="#cb238-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb238-8"><a href="#cb238-8" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells)</span>
<span id="cb238-9"><a href="#cb238-9" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb238-10"><a href="#cb238-10" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells)</span>
<span id="cb238-11"><a href="#cb238-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-12"><a href="#cb238-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells[<span class="fu">order</span>(res_df_B_Cells<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb238-13"><a href="#cb238-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-14"><a href="#cb238-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb238-15"><a href="#cb238-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb238-16"><a href="#cb238-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-17"><a href="#cb238-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_B_Cells)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb238-18"><a href="#cb238-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-19"><a href="#cb238-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb238-20"><a href="#cb238-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb238-21"><a href="#cb238-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb238-22"><a href="#cb238-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb238-23"><a href="#cb238-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb238-24"><a href="#cb238-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb238-25"><a href="#cb238-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb238-26"><a href="#cb238-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in B Cells (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-t-cells-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-t-cells-ifnpos-vs-hc">6.1.4 Differential Expression Analysis for T Cells (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a>cell_type_T_Cells <span class="ot">&lt;-</span> <span class="st">"T cells"</span></span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for cDC cells</span></span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_T_Cells]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>keep_T_Cells <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_T_Cells) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> dds_T_Cells[keep_T_Cells, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a>cts_T_Cells <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_T_Cells)</span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>coldata_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_T_Cells))</span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>coldata_T_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_T_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_T_Cells,</span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_T_Cells,</span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 50 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb252-2"><a href="#cb252-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_T_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells <span class="ot">&lt;-</span> res_T_Cells[<span class="fu">order</span>(res_T_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20883 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 949, 4.5%
LFC &lt; 0 (down)     : 518, 2.5%
outliers [1]       : 0, 0%
low counts [2]     : 5264, 25%
(mean count &lt; 13)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI6    2116.879        2.52368  0.200713   12.5736 2.95069e-36 4.60868e-32
OAS3     622.486        1.92040  0.165602   11.5965 4.29171e-31 3.35161e-27
IFI44L  2005.728        4.29735  0.376195   11.4232 3.20170e-30 1.66691e-26
IFI44   1137.650        2.71482  0.241037   11.2631 1.99666e-29 7.79645e-26
IFIT3    346.838        3.31500  0.305503   10.8510 1.97325e-27 6.16404e-24
OAS1     712.529        2.66137  0.245767   10.8288 2.51351e-27 6.54309e-24</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb257-2"><a href="#cb257-2" aria-hidden="true" tabindex="-1"></a>resLFC_T_Cells <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_T_Cells,</span>
<span id="cb257-3"><a href="#cb257-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb257-4"><a href="#cb257-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-t-cells" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-t-cells">6.1.4.1 Exporting results for T Cells</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>res_sig_T_Cells <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_T_Cells, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb259-2"><a href="#cb259-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_T_Cells), <span class="at">file =</span> <span class="st">"results/DEG_significant_T_Cells_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-3" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-3">6.1.4.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_T_Cells, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in T Cells"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-hc">6.1.4.3 Top 10 Differentially Expressed Genes in T Cells (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_T_Cells[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_T_Cells<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27   807.3096        7.04081  0.682009  10.32363 5.50986e-25 6.14704e-22
IFI44L 2005.7283        4.29735  0.376195  11.42321 3.20170e-30 1.66691e-26
OTOF     32.4095        3.54321  0.630762   5.61734 1.93922e-08 5.13369e-06
IFIT3   346.8381        3.31500  0.305503  10.85097 1.97325e-27 6.16404e-24
NRIR     15.4155        3.04216  0.373345   8.14839 3.68799e-16 1.74554e-13
USP41    59.5937        2.98922  0.282578  10.57839 3.75354e-26 6.51406e-23
RSAD2   345.9063        2.92321  0.285344  10.24454 1.25225e-24 1.30392e-21
CMPK2   205.6176        2.86353  0.266171  10.75825 5.41901e-27 1.20914e-23
USP18   376.3179        2.78829  0.273649  10.18928 2.21407e-24 2.16135e-21
IFIT1   267.1082        2.78112  0.262896  10.57880 3.73696e-26 6.51406e-23</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-3" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-3">6.1.4.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells)</span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells)</span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> res_df_T_Cells[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb263-5"><a href="#cb263-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-6"><a href="#cb263-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_T_Cells,</span>
<span id="cb263-7"><a href="#cb263-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_T_Cells<span class="sc">$</span>gene,                </span>
<span id="cb263-8"><a href="#cb263-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb263-9"><a href="#cb263-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb263-10"><a href="#cb263-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb263-11"><a href="#cb263-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb263-12"><a href="#cb263-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in T Cells'</span>,</span>
<span id="cb263-13"><a href="#cb263-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb263-14"><a href="#cb263-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb263-15"><a href="#cb263-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb263-16"><a href="#cb263-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb263-17"><a href="#cb263-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb263-18"><a href="#cb263-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb263-19"><a href="#cb263-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb263-20"><a href="#cb263-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb263-21"><a href="#cb263-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb263-22"><a href="#cb263-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb263-23"><a href="#cb263-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb263-24"><a href="#cb263-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb263-25"><a href="#cb263-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1453 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-3" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-3">6.1.4.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>vsd_T_Cells <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_T_Cells, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb265-3"><a href="#cb265-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-4"><a href="#cb265-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb265-5"><a href="#cb265-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_T_Cells)</span>
<span id="cb265-6"><a href="#cb265-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-7"><a href="#cb265-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb265-8"><a href="#cb265-8" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells)</span>
<span id="cb265-9"><a href="#cb265-9" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells<span class="sc">$</span>padj), ]</span>
<span id="cb265-10"><a href="#cb265-10" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells)</span>
<span id="cb265-11"><a href="#cb265-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-12"><a href="#cb265-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells[<span class="fu">order</span>(res_df_T_Cells<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb265-13"><a href="#cb265-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-14"><a href="#cb265-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb265-15"><a href="#cb265-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb265-16"><a href="#cb265-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-17"><a href="#cb265-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_T_Cells)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb265-18"><a href="#cb265-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-19"><a href="#cb265-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb265-20"><a href="#cb265-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb265-21"><a href="#cb265-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb265-22"><a href="#cb265-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb265-23"><a href="#cb265-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb265-24"><a href="#cb265-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb265-25"><a href="#cb265-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb265-26"><a href="#cb265-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in T Cells (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pdc-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pdc-ifnpos-vs-hc">6.1.5 Differential Expression Analysis for pDC (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb266"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb266-1"><a href="#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb266-2"><a href="#cb266-2" aria-hidden="true" tabindex="-1"></a>cell_type_pDC <span class="ot">&lt;-</span> <span class="st">"pDC"</span></span>
<span id="cb266-3"><a href="#cb266-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-4"><a href="#cb266-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for pDC cells</span></span>
<span id="cb266-5"><a href="#cb266-5" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_pDC]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a>keep_pDC <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_pDC) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb267-3"><a href="#cb267-3" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> dds_pDC[keep_pDC, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb268"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb268-1"><a href="#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb268-2"><a href="#cb268-2" aria-hidden="true" tabindex="-1"></a>cts_pDC <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_pDC)</span>
<span id="cb268-3"><a href="#cb268-3" aria-hidden="true" tabindex="-1"></a>coldata_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_pDC))</span>
<span id="cb268-4"><a href="#cb268-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-5"><a href="#cb268-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb268-6"><a href="#cb268-6" aria-hidden="true" tabindex="-1"></a>coldata_pDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_pDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb268-7"><a href="#cb268-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-8"><a href="#cb268-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb268-9"><a href="#cb268-9" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_pDC,</span>
<span id="cb268-10"><a href="#cb268-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_pDC,</span>
<span id="cb268-11"><a href="#cb268-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb269"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb269-1"><a href="#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb269-2"><a href="#cb269-2" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 78 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a>res_pDC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_pDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb280"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb280-1"><a href="#cb280-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb280-2"><a href="#cb280-2" aria-hidden="true" tabindex="-1"></a>res_pDC <span class="ot">&lt;-</span> res_pDC[<span class="fu">order</span>(res_pDC<span class="sc">$</span>padj), ]</span>
<span id="cb280-3"><a href="#cb280-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 19119 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 454, 2.4%
LFC &lt; 0 (down)     : 354, 1.9%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 0)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb282"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
PARP14   419.5817        3.57201  0.289372   12.3440 5.24719e-35 1.00321e-30
TNFSF10  197.4279        3.06011  0.258398   11.8426 2.35051e-32 2.24697e-28
STAT1   4832.2032        1.95411  0.186362   10.4855 1.00674e-25 6.41594e-22
EIF2AK2 1075.8132        2.18572  0.212010   10.3095 6.38506e-25 2.03460e-21
IFI27    245.1479        7.50725  0.727046   10.3257 5.39294e-25 2.03460e-21
RSAD2     84.3289        3.98667  0.385998   10.3282 5.25159e-25 2.03460e-21</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a>resLFC_pDC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_pDC,</span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-pdc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pdc">6.1.5.1 Exporting results for pDC</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb286"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1" aria-hidden="true" tabindex="-1"></a>res_sig_pDC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_pDC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb286-2"><a href="#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_pDC), <span class="at">file =</span> <span class="st">"results/DEG_significant_pDC_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-4" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-4">6.1.5.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb287"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb287-1"><a href="#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_pDC, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in pDC"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-66-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-hc">6.1.5.3 Top 10 Differentially Expressed Genes in pDC (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb288"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb288-1"><a href="#cb288-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_pDC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_pDC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
              baseMean log2FoldChange     lfcSE      stat      pvalue
             &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
IL36A        200.67437       10.01732  1.730896   5.78736 7.15002e-09
IFI27        245.14788        7.50725  0.727046  10.32569 5.39294e-25
C1GALT1P3      5.44997        5.65935  1.894706   2.98693 2.81796e-03
USP18        410.60234        5.12661  0.587139   8.73152 2.51274e-18
USP41         67.68563        4.98772  0.580768   8.58815 8.83797e-18
AC005780.1     4.00141       -4.97312  1.045627  -4.75611 1.97356e-06
KDM5D         23.26211        4.97233  0.914887   5.43491 5.48231e-08
LGALS3BP      69.19806        4.94647  0.531916   9.29935 1.41312e-20
NRIR           9.60052        4.73858  0.782365   6.05674 1.38905e-09
RP11-640L9.2   2.43023        4.64686  1.057428   4.39449 1.11034e-05
                    padj
               &lt;numeric&gt;
IL36A        1.64700e-06
IFI27        2.03460e-21
C1GALT1P3    7.90704e-02
USP18        2.40205e-15
USP41        7.68060e-15
AC005780.1   2.45016e-04
KDM5D        1.04816e-05
LGALS3BP     2.25145e-17
NRIR         3.84889e-07
RP11-640L9.2 1.12321e-03</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-4" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-4">6.1.5.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb290"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb290-2"><a href="#cb290-2" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC)</span>
<span id="cb290-3"><a href="#cb290-3" aria-hidden="true" tabindex="-1"></a>res_df_pDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC)</span>
<span id="cb290-4"><a href="#cb290-4" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> res_df_pDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb290-5"><a href="#cb290-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-6"><a href="#cb290-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_pDC,</span>
<span id="cb290-7"><a href="#cb290-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_pDC<span class="sc">$</span>gene,                </span>
<span id="cb290-8"><a href="#cb290-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb290-9"><a href="#cb290-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb290-10"><a href="#cb290-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb290-11"><a href="#cb290-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb290-12"><a href="#cb290-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in pDC'</span>,</span>
<span id="cb290-13"><a href="#cb290-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb290-14"><a href="#cb290-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb290-15"><a href="#cb290-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb290-16"><a href="#cb290-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb290-17"><a href="#cb290-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb290-18"><a href="#cb290-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb290-19"><a href="#cb290-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb290-20"><a href="#cb290-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb290-21"><a href="#cb290-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb290-22"><a href="#cb290-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb290-23"><a href="#cb290-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb290-24"><a href="#cb290-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb290-25"><a href="#cb290-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 787 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-4" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-4">6.1.5.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb292"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb292-1"><a href="#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb292-2"><a href="#cb292-2" aria-hidden="true" tabindex="-1"></a>vsd_pDC <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_pDC, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb292-3"><a href="#cb292-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-4"><a href="#cb292-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb292-5"><a href="#cb292-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_pDC)</span>
<span id="cb292-6"><a href="#cb292-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-7"><a href="#cb292-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb292-8"><a href="#cb292-8" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC)</span>
<span id="cb292-9"><a href="#cb292-9" aria-hidden="true" tabindex="-1"></a>res_df_pDC <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC<span class="sc">$</span>padj), ]</span>
<span id="cb292-10"><a href="#cb292-10" aria-hidden="true" tabindex="-1"></a>res_df_pDC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC)</span>
<span id="cb292-11"><a href="#cb292-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-12"><a href="#cb292-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC[<span class="fu">order</span>(res_df_pDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb292-13"><a href="#cb292-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-14"><a href="#cb292-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb292-15"><a href="#cb292-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb292-16"><a href="#cb292-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-17"><a href="#cb292-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_pDC)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb292-18"><a href="#cb292-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-19"><a href="#cb292-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb292-20"><a href="#cb292-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb292-21"><a href="#cb292-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb292-22"><a href="#cb292-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>,  </span>
<span id="cb292-23"><a href="#cb292-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb292-24"><a href="#cb292-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb292-25"><a href="#cb292-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb292-26"><a href="#cb292-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in pDC (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pmn-ifnpos-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pmn-ifnpos-vs-hc">6.1.6 Differential Expression Analysis for PMN (IFNpos vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb293"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb293-1"><a href="#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the first cell type</span></span>
<span id="cb293-2"><a href="#cb293-2" aria-hidden="true" tabindex="-1"></a>cell_type_PMN <span class="ot">&lt;-</span> <span class="st">"PMN"</span></span>
<span id="cb293-3"><a href="#cb293-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb293-4"><a href="#cb293-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the cleaned dataset for PMN cells</span></span>
<span id="cb293-5"><a href="#cb293-5" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> dds_clean[, <span class="fu">colData</span>(dds_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type_PMN]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb294"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb294-1"><a href="#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep genes with enough reads</span></span>
<span id="cb294-2"><a href="#cb294-2" aria-hidden="true" tabindex="-1"></a>keep_PMN <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds_PMN) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb294-3"><a href="#cb294-3" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> dds_PMN[keep_PMN, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb295"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb295-1"><a href="#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create new DESeqDataSet</span></span>
<span id="cb295-2"><a href="#cb295-2" aria-hidden="true" tabindex="-1"></a>cts_PMN <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_PMN)</span>
<span id="cb295-3"><a href="#cb295-3" aria-hidden="true" tabindex="-1"></a>coldata_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_PMN))</span>
<span id="cb295-4"><a href="#cb295-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-5"><a href="#cb295-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set IFN status as factor with reference level HC</span></span>
<span id="cb295-6"><a href="#cb295-6" aria-hidden="true" tabindex="-1"></a>coldata_PMN<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_PMN<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"HC"</span>)</span>
<span id="cb295-7"><a href="#cb295-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-8"><a href="#cb295-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Build new DESeq2 object</span></span>
<span id="cb295-9"><a href="#cb295-9" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_PMN,</span>
<span id="cb295-10"><a href="#cb295-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_PMN,</span>
<span id="cb295-11"><a href="#cb295-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb296"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb296-1"><a href="#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2</span></span>
<span id="cb296-2"><a href="#cb296-2" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 230 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb306"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb306-1"><a href="#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs HC</span></span>
<span id="cb306-2"><a href="#cb306-2" aria-hidden="true" tabindex="-1"></a>res_PMN <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_PMN, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"HC"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb307"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb307-1"><a href="#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order and inspect</span></span>
<span id="cb307-2"><a href="#cb307-2" aria-hidden="true" tabindex="-1"></a>res_PMN <span class="ot">&lt;-</span> res_PMN[<span class="fu">order</span>(res_PMN<span class="sc">$</span>padj), ]</span>
<span id="cb307-3"><a href="#cb307-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 17241 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1333, 7.7%
LFC &lt; 0 (down)     : 1212, 7%
outliers [1]       : 0, 0%
low counts [2]     : 335, 1.9%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb309"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb309-1"><a href="#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
SPATS2L   283.842        4.89608  0.386544   12.6663 9.08962e-37 1.53669e-32
IFIT3   26824.278        3.72780  0.300865   12.3903 2.95049e-35 1.66270e-31
IRF7     2198.851        2.46721  0.199080   12.3930 2.85016e-35 1.66270e-31
PARP14   2981.760        2.48630  0.202578   12.2733 1.26063e-34 5.32805e-31
TMEM123  4779.615        2.22638  0.187922   11.8474 2.22082e-32 7.50904e-29
CHMP5    1993.580        1.74595  0.147942   11.8016 3.83169e-32 1.07964e-28</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb311"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb311-1"><a href="#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply shrinkage to log2 fold changes</span></span>
<span id="cb311-2"><a href="#cb311-2" aria-hidden="true" tabindex="-1"></a>resLFC_PMN <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_PMN,</span>
<span id="cb311-3"><a href="#cb311-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_HC"</span>,</span>
<span id="cb311-4"><a href="#cb311-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
</div>
<section id="exporting-results-for-pmn" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pmn">6.1.6.1 Exporting results for PMN</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb313"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb313-1"><a href="#cb313-1" aria-hidden="true" tabindex="-1"></a>res_sig_PMN <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_PMN, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb313-2"><a href="#cb313-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_PMN), <span class="at">file =</span> <span class="st">"results/DEG_significant_PMN_IFNpos_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-5" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-5">6.1.6.2 MA-plot of shrunken log2 fold changes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb314"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb314-1"><a href="#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_PMN, <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs HC in PMN"</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-77-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-hc">6.1.6.3 Top 10 Differentially Expressed Genes in PMN (IFNpos vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb315"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb315-1"><a href="#cb315-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_PMN[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_PMN<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs HC 
Wald test p-value: ifn status IFNpos vs HC 
DataFrame with 10 rows and 6 columns
                baseMean log2FoldChange     lfcSE      stat      pvalue
               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
OTOF           130.96602       10.98273  1.010050  10.87345 1.54248e-27
IFI27          317.09943        9.62776  0.909950  10.58055 3.66817e-26
SIGLEC1         61.42598        8.34894  0.898786   9.28913 1.55557e-20
LY6E          9047.49266        6.81384  0.650138  10.48060 1.06067e-25
RP11-288L9.1     7.14567        6.53839  0.900625   7.25984 3.87544e-13
IFI44L        3549.05385        5.96718  0.645829   9.23956 2.47508e-20
USP18          423.57521        5.72042  0.781464   7.32012 2.47744e-13
HESX1            2.83300        5.64466  1.499583   3.76415 1.67114e-04
RSAD2        11379.20506        5.61395  0.543520  10.32887 5.21700e-25
OAS2          1926.24454        5.57061  0.671548   8.29518 1.08418e-16
                    padj
               &lt;numeric&gt;
OTOF         2.17310e-24
IFI27        3.87588e-23
SIGLEC1      5.25969e-18
LY6E         9.43775e-23
RP11-288L9.1 5.32668e-11
IFI44L       7.89504e-18
USP18        3.51963e-11
HESX1        3.31212e-03
RSAD2        3.67494e-22
OAS2         2.32015e-14</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-5" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-5">6.1.6.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb317"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb317-1"><a href="#cb317-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb317-2"><a href="#cb317-2" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN)</span>
<span id="cb317-3"><a href="#cb317-3" aria-hidden="true" tabindex="-1"></a>res_df_PMN<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN)</span>
<span id="cb317-4"><a href="#cb317-4" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> res_df_pDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb317-5"><a href="#cb317-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb317-6"><a href="#cb317-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_PMN,</span>
<span id="cb317-7"><a href="#cb317-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_PMN<span class="sc">$</span>gene,                </span>
<span id="cb317-8"><a href="#cb317-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,                  </span>
<span id="cb317-9"><a href="#cb317-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,                           </span>
<span id="cb317-10"><a href="#cb317-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,                         </span>
<span id="cb317-11"><a href="#cb317-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,                         </span>
<span id="cb317-12"><a href="#cb317-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos vs HC in PMN'</span>,</span>
<span id="cb317-13"><a href="#cb317-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb317-14"><a href="#cb317-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb317-15"><a href="#cb317-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb317-16"><a href="#cb317-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb317-17"><a href="#cb317-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb317-18"><a href="#cb317-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb317-19"><a href="#cb317-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb317-20"><a href="#cb317-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb317-21"><a href="#cb317-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb317-22"><a href="#cb317-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb317-23"><a href="#cb317-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb317-24"><a href="#cb317-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span></span>
<span id="cb317-25"><a href="#cb317-25" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2776 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-79-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-of-top-differentially-expressed-genes-5" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-of-top-differentially-expressed-genes-5">6.1.6.5 Heatmap of top differentially expressed genes</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb319"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb319-1"><a href="#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform with vst (variance stabilizing transformation)</span></span>
<span id="cb319-2"><a href="#cb319-2" aria-hidden="true" tabindex="-1"></a>vsd_PMN <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_PMN, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb319-3"><a href="#cb319-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-4"><a href="#cb319-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract expression matrix</span></span>
<span id="cb319-5"><a href="#cb319-5" aria-hidden="true" tabindex="-1"></a>expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_PMN)</span>
<span id="cb319-6"><a href="#cb319-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-7"><a href="#cb319-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 20 significant genes by padj</span></span>
<span id="cb319-8"><a href="#cb319-8" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN)</span>
<span id="cb319-9"><a href="#cb319-9" aria-hidden="true" tabindex="-1"></a>res_df_PMN <span class="ot">&lt;-</span> res_df_cDC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN<span class="sc">$</span>padj), ]</span>
<span id="cb319-10"><a href="#cb319-10" aria-hidden="true" tabindex="-1"></a>res_df_PMN<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN)</span>
<span id="cb319-11"><a href="#cb319-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-12"><a href="#cb319-12" aria-hidden="true" tabindex="-1"></a>top20_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC[<span class="fu">order</span>(res_df_pDC<span class="sc">$</span>padj), ])[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb319-13"><a href="#cb319-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-14"><a href="#cb319-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset expression matrix</span></span>
<span id="cb319-15"><a href="#cb319-15" aria-hidden="true" tabindex="-1"></a>mat_top20 <span class="ot">&lt;-</span> expr_mat[top20_genes, ]</span>
<span id="cb319-16"><a href="#cb319-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-17"><a href="#cb319-17" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_PMN)[, <span class="st">"ifn_status"</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb319-18"><a href="#cb319-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-19"><a href="#cb319-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb319-20"><a href="#cb319-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(mat_top20,</span>
<span id="cb319-21"><a href="#cb319-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno,</span>
<span id="cb319-22"><a href="#cb319-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">"row"</span>, </span>
<span id="cb319-23"><a href="#cb319-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">8</span>,</span>
<span id="cb319-24"><a href="#cb319-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb319-25"><a href="#cb319-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">"correlation"</span>,</span>
<span id="cb319-26"><a href="#cb319-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"DE Genes in PMN (IFNpos vs HC)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="differential-expression-analysis-for-ifnneg-vs-hc" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-ifnneg-vs-hc">6.2 Differential Expression Analysis for IFNneg vs HC</h2>
<p>In this section, we perform differential expression analysis to identify genes that are significantly differentially expressed between interferon-negative (IFNpos) patients and healthy controls (HC) across various immune cell types.</p>
<section id="differential-expression-analysis-for-cmo-cells-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cmo-cells-ifnneg-vs-hc">6.2.1 Differential Expression Analysis for cMo Cells (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb320"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb320-1"><a href="#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb320-2"><a href="#cb320-2" aria-hidden="true" tabindex="-1"></a>res_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cMo, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb320-3"><a href="#cb320-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb320-4"><a href="#cb320-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb320-5"><a href="#cb320-5" aria-hidden="true" tabindex="-1"></a>res_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> res_cMo_IFNneg_vs_HC[<span class="fu">order</span>(res_cMo_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb320-6"><a href="#cb320-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb320-7"><a href="#cb320-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb320-8"><a href="#cb320-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cMo_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 23365 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 18, 0.077%
LFC &lt; 0 (down)     : 13, 0.056%
outliers [1]       : 0, 0%
low counts [2]     : 1359, 5.8%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb322"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb322-1"><a href="#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cMo_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue       padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;
ADAMTS2   89.3131       3.386145 0.6352283   5.33060 9.78913e-08 0.00215420
MYADM   1989.5331       0.561328 0.1136306   4.93994 7.81479e-07 0.00859861
DUSP6   7530.3104       0.636587 0.1328501   4.79177 1.65318e-06 0.00987081
EIF1B    825.8077       0.374432 0.0784099   4.77532 1.79420e-06 0.00987081
IGF2R    540.6360       0.442790 0.0966145   4.58306 4.58220e-06 0.02016719
MIR24-2  427.3148       0.901304 0.2004823   4.49568 6.93478e-06 0.02269228</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb324"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb324-1"><a href="#cb324-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb324-2"><a href="#cb324-2" aria-hidden="true" tabindex="-1"></a>resLFC_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cMo,</span>
<span id="cb324-3"><a href="#cb324-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb324-4"><a href="#cb324-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb326"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb326-1"><a href="#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb326-2"><a href="#cb326-2" aria-hidden="true" tabindex="-1"></a>res_sig_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cMo_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-cmo-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cmo-cells-ifnneg-vs-hc">6.2.1.1 Exporting results for cMo Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb327"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb327-1"><a href="#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb327-2"><a href="#cb327-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cMo_IFNneg_vs_HC), </span>
<span id="cb327-3"><a href="#cb327-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_cMo_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc">6.2.1.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb328"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb328-1"><a href="#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cMo_IFNneg_vs_HC,</span>
<span id="cb328-2"><a href="#cb328-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in cMo"</span>,</span>
<span id="cb328-3"><a href="#cb328-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-84-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cmo-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cmo-ifnneg-vs-hc">6.2.1.3 Top 10 Differentially Expressed Genes in cMo (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb329"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb329-1"><a href="#cb329-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cMo_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cMo_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 10 rows and 6 columns
              baseMean log2FoldChange     lfcSE      stat      pvalue      padj
             &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
ADAMTS2       89.31312       3.386145  0.635228   5.33060 9.78913e-08 0.0021542
PPP2R2B        2.25227       2.120025  0.554835   3.82100 1.32910e-04 0.0991050
GRAMD1C        6.23128       1.740362  0.411890   4.22531 2.38612e-05 0.0350060
FAM229B        2.18830      -1.643765  0.391445  -4.19923 2.67828e-05 0.0368363
RN7SL368P     29.12953      -1.186529  0.283916  -4.17916 2.92586e-05 0.0378743
RBKS          55.43416      -1.147650  0.257338  -4.45969 8.20779e-06 0.0226923
RP5-858L17.1   9.82502       1.146711  0.296603   3.86614 1.10570e-04 0.0935845
MTCO1P40     709.82125      -1.007199  0.235038  -4.28526 1.82523e-05 0.0334717
MIR24-2      427.31476       0.901304  0.200482   4.49568 6.93478e-06 0.0226923
RHOB         540.48661      -0.851456  0.223072  -3.81696 1.35106e-04 0.0991050</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-6" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-6">6.2.1.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb331"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb331-1"><a href="#cb331-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb331-2"><a href="#cb331-2" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo_IFNneg_vs_HC)</span>
<span id="cb331-3"><a href="#cb331-3" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo_IFNneg_vs_HC)</span>
<span id="cb331-4"><a href="#cb331-4" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_cMo_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb331-5"><a href="#cb331-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb331-6"><a href="#cb331-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cMo_IFNneg_vs_HC,</span>
<span id="cb331-7"><a href="#cb331-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cMo_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb331-8"><a href="#cb331-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb331-9"><a href="#cb331-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb331-10"><a href="#cb331-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb331-11"><a href="#cb331-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb331-12"><a href="#cb331-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in cMo'</span>,</span>
<span id="cb331-13"><a href="#cb331-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb331-14"><a href="#cb331-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb331-15"><a href="#cb331-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb331-16"><a href="#cb331-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb331-17"><a href="#cb331-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb331-18"><a href="#cb331-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb331-19"><a href="#cb331-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb331-20"><a href="#cb331-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb331-21"><a href="#cb331-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb331-22"><a href="#cb331-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb331-23"><a href="#cb331-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb331-24"><a href="#cb331-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 21 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-86-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-cdc-cells-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cdc-cells-ifnneg-vs-hc">6.2.2 Differential Expression Analysis for cDC Cells (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb333"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb333-1"><a href="#cb333-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb333-2"><a href="#cb333-2" aria-hidden="true" tabindex="-1"></a>res_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb333-3"><a href="#cb333-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb333-4"><a href="#cb333-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb333-5"><a href="#cb333-5" aria-hidden="true" tabindex="-1"></a>res_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> res_cDC_IFNneg_vs_HC[<span class="fu">order</span>(res_cDC_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb333-6"><a href="#cb333-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb333-7"><a href="#cb333-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb333-8"><a href="#cb333-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cDC_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 18625 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 35, 0.19%
LFC &lt; 0 (down)     : 93, 0.5%
outliers [1]       : 0, 0%
low counts [2]     : 13721, 74%
(mean count &lt; 293)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb335"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb335-1"><a href="#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cDC_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
          baseMean log2FoldChange     lfcSE      stat      pvalue      padj
         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
TMEM261    295.942       0.455627 0.1028862   4.42845 9.49123e-06 0.0232725
TOMM22    1077.611       0.261988 0.0581670   4.50407 6.66646e-06 0.0232725
TLK1       831.273      -0.459340 0.1091860  -4.20695 2.58842e-05 0.0317341
TOB1      2263.746      -0.761976 0.1785421  -4.26776 1.97442e-05 0.0317341
ATXN7L3B   710.407      -0.383541 0.0960882  -3.99155 6.56437e-05 0.0366542
BCL11A     802.557      -0.558920 0.1404910  -3.97833 6.94005e-05 0.0366542</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb337"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb337-1"><a href="#cb337-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb337-2"><a href="#cb337-2" aria-hidden="true" tabindex="-1"></a>resLFC_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cDC,</span>
<span id="cb337-3"><a href="#cb337-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb337-4"><a href="#cb337-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb339"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb339-1"><a href="#cb339-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb339-2"><a href="#cb339-2" aria-hidden="true" tabindex="-1"></a>res_sig_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cDC_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-cdc-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cdc-cells-ifnneg-vs-hc">6.2.2.1 Exporting results for cDC Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb340"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb340-1"><a href="#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb340-2"><a href="#cb340-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cDC_IFNneg_vs_HC), </span>
<span id="cb340-3"><a href="#cb340-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_cDC_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-1" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-1">6.2.2.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb341"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb341-1"><a href="#cb341-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cDC_IFNneg_vs_HC,</span>
<span id="cb341-2"><a href="#cb341-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in cDC"</span>,</span>
<span id="cb341-3"><a href="#cb341-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-90-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cdc-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cdc-ifnneg-vs-hc">6.2.2.3 Top 10 Differentially Expressed Genes in cDC (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb342"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb342-1"><a href="#cb342-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cDC_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cDC_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 10 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue      padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
TAPBP    3228.650      -0.783863  0.192403  -4.07407 4.61981e-05 0.0366542
TOB1     2263.746      -0.761976  0.178542  -4.26776 1.97442e-05 0.0317341
ETS2     1841.674      -0.714411  0.177141  -4.03301 5.50674e-05 0.0366542
EIF4G1    747.102      -0.691823  0.202212  -3.42128 6.23263e-04 0.0795757
THEMIS2  2898.306      -0.682240  0.172254  -3.96066 7.47435e-05 0.0366542
CD52     6427.546       0.667926  0.220819   3.02477 2.48824e-03 0.0995194
BRD3      329.014      -0.653534  0.171489  -3.81094 1.38437e-04 0.0473017
SEMA4A   2661.197      -0.636192  0.166189  -3.82813 1.29123e-04 0.0473017
ARHGAP1   688.056      -0.611019  0.186027  -3.28457 1.02136e-03 0.0902210
PLBD2     394.616      -0.604826  0.200633  -3.01459 2.57330e-03 0.0995194</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-7" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-7">6.2.2.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb344"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb344-1"><a href="#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb344-2"><a href="#cb344-2" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC_IFNneg_vs_HC)</span>
<span id="cb344-3"><a href="#cb344-3" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC_IFNneg_vs_HC)</span>
<span id="cb344-4"><a href="#cb344-4" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_cDC_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb344-5"><a href="#cb344-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb344-6"><a href="#cb344-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cDC_IFNneg_vs_HC,</span>
<span id="cb344-7"><a href="#cb344-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cDC_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb344-8"><a href="#cb344-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb344-9"><a href="#cb344-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb344-10"><a href="#cb344-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb344-11"><a href="#cb344-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb344-12"><a href="#cb344-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in cDC'</span>,</span>
<span id="cb344-13"><a href="#cb344-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb344-14"><a href="#cb344-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb344-15"><a href="#cb344-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb344-16"><a href="#cb344-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb344-17"><a href="#cb344-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb344-18"><a href="#cb344-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb344-19"><a href="#cb344-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb344-20"><a href="#cb344-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb344-21"><a href="#cb344-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb344-22"><a href="#cb344-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb344-23"><a href="#cb344-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb344-24"><a href="#cb344-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 127 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-92-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-b-cells-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-b-cells-ifnneg-vs-hc">6.2.3 Differential Expression Analysis for B Cells (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb346"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb346-1"><a href="#cb346-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb346-2"><a href="#cb346-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_B_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb346-3"><a href="#cb346-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-4"><a href="#cb346-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb346-5"><a href="#cb346-5" aria-hidden="true" tabindex="-1"></a>res_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> res_B_Cells_IFNneg_vs_HC[<span class="fu">order</span>(res_B_Cells_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb346-6"><a href="#cb346-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-7"><a href="#cb346-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb346-8"><a href="#cb346-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_B_Cells_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20081 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 47, 0.23%
LFC &lt; 0 (down)     : 6, 0.03%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb348"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb348-1"><a href="#cb348-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_B_Cells_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
          baseMean log2FoldChange     lfcSE      stat      pvalue        padj
         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
GSTM1      54.6197       3.283811  0.353131   9.29913 1.41593e-20 2.84333e-16
GSTM2      80.3298       1.079795  0.193765   5.57271 2.50810e-08 2.51826e-04
GPAT3      23.1154       2.839002  0.558383   5.08433 3.68921e-07 2.46943e-03
ATP6V0E2  167.4293      -0.718782  0.153367  -4.68669 2.77663e-06 1.39394e-02
GIMAP7    119.6212       2.170722  0.469727   4.62125 3.81443e-06 1.53195e-02
PLEKHG3    18.0420       1.363192  0.298479   4.56713 4.94449e-06 1.65484e-02</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb350"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb350-1"><a href="#cb350-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb350-2"><a href="#cb350-2" aria-hidden="true" tabindex="-1"></a>resLFC_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_B_Cells,</span>
<span id="cb350-3"><a href="#cb350-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb350-4"><a href="#cb350-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb352"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb352-1"><a href="#cb352-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb352-2"><a href="#cb352-2" aria-hidden="true" tabindex="-1"></a>res_sig_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_B_Cells_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-b-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-b-cells-ifnneg-vs-hc">6.2.3.1 Exporting results for B Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb353"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb353-1"><a href="#cb353-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb353-2"><a href="#cb353-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_B_Cells_IFNneg_vs_HC), </span>
<span id="cb353-3"><a href="#cb353-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_B_Cells_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-2" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-2">6.2.3.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb354"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb354-1"><a href="#cb354-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_B_Cells_IFNneg_vs_HC,</span>
<span id="cb354-2"><a href="#cb354-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in B Cells"</span>,</span>
<span id="cb354-3"><a href="#cb354-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-96-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-b-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-b-cells-ifnneg-vs-hc">6.2.3.3 Top 10 Differentially Expressed Genes in B Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb355"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb355-1"><a href="#cb355-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_B_Cells_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_B_Cells_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 10 rows and 6 columns
          baseMean log2FoldChange     lfcSE      stat      pvalue        padj
         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A    117.48979       11.48314  2.670188   4.30050 1.70414e-05 3.42207e-02
STON2      3.79581        4.43217  1.111532   3.98745 6.67882e-05 6.61849e-02
DLEU7      3.39385        4.17314  1.057640   3.94571 7.95628e-05 6.61849e-02
SEMA3C     7.84296        3.46105  0.781083   4.43109 9.37595e-06 2.40427e-02
LGALS12    8.39750        3.31158  0.876472   3.77831 1.57898e-04 8.56961e-02
GSTM1     54.61974        3.28381  0.353131   9.29913 1.41593e-20 2.84333e-16
ARHGAP10   7.00551        3.00616  0.770506   3.90154 9.55809e-05 6.61849e-02
CES1      46.74957        2.98888  0.777640   3.84353 1.21278e-04 7.85610e-02
GPAT3     23.11542        2.83900  0.558383   5.08433 3.68921e-07 2.46943e-03
FRMD4B     8.42800        2.72120  0.713018   3.81646 1.35381e-04 8.23181e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-8" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-8">6.2.3.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb357"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb357-1"><a href="#cb357-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb357-2"><a href="#cb357-2" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells_IFNneg_vs_HC)</span>
<span id="cb357-3"><a href="#cb357-3" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells_IFNneg_vs_HC)</span>
<span id="cb357-4"><a href="#cb357-4" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_B_Cells_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb357-5"><a href="#cb357-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb357-6"><a href="#cb357-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_B_Cells_IFNneg_vs_HC,</span>
<span id="cb357-7"><a href="#cb357-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_B_Cells_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb357-8"><a href="#cb357-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb357-9"><a href="#cb357-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb357-10"><a href="#cb357-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb357-11"><a href="#cb357-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb357-12"><a href="#cb357-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in B Cells'</span>,</span>
<span id="cb357-13"><a href="#cb357-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb357-14"><a href="#cb357-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb357-15"><a href="#cb357-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb357-16"><a href="#cb357-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb357-17"><a href="#cb357-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb357-18"><a href="#cb357-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb357-19"><a href="#cb357-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb357-20"><a href="#cb357-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb357-21"><a href="#cb357-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb357-22"><a href="#cb357-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb357-23"><a href="#cb357-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb357-24"><a href="#cb357-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 48 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-98-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-t-cells-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-t-cells-ifnneg-vs-hc">6.2.4 Differential Expression Analysis for T Cells (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb359"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb359-1"><a href="#cb359-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb359-2"><a href="#cb359-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_T_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb359-3"><a href="#cb359-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb359-4"><a href="#cb359-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb359-5"><a href="#cb359-5" aria-hidden="true" tabindex="-1"></a>res_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> res_T_Cells_IFNneg_vs_HC[<span class="fu">order</span>(res_T_Cells_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb359-6"><a href="#cb359-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb359-7"><a href="#cb359-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb359-8"><a href="#cb359-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_T_Cells_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20883 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2, 0.0096%
LFC &lt; 0 (down)     : 2, 0.0096%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb361"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb361-1"><a href="#cb361-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_T_Cells_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
                baseMean log2FoldChange     lfcSE      stat      pvalue
               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
GSTM1          120.05630       3.238239  0.461119   7.02256 2.17836e-12
NLGN3           13.38667      -1.594066  0.315134  -5.05837 4.22861e-07
GSTM5           13.22514       1.829950  0.367661   4.97727 6.44875e-07
XX-FW83563B9.5  46.19150      -0.801303  0.167044  -4.79695 1.61102e-06
CNTF             3.21368       3.710791  0.918041   4.04207 5.29804e-05
DHCR24          97.38596      -1.022439  0.256845  -3.98076 6.86941e-05
                      padj
                 &lt;numeric&gt;
GSTM1          4.54906e-08
NLGN3          4.41530e-03
GSTM5          4.48898e-03
XX-FW83563B9.5 8.41072e-03
CNTF           2.21278e-01
DHCR24         2.39090e-01</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb363"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb363-1"><a href="#cb363-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb363-2"><a href="#cb363-2" aria-hidden="true" tabindex="-1"></a>resLFC_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_T_Cells,</span>
<span id="cb363-3"><a href="#cb363-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb363-4"><a href="#cb363-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in nbinomGLM(x = x, Y = YNZ, size = size, weights = weightsNZ, offset =
offsetNZ, : the line search routine failed, unable to sufficiently decrease the
function value
Warning in nbinomGLM(x = x, Y = YNZ, size = size, weights = weightsNZ, offset =
offsetNZ, : the line search routine failed, unable to sufficiently decrease the
function value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in nbinomGLM(x = x, Y = YNZ, size = size, weights = weightsNZ, offset =
offsetNZ, : the line search routine failed, unable to sufficiently decrease the
function value</code></pre>
</div>
<div class="sourceCode cell-code" id="cb367"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb367-1"><a href="#cb367-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb367-2"><a href="#cb367-2" aria-hidden="true" tabindex="-1"></a>res_sig_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_T_Cells_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-t-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-t-cells-ifnneg-vs-hc">6.2.4.1 Exporting results for T Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb368"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb368-1"><a href="#cb368-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb368-2"><a href="#cb368-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_T_Cells_IFNneg_vs_HC), </span>
<span id="cb368-3"><a href="#cb368-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_T_Cells_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-3" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-3">6.2.4.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb369"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb369-1"><a href="#cb369-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_T_Cells_IFNneg_vs_HC,</span>
<span id="cb369-2"><a href="#cb369-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in T Cells"</span>,</span>
<span id="cb369-3"><a href="#cb369-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-102-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-t-cells-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-t-cells-ifnneg-vs-hc">6.2.4.3 Top 10 Differentially Expressed Genes in T Cells (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb370"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb370-1"><a href="#cb370-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_T_Cells_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_T_Cells_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 4 rows and 6 columns
                baseMean log2FoldChange     lfcSE      stat      pvalue
               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
GSTM1           120.0563       3.238239  0.461119   7.02256 2.17836e-12
GSTM5            13.2251       1.829950  0.367661   4.97727 6.44875e-07
NLGN3            13.3867      -1.594066  0.315134  -5.05837 4.22861e-07
XX-FW83563B9.5   46.1915      -0.801303  0.167044  -4.79695 1.61102e-06
                      padj
                 &lt;numeric&gt;
GSTM1          4.54906e-08
GSTM5          4.48898e-03
NLGN3          4.41530e-03
XX-FW83563B9.5 8.41072e-03</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-9" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-9">6.2.4.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb372"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb372-1"><a href="#cb372-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb372-2"><a href="#cb372-2" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells_IFNneg_vs_HC)</span>
<span id="cb372-3"><a href="#cb372-3" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells_IFNneg_vs_HC)</span>
<span id="cb372-4"><a href="#cb372-4" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_T_Cells_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb372-5"><a href="#cb372-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-6"><a href="#cb372-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_T_Cells_IFNneg_vs_HC,</span>
<span id="cb372-7"><a href="#cb372-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_T_Cells_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb372-8"><a href="#cb372-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb372-9"><a href="#cb372-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb372-10"><a href="#cb372-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb372-11"><a href="#cb372-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb372-12"><a href="#cb372-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in T Cells'</span>,</span>
<span id="cb372-13"><a href="#cb372-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb372-14"><a href="#cb372-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb372-15"><a href="#cb372-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb372-16"><a href="#cb372-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb372-17"><a href="#cb372-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb372-18"><a href="#cb372-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb372-19"><a href="#cb372-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb372-20"><a href="#cb372-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb372-21"><a href="#cb372-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb372-22"><a href="#cb372-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb372-23"><a href="#cb372-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb372-24"><a href="#cb372-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-104-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pdc-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pdc-ifnneg-vs-hc">6.2.5 Differential Expression Analysis for pDC (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb373"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb373-1"><a href="#cb373-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb373-2"><a href="#cb373-2" aria-hidden="true" tabindex="-1"></a>res_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_pDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb373-3"><a href="#cb373-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb373-4"><a href="#cb373-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb373-5"><a href="#cb373-5" aria-hidden="true" tabindex="-1"></a>res_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> res_pDC_IFNneg_vs_HC[<span class="fu">order</span>(res_pDC_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb373-6"><a href="#cb373-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb373-7"><a href="#cb373-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb373-8"><a href="#cb373-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_pDC_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 19119 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 31, 0.16%
LFC &lt; 0 (down)     : 75, 0.39%
outliers [1]       : 0, 0%
low counts [2]     : 4819, 25%
(mean count &lt; 13)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb375"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb375-1"><a href="#cb375-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_pDC_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
          baseMean log2FoldChange     lfcSE      stat      pvalue        padj
         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
GSTM1     161.8400       3.008596 0.4752542   6.33050 2.44372e-10 3.49452e-06
IL36A     200.6744      10.618429 1.7308252   6.13489 8.52161e-10 6.09295e-06
ADGRL1     16.1937      -2.781083 0.5797667  -4.79690 1.61140e-06 7.68102e-03
CDC42EP3   31.7573       1.647111 0.3721727   4.42566 9.61463e-06 2.69092e-02
CSF3R     108.4947       1.154150 0.2703152   4.26965 1.95783e-05 2.69092e-02
EEF2     1535.5608      -0.414938 0.0960719  -4.31903 1.56713e-05 2.69092e-02</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb377"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb377-1"><a href="#cb377-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb377-2"><a href="#cb377-2" aria-hidden="true" tabindex="-1"></a>resLFC_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_pDC,</span>
<span id="cb377-3"><a href="#cb377-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb377-4"><a href="#cb377-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb379"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb379-1"><a href="#cb379-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb379-2"><a href="#cb379-2" aria-hidden="true" tabindex="-1"></a>res_sig_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_pDC_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-pdc-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pdc-ifnneg-vs-hc">6.2.5.1 Exporting results for pDC (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb380"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb380-1"><a href="#cb380-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb380-2"><a href="#cb380-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_pDC_IFNneg_vs_HC), </span>
<span id="cb380-3"><a href="#cb380-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_pDC_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-4" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-4">6.2.5.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb381"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb381-1"><a href="#cb381-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_pDC_IFNneg_vs_HC,</span>
<span id="cb381-2"><a href="#cb381-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in pDC"</span>,</span>
<span id="cb381-3"><a href="#cb381-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-108-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pdc-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pdc-ifnneg-vs-hc">6.2.5.3 Top 10 Differentially Expressed Genes in pDC (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb382"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb382-1"><a href="#cb382-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_pDC_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_pDC_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 10 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A   200.6744       10.61843  1.730825   6.13489 8.52161e-10 6.09295e-06
GSTM1   161.8400        3.00860  0.475254   6.33050 2.44372e-10 3.49452e-06
ADGRL1   16.1937       -2.78108  0.579767  -4.79690 1.61140e-06 7.68102e-03
ASPM     46.2568        2.61048  0.622069   4.19644 2.71138e-05 2.98252e-02
HK3      14.9468        2.54071  0.717050   3.54328 3.95179e-04 7.24494e-02
IL2RB    14.2111        2.53780  0.691760   3.66861 2.43871e-04 6.17193e-02
VCAN     79.7772        2.30718  0.609475   3.78553 1.53383e-04 5.66734e-02
FCN1    195.2286        2.19370  0.526537   4.16628 3.09609e-05 3.05858e-02
GIMAP7   14.1011        1.99729  0.468686   4.26147 2.03086e-05 2.69092e-02
TLR2     14.2851        1.93756  0.491523   3.94196 8.08186e-05 4.50371e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-10" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-10">6.2.5.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb384"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb384-1"><a href="#cb384-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb384-2"><a href="#cb384-2" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC_IFNneg_vs_HC)</span>
<span id="cb384-3"><a href="#cb384-3" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC_IFNneg_vs_HC)</span>
<span id="cb384-4"><a href="#cb384-4" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_pDC_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb384-5"><a href="#cb384-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb384-6"><a href="#cb384-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_pDC_IFNneg_vs_HC,</span>
<span id="cb384-7"><a href="#cb384-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_pDC_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb384-8"><a href="#cb384-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb384-9"><a href="#cb384-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb384-10"><a href="#cb384-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb384-11"><a href="#cb384-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb384-12"><a href="#cb384-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in pDC'</span>,</span>
<span id="cb384-13"><a href="#cb384-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb384-14"><a href="#cb384-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb384-15"><a href="#cb384-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb384-16"><a href="#cb384-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb384-17"><a href="#cb384-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb384-18"><a href="#cb384-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb384-19"><a href="#cb384-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb384-20"><a href="#cb384-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb384-21"><a href="#cb384-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb384-22"><a href="#cb384-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb384-23"><a href="#cb384-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb384-24"><a href="#cb384-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 103 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-110-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pmn-ifnneg-vs-hc" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pmn-ifnneg-vs-hc">6.2.6 Differential Expression Analysis for PMN (IFNneg vs HC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb386"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb386-1"><a href="#cb386-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNneg vs HC</span></span>
<span id="cb386-2"><a href="#cb386-2" aria-hidden="true" tabindex="-1"></a>res_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_PMN, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNneg"</span>, <span class="st">"HC"</span>))</span>
<span id="cb386-3"><a href="#cb386-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-4"><a href="#cb386-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb386-5"><a href="#cb386-5" aria-hidden="true" tabindex="-1"></a>res_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> res_PMN_IFNneg_vs_HC[<span class="fu">order</span>(res_PMN_IFNneg_vs_HC<span class="sc">$</span>padj), ]</span>
<span id="cb386-6"><a href="#cb386-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-7"><a href="#cb386-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb386-8"><a href="#cb386-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_PMN_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 17241 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 18, 0.1%
LFC &lt; 0 (down)     : 15, 0.087%
outliers [1]       : 0, 0%
low counts [2]     : 335, 1.9%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb388"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb388-1"><a href="#cb388-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_PMN_IFNneg_vs_HC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 6 rows and 6 columns
             baseMean log2FoldChange     lfcSE      stat      pvalue
            &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
FOSB         97.35885       1.860304  0.341880   5.44139 5.28667e-08
C16orf54   6164.07769       0.510193  0.112096   4.55138 5.32946e-06
PPP1R16A      8.63263      -1.893807  0.416025  -4.55214 5.31019e-06
AF001548.5   89.37107       1.043330  0.237618   4.39079 1.12940e-05
EGR1        101.46674       2.462043  0.560568   4.39205 1.12286e-05
MTR          16.65358      -2.043840  0.472964  -4.32134 1.55081e-05
                  padj
             &lt;numeric&gt;
FOSB       0.000893764
C16orf54   0.030033269
PPP1R16A   0.030033269
AF001548.5 0.036667080
EGR1       0.036667080
MTR        0.036667080</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb390"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb390-1"><a href="#cb390-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb390-2"><a href="#cb390-2" aria-hidden="true" tabindex="-1"></a>resLFC_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_PMN,</span>
<span id="cb390-3"><a href="#cb390-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNneg_vs_HC"</span>, </span>
<span id="cb390-4"><a href="#cb390-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb392"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb392-1"><a href="#cb392-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb392-2"><a href="#cb392-2" aria-hidden="true" tabindex="-1"></a>res_sig_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_PMN_IFNneg_vs_HC, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-pmn-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pmn-ifnneg-vs-hc">6.2.5.1 Exporting results for PMN (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb393"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb393-1"><a href="#cb393-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb393-2"><a href="#cb393-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_PMN_IFNneg_vs_HC), </span>
<span id="cb393-3"><a href="#cb393-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_PMN_IFNneg_vs_HC.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-5" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnneg-vs-hc-5">6.2.5.2 MA-plot of shrunken log2 fold changes (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb394"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb394-1"><a href="#cb394-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_PMN_IFNneg_vs_HC,</span>
<span id="cb394-2"><a href="#cb394-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNneg vs HC in PMN"</span>,</span>
<span id="cb394-3"><a href="#cb394-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-114-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pmn-ifnneg-vs-hc" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pmn-ifnneg-vs-hc">6.2.5.3 Top 10 Differentially Expressed Genes in PMN (IFNneg vs HC)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb395"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb395-1"><a href="#cb395-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_PMN_IFNneg_vs_HC[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_PMN_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNneg vs HC 
Wald test p-value: ifn status IFNneg vs HC 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue      padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
CR2          5.51780       -5.89313  1.445509  -4.07685 4.56497e-05 0.0580496
RGS16        4.53924       -3.62756  0.899296  -4.03378 5.48871e-05 0.0618615
CTSF         4.30842       -3.36229  0.881155  -3.81577 1.35760e-04 0.0791431
TAPT1-AS1    3.89926       -2.89491  0.678759  -4.26499 1.99908e-05 0.0366671
ZNF831       5.12525       -2.84604  0.691161  -4.11777 3.82556e-05 0.0538957
NKAPL        6.01965        2.84565  0.663079   4.29157 1.77415e-05 0.0366671
AC007950.2  13.84506        2.65930  0.663530   4.00780 6.12865e-05 0.0645659
REPS1       14.89053       -2.53631  0.671719  -3.77585 1.59460e-04 0.0881116
EGR1       101.46674        2.46204  0.560568   4.39205 1.12286e-05 0.0366671
EPB41L5     37.66630        2.17629  0.544869   3.99416 6.49249e-05 0.0645659</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-11" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-11">6.2.5.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb397"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb397-1"><a href="#cb397-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb397-2"><a href="#cb397-2" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN_IFNneg_vs_HC)</span>
<span id="cb397-3"><a href="#cb397-3" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNneg_vs_HC<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN_IFNneg_vs_HC)</span>
<span id="cb397-4"><a href="#cb397-4" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNneg_vs_HC <span class="ot">&lt;-</span> res_df_PMN_IFNneg_vs_HC[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN_IFNneg_vs_HC<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN_IFNneg_vs_HC<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb397-5"><a href="#cb397-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb397-6"><a href="#cb397-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_PMN_IFNneg_vs_HC,</span>
<span id="cb397-7"><a href="#cb397-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_PMN_IFNneg_vs_HC<span class="sc">$</span>gene,</span>
<span id="cb397-8"><a href="#cb397-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb397-9"><a href="#cb397-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb397-10"><a href="#cb397-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb397-11"><a href="#cb397-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb397-12"><a href="#cb397-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNneg vs HC in PMN'</span>,</span>
<span id="cb397-13"><a href="#cb397-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb397-14"><a href="#cb397-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb397-15"><a href="#cb397-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb397-16"><a href="#cb397-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb397-17"><a href="#cb397-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb397-18"><a href="#cb397-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb397-19"><a href="#cb397-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb397-20"><a href="#cb397-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb397-21"><a href="#cb397-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb397-22"><a href="#cb397-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb397-23"><a href="#cb397-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb397-24"><a href="#cb397-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-116-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="differential-expression-analysis-for-ifnpos-vs-ifnneg" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-for-ifnpos-vs-ifnneg">6.3 Differential Expression Analysis for IFNpos vs IFNneg</h2>
<p>In this section, we perform differential expression analysis to identify genes that are significantly differentially expressed between interferon-positive (IFNpos) patients and interferon-negative (IFNneg) across various immune cell types.</p>
<section id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cmo-cells-ifnpos-vs-ifnneg">6.3.1 Differential Expression Analysis for cMo Cells (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb399"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb399-1"><a href="#cb399-1" aria-hidden="true" tabindex="-1"></a>coldata_cMo<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cMo<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb399-2"><a href="#cb399-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb399-3"><a href="#cb399-3" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cMo,</span>
<span id="cb399-4"><a href="#cb399-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cMo,</span>
<span id="cb399-5"><a href="#cb399-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb399-6"><a href="#cb399-6" aria-hidden="true" tabindex="-1"></a>dds_cMo <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cMo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 673 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb409"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb409-1"><a href="#cb409-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb409-2"><a href="#cb409-2" aria-hidden="true" tabindex="-1"></a>res_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cMo, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb409-3"><a href="#cb409-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb409-4"><a href="#cb409-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb409-5"><a href="#cb409-5" aria-hidden="true" tabindex="-1"></a>res_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_cMo_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb409-6"><a href="#cb409-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb409-7"><a href="#cb409-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb409-8"><a href="#cb409-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cMo_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 23365 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1445, 6.2%
LFC &lt; 0 (down)     : 1075, 4.6%
outliers [1]       : 0, 0%
low counts [2]     : 906, 3.9%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb411"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb411-1"><a href="#cb411-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cMo_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
ISG15     2959.12        3.00044 0.1477120   20.3128 9.91050e-92 2.22580e-87
EIF2AK2   1371.42        1.81534 0.0931013   19.4986 1.12835e-84 8.44719e-81
LY6E      6767.12        3.02681 0.1551057   19.5145 8.26601e-85 8.44719e-81
IFI6      7850.30        3.19131 0.1643133   19.4221 5.01828e-84 2.81764e-80
IFI44L    3117.70        3.88749 0.2122436   18.3162 6.15043e-75 2.76265e-71
IFITM1    2738.47        3.24053 0.1770564   18.3023 7.93689e-75 2.97091e-71</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb413"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb413-1"><a href="#cb413-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb413-2"><a href="#cb413-2" aria-hidden="true" tabindex="-1"></a>resLFC_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cMo,</span>
<span id="cb413-3"><a href="#cb413-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb413-4"><a href="#cb413-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb415"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb415-1"><a href="#cb415-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb415-2"><a href="#cb415-2" aria-hidden="true" tabindex="-1"></a>res_sig_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cMo_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-cmo-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cmo-cells-ifnpos-vs-ifnneg">6.3.1.1 Exporting results for cMo Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb416"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb416-1"><a href="#cb416-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb416-2"><a href="#cb416-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cMo_IFNpos_vs_IFNneg), </span>
<span id="cb416-3"><a href="#cb416-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_cMo_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg">6.3.1.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb417"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb417-1"><a href="#cb417-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cMo_IFNpos_vs_IFNneg,</span>
<span id="cb417-2"><a href="#cb417-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in cMo"</span>,</span>
<span id="cb417-3"><a href="#cb417-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-121-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cmo-ifnpos-vs-ifnneg">6.3.1.3 Top 10 Differentially Expressed Genes in cMo (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb418"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb418-1"><a href="#cb418-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cMo_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
                 baseMean log2FoldChange     lfcSE      stat      pvalue
                &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
IFI27          1601.09952        7.27103  0.462979  15.70489 1.40024e-55
OTOF            274.27434        7.00243  0.602672  11.61897 3.30100e-31
PPP1R17           2.43508       -4.43409  1.354606  -3.27334 1.06283e-03
STAP1            15.04623        4.30865  0.813668   5.29534 1.18796e-07
USP18           469.19372        4.26835  0.288760  14.78165 1.92401e-49
GFAP              1.67187       -4.24809  1.520515  -2.79385 5.20843e-03
USP41            83.92032        4.19896  0.277910  15.10906 1.41142e-51
RP11-1260E13.4    1.29876        4.04060  1.365978   2.95803 3.09615e-03
KCTD14            4.98449        3.96408  0.876245   4.52394 6.06990e-06
CCDC81            1.63094       -3.89162  1.248573  -3.11685 1.82794e-03
                      padj
                 &lt;numeric&gt;
IFI27          1.31033e-52
OTOF           9.15273e-29
PPP1R17        1.74489e-02
STAP1          6.58776e-06
USP18          1.23461e-46
GFAP           5.79375e-02
USP41          9.90593e-49
RP11-1260E13.4 3.95843e-02
KCTD14         2.22026e-04
CCDC81         2.64863e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-12" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-12">6.3.1.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb420"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb420-1"><a href="#cb420-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb420-2"><a href="#cb420-2" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cMo_IFNpos_vs_IFNneg)</span>
<span id="cb420-3"><a href="#cb420-3" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cMo_IFNpos_vs_IFNneg)</span>
<span id="cb420-4"><a href="#cb420-4" aria-hidden="true" tabindex="-1"></a>res_df_cMo_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_cMo_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb420-5"><a href="#cb420-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb420-6"><a href="#cb420-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cMo_IFNpos_vs_IFNneg,</span>
<span id="cb420-7"><a href="#cb420-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cMo_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb420-8"><a href="#cb420-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb420-9"><a href="#cb420-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb420-10"><a href="#cb420-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb420-11"><a href="#cb420-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb420-12"><a href="#cb420-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in cMo'</span>,</span>
<span id="cb420-13"><a href="#cb420-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb420-14"><a href="#cb420-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb420-15"><a href="#cb420-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb420-16"><a href="#cb420-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb420-17"><a href="#cb420-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb420-18"><a href="#cb420-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb420-19"><a href="#cb420-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb420-20"><a href="#cb420-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb420-21"><a href="#cb420-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb420-22"><a href="#cb420-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb420-23"><a href="#cb420-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb420-24"><a href="#cb420-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2502 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-123-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-cdc-cells-ifnpos-vs-ifnneg">6.3.2 Differential Expression Analysis for cDC Cells (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb422"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb422-1"><a href="#cb422-1" aria-hidden="true" tabindex="-1"></a>coldata_cDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_cDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb422-2"><a href="#cb422-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-3"><a href="#cb422-3" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_cDC,</span>
<span id="cb422-4"><a href="#cb422-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_cDC,</span>
<span id="cb422-5"><a href="#cb422-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb422-6"><a href="#cb422-6" aria-hidden="true" tabindex="-1"></a>dds_cDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_cDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 102 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb432"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb432-1"><a href="#cb432-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb432-2"><a href="#cb432-2" aria-hidden="true" tabindex="-1"></a>res_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_cDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb432-3"><a href="#cb432-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb432-4"><a href="#cb432-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb432-5"><a href="#cb432-5" aria-hidden="true" tabindex="-1"></a>res_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_cDC_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb432-6"><a href="#cb432-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb432-7"><a href="#cb432-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb432-8"><a href="#cb432-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_cDC_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 18625 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 753, 4%
LFC &lt; 0 (down)     : 625, 3.4%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb434"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb434-1"><a href="#cb434-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_cDC_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFITM1   3325.129        3.52465  0.276566   12.7443 3.35228e-37 6.24363e-33
IFIT1     688.106        3.63790  0.298831   12.1738 4.28912e-34 3.99424e-30
IFI27    1272.994        9.33333  0.809070   11.5359 8.69981e-31 5.40113e-27
IFI6     2847.881        3.47113  0.317330   10.9385 7.54065e-28 3.51112e-24
ISG15    1460.197        2.81220  0.265409   10.5957 3.11916e-26 1.16189e-22
EIF2AK2  1024.439        1.76514  0.167523   10.5367 5.85259e-26 1.81674e-22</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb436"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb436-1"><a href="#cb436-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb436-2"><a href="#cb436-2" aria-hidden="true" tabindex="-1"></a>resLFC_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_cDC,</span>
<span id="cb436-3"><a href="#cb436-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb436-4"><a href="#cb436-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb438"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb438-1"><a href="#cb438-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb438-2"><a href="#cb438-2" aria-hidden="true" tabindex="-1"></a>res_sig_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_cDC_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-cdc-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-cdc-cells-ifnpos-vs-ifnneg">6.3.2.1 Exporting results for cDC Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb439"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb439-1"><a href="#cb439-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb439-2"><a href="#cb439-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_cDC_IFNpos_vs_IFNneg), </span>
<span id="cb439-3"><a href="#cb439-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_cDC_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-1" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-1">6.3.2.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb440"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb440-1"><a href="#cb440-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_cDC_IFNpos_vs_IFNneg,</span>
<span id="cb440-2"><a href="#cb440-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in cDC"</span>,</span>
<span id="cb440-3"><a href="#cb440-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-128-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-cdc-ifnpos-vs-ifnneg">6.3.2.3 Top 10 Differentially Expressed Genes in cDC (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb441"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb441-1"><a href="#cb441-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_cDC_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
           baseMean log2FoldChange     lfcSE      stat      pvalue        padj
          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A     101.84935      -11.05412  2.833627  -3.90105 9.57768e-05 4.18743e-03
IFI27    1272.99387        9.33333  0.809070  11.53587 8.69981e-31 5.40113e-27
OTOF       50.04154        9.03694  1.492563   6.05464 1.40729e-09 3.01273e-07
PTPRH       4.49211       -4.93292  1.164299  -4.23682 2.26707e-05 1.26042e-03
IFI44L   1998.83463        4.89002  0.492492   9.92914 3.10938e-23 5.26474e-20
TNFAIP6     9.92146        4.79073  0.932266   5.13880 2.76496e-07 3.35550e-05
GALR2       3.56876       -4.52439  1.467549  -3.08295 2.04957e-03 4.07398e-02
NTSR1       1.71336        4.48742  1.488547   3.01463 2.57291e-03 4.76497e-02
MTND1P32    3.80621       -4.46906  1.336434  -3.34402 8.25748e-04 2.11840e-02
EGR3        4.55683       -4.45140  1.532197  -2.90524 3.66968e-03 6.16435e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-13" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-13">6.3.2.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb443"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb443-1"><a href="#cb443-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb443-2"><a href="#cb443-2" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_cDC_IFNpos_vs_IFNneg)</span>
<span id="cb443-3"><a href="#cb443-3" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_cDC_IFNpos_vs_IFNneg)</span>
<span id="cb443-4"><a href="#cb443-4" aria-hidden="true" tabindex="-1"></a>res_df_cDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_cDC_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb443-5"><a href="#cb443-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-6"><a href="#cb443-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_cDC_IFNpos_vs_IFNneg,</span>
<span id="cb443-7"><a href="#cb443-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_cDC_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb443-8"><a href="#cb443-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb443-9"><a href="#cb443-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb443-10"><a href="#cb443-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb443-11"><a href="#cb443-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb443-12"><a href="#cb443-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in cDC'</span>,</span>
<span id="cb443-13"><a href="#cb443-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb443-14"><a href="#cb443-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb443-15"><a href="#cb443-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb443-16"><a href="#cb443-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb443-17"><a href="#cb443-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb443-18"><a href="#cb443-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb443-19"><a href="#cb443-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb443-20"><a href="#cb443-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb443-21"><a href="#cb443-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb443-22"><a href="#cb443-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-23"><a href="#cb443-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb443-24"><a href="#cb443-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1369 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-130-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-b-cells-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-b-cells-ifnpos-vs-ifnneg">6.3.3 Differential Expression Analysis for B Cells (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb445"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb445-1"><a href="#cb445-1" aria-hidden="true" tabindex="-1"></a>coldata_B_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_B_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb445-2"><a href="#cb445-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb445-3"><a href="#cb445-3" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_B_Cells,</span>
<span id="cb445-4"><a href="#cb445-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_B_Cells,</span>
<span id="cb445-5"><a href="#cb445-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb445-6"><a href="#cb445-6" aria-hidden="true" tabindex="-1"></a>dds_B_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_B_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 116 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb455"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb455-1"><a href="#cb455-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb455-2"><a href="#cb455-2" aria-hidden="true" tabindex="-1"></a>res_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_B_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb455-3"><a href="#cb455-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb455-4"><a href="#cb455-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb455-5"><a href="#cb455-5" aria-hidden="true" tabindex="-1"></a>res_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_B_Cells_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb455-6"><a href="#cb455-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb455-7"><a href="#cb455-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb455-8"><a href="#cb455-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_B_Cells_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20081 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 166, 0.83%
LFC &lt; 0 (down)     : 231, 1.2%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb457"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb457-1"><a href="#cb457-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_B_Cells_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
CMPK2     168.043        4.07385  0.294133  13.85036 1.26579e-43 2.54183e-39
IFI44L   1479.816        4.46368  0.405106  11.01855 3.11017e-28 3.12277e-24
IFI44     536.649        3.43363  0.364674   9.41562 4.70298e-21 3.14802e-17
RSAD2     187.138        3.56361  0.383576   9.29050 1.53562e-20 7.70921e-17
IFIT3     488.133        3.12375  0.337354   9.25957 2.05257e-20 8.24354e-17
EIF2AK2   858.377        1.58040  0.175377   9.01149 2.03284e-19 6.80357e-16</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb459"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb459-1"><a href="#cb459-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb459-2"><a href="#cb459-2" aria-hidden="true" tabindex="-1"></a>resLFC_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_B_Cells,</span>
<span id="cb459-3"><a href="#cb459-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb459-4"><a href="#cb459-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb461"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb461-1"><a href="#cb461-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb461-2"><a href="#cb461-2" aria-hidden="true" tabindex="-1"></a>res_sig_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_B_Cells_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-b-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-b-cells-ifnpos-vs-ifnneg">6.3.3.1 Exporting results for B Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb462"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb462-1"><a href="#cb462-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb462-2"><a href="#cb462-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_B_Cells_IFNpos_vs_IFNneg), </span>
<span id="cb462-3"><a href="#cb462-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_B_Cells_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-2" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-2">6.3.3.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb463"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb463-1"><a href="#cb463-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_B_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb463-2"><a href="#cb463-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in B Cells"</span>,</span>
<span id="cb463-3"><a href="#cb463-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-135-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-b-cells-ifnpos-vs-ifnneg">6.3.3.3 Top 10 Differentially Expressed Genes in B Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb464"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb464-1"><a href="#cb464-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_B_Cells_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
            baseMean log2FoldChange     lfcSE      stat      pvalue        padj
           &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IL36A      117.48979      -10.52372  2.789373  -3.77279 1.61433e-04 2.55255e-02
OTOF         6.11952        6.58872  1.860004   3.54231 3.96633e-04 4.12684e-02
FRZB         5.14098       -5.39181  1.271656  -4.23999 2.23527e-05 6.50530e-03
LINC00487    4.18743        5.35515  1.100938   4.86417 1.14939e-06 5.59412e-04
NRIR        16.26801        4.66356  0.781035   5.97100 2.35810e-09 1.89412e-06
AHSP         5.01070        4.49716  1.432260   3.13991 1.69002e-03 9.27564e-02
IFI44L    1479.81593        4.46368  0.405106  11.01855 3.11017e-28 3.12277e-24
IFI27      254.35770        4.39135  1.034534   4.24476 2.18828e-05 6.46218e-03
CMPK2      168.04341        4.07385  0.294133  13.85036 1.26579e-43 2.54183e-39
SORBS2       2.75941       -3.61693  1.138929  -3.17573 1.49459e-03 8.70689e-02</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-14" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-14">6.3.3.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb466"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb466-1"><a href="#cb466-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb466-2"><a href="#cb466-2" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_B_Cells_IFNpos_vs_IFNneg)</span>
<span id="cb466-3"><a href="#cb466-3" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_B_Cells_IFNpos_vs_IFNneg)</span>
<span id="cb466-4"><a href="#cb466-4" aria-hidden="true" tabindex="-1"></a>res_df_B_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_B_Cells_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb466-5"><a href="#cb466-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb466-6"><a href="#cb466-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_B_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb466-7"><a href="#cb466-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_B_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb466-8"><a href="#cb466-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb466-9"><a href="#cb466-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb466-10"><a href="#cb466-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb466-11"><a href="#cb466-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb466-12"><a href="#cb466-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in B Cells'</span>,</span>
<span id="cb466-13"><a href="#cb466-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb466-14"><a href="#cb466-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb466-15"><a href="#cb466-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb466-16"><a href="#cb466-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb466-17"><a href="#cb466-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb466-18"><a href="#cb466-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb466-19"><a href="#cb466-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb466-20"><a href="#cb466-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb466-21"><a href="#cb466-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb466-22"><a href="#cb466-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb466-23"><a href="#cb466-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb466-24"><a href="#cb466-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 383 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-137-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-t-cells-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-t-cells-ifnpos-vs-ifnneg">6.3.4 Differential Expression Analysis for T Cells (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb468"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb468-1"><a href="#cb468-1" aria-hidden="true" tabindex="-1"></a>coldata_T_Cells<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_T_Cells<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb468-2"><a href="#cb468-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-3"><a href="#cb468-3" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_T_Cells,</span>
<span id="cb468-4"><a href="#cb468-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_T_Cells,</span>
<span id="cb468-5"><a href="#cb468-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb468-6"><a href="#cb468-6" aria-hidden="true" tabindex="-1"></a>dds_T_Cells <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_T_Cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 50 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb478"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb478-1"><a href="#cb478-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb478-2"><a href="#cb478-2" aria-hidden="true" tabindex="-1"></a>res_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_T_Cells, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb478-3"><a href="#cb478-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-4"><a href="#cb478-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb478-5"><a href="#cb478-5" aria-hidden="true" tabindex="-1"></a>res_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_T_Cells_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb478-6"><a href="#cb478-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-7"><a href="#cb478-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb478-8"><a href="#cb478-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_T_Cells_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 20883 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 651, 3.1%
LFC &lt; 0 (down)     : 283, 1.4%
outliers [1]       : 0, 0%
low counts [2]     : 4454, 21%
(mean count &lt; 11)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb480"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb480-1"><a href="#cb480-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_T_Cells_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI6     2116.879        2.45020 0.2006343   12.2123 2.67244e-34 4.39056e-30
IFI44L   2005.728        4.36027 0.3761473   11.5919 4.52886e-31 3.72023e-27
MX1      1102.398        2.33735 0.2141153   10.9163 9.63411e-28 5.27596e-24
EIF2AK2  1200.942        1.49191 0.1396506   10.6832 1.22045e-26 4.63640e-23
MX2       644.440        1.03145 0.0966714   10.6697 1.41104e-26 4.63640e-23
OAS3      622.486        1.73400 0.1652433   10.4936 9.24393e-26 2.53114e-22</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb482"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb482-1"><a href="#cb482-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb482-2"><a href="#cb482-2" aria-hidden="true" tabindex="-1"></a>resLFC_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_T_Cells,</span>
<span id="cb482-3"><a href="#cb482-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb482-4"><a href="#cb482-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb484"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb484-1"><a href="#cb484-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb484-2"><a href="#cb484-2" aria-hidden="true" tabindex="-1"></a>res_sig_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_T_Cells_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-t-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-t-cells-ifnpos-vs-ifnneg">6.3.4.1 Exporting results for T Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb485"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb485-1"><a href="#cb485-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb485-2"><a href="#cb485-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_T_Cells_IFNpos_vs_IFNneg), </span>
<span id="cb485-3"><a href="#cb485-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_T_Cells_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-3" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-3">6.3.4.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb486"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb486-1"><a href="#cb486-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_T_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb486-2"><a href="#cb486-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in T Cells"</span>,</span>
<span id="cb486-3"><a href="#cb486-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-142-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-t-cells-ifnpos-vs-ifnneg">6.3.4.3 Top 10 Differentially Expressed Genes in T Cells (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb487"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb487-1"><a href="#cb487-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_T_Cells_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
        baseMean log2FoldChange     lfcSE      stat      pvalue        padj
       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
IFI27   807.3096        6.27660  0.678512   9.25054 2.23352e-20 2.03858e-17
IFI44L 2005.7283        4.36027  0.376147  11.59192 4.52886e-31 3.72023e-27
OTOF     32.4095        3.83014  0.632953   6.05122 1.43749e-09 4.72329e-07
IFIT3   346.8381        3.10778  0.304750  10.19780 2.02804e-24 3.33187e-21
RSAD2   345.9063        2.94103  0.285100  10.31581 5.97771e-25 1.09120e-21
NRIR     15.4155        2.71360  0.358356   7.57236 3.66498e-14 1.77094e-11
USP41    59.5937        2.63105  0.277205   9.49135 2.28055e-21 2.34169e-18
CMPK2   205.6176        2.61257  0.264911   9.86208 6.07759e-23 8.32073e-20
USP18   376.3179        2.56782  0.273049   9.40426 5.23973e-21 5.06374e-18
IFIT1   267.1082        2.53234  0.261959   9.66691 4.16777e-22 4.56482e-19</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-15" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-15">6.3.4.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb489"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb489-1"><a href="#cb489-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb489-2"><a href="#cb489-2" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_T_Cells_IFNpos_vs_IFNneg)</span>
<span id="cb489-3"><a href="#cb489-3" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_T_Cells_IFNpos_vs_IFNneg)</span>
<span id="cb489-4"><a href="#cb489-4" aria-hidden="true" tabindex="-1"></a>res_df_T_Cells_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_T_Cells_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb489-5"><a href="#cb489-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb489-6"><a href="#cb489-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_T_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb489-7"><a href="#cb489-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_T_Cells_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb489-8"><a href="#cb489-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb489-9"><a href="#cb489-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb489-10"><a href="#cb489-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb489-11"><a href="#cb489-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb489-12"><a href="#cb489-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in T Cells'</span>,</span>
<span id="cb489-13"><a href="#cb489-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb489-14"><a href="#cb489-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb489-15"><a href="#cb489-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb489-16"><a href="#cb489-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb489-17"><a href="#cb489-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb489-18"><a href="#cb489-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb489-19"><a href="#cb489-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb489-20"><a href="#cb489-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb489-21"><a href="#cb489-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb489-22"><a href="#cb489-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb489-23"><a href="#cb489-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb489-24"><a href="#cb489-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 916 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-144-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pdc-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pdc-ifnpos-vs-ifnneg">6.3.5 Differential Expression Analysis for pDC (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb491"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb491-1"><a href="#cb491-1" aria-hidden="true" tabindex="-1"></a>coldata_pDC<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_pDC<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb491-2"><a href="#cb491-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb491-3"><a href="#cb491-3" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_pDC,</span>
<span id="cb491-4"><a href="#cb491-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_pDC,</span>
<span id="cb491-5"><a href="#cb491-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb491-6"><a href="#cb491-6" aria-hidden="true" tabindex="-1"></a>dds_pDC <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_pDC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 78 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb501"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb501-1"><a href="#cb501-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb501-2"><a href="#cb501-2" aria-hidden="true" tabindex="-1"></a>res_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_pDC, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb501-3"><a href="#cb501-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb501-4"><a href="#cb501-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb501-5"><a href="#cb501-5" aria-hidden="true" tabindex="-1"></a>res_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_pDC_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb501-6"><a href="#cb501-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb501-7"><a href="#cb501-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb501-8"><a href="#cb501-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_pDC_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 19119 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 288, 1.5%
LFC &lt; 0 (down)     : 168, 0.88%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 0)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb503"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb503-1"><a href="#cb503-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_pDC_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
PARP14    419.582        3.16716  0.288484  10.97863 4.84192e-28 9.25726e-24
STAT1    4832.203        2.01657  0.186373  10.82012 2.76426e-27 2.64249e-23
IFI27     245.148        7.78971  0.734484  10.60569 2.80383e-26 1.78688e-22
EIF2AK2  1075.813        2.20925  0.212031  10.41950 2.02019e-25 9.65601e-22
LY6E     1018.817        3.83403  0.383003  10.01045 1.37130e-23 5.24358e-20
IFITM1    501.606        2.90149  0.293471   9.88681 4.74940e-23 1.51340e-19</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb505"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb505-1"><a href="#cb505-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb505-2"><a href="#cb505-2" aria-hidden="true" tabindex="-1"></a>resLFC_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_pDC,</span>
<span id="cb505-3"><a href="#cb505-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb505-4"><a href="#cb505-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb507"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb507-1"><a href="#cb507-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb507-2"><a href="#cb507-2" aria-hidden="true" tabindex="-1"></a>res_sig_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_pDC_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-pdc-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pdc-ifnpos-vs-ifnneg">6.3.5.1 Exporting results for pDC (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb508"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb508-1"><a href="#cb508-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb508-2"><a href="#cb508-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_pDC_IFNpos_vs_IFNneg), </span>
<span id="cb508-3"><a href="#cb508-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_pDC_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-4" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-4">6.3.5.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb509"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb509-1"><a href="#cb509-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_pDC_IFNpos_vs_IFNneg,</span>
<span id="cb509-2"><a href="#cb509-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in pDC"</span>,</span>
<span id="cb509-3"><a href="#cb509-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-149-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pdc-ifnpos-vs-ifnneg">6.3.5.3 Top 10 Differentially Expressed Genes in pDC (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb510"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb510-1"><a href="#cb510-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_pDC_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
                baseMean log2FoldChange     lfcSE      stat      pvalue
               &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
IFI27          245.14788        7.78971  0.734484  10.60569 2.80383e-26
RP11-705O24.1    6.51858        5.58169  0.904628   6.17015 6.82241e-10
KDM5D           23.26211        4.81656  0.910466   5.29022 1.22171e-07
USP18          410.60234        4.74830  0.586033   8.10244 5.38662e-16
USP41           67.68563        4.69729  0.576113   8.15342 3.53762e-16
AC005780.1       4.00141       -4.56840  1.050162  -4.35019 1.36023e-05
IFI44L        3152.15000        4.29603  0.561802   7.64687 2.05933e-14
IFI44          732.66323        3.94476  0.502837   7.84501 4.32913e-15
UTY             23.33159        3.89262  0.749249   5.19536 2.04323e-07
CTNNAL1          9.03136        3.84661  0.847639   4.53803 5.67825e-06
                     padj
                &lt;numeric&gt;
IFI27         1.78688e-22
RP11-705O24.1 1.97633e-07
KDM5D         2.38981e-05
USP18         3.96103e-13
USP41         2.70543e-13
AC005780.1    1.66002e-03
IFI44L        1.19310e-11
IFI44         2.75896e-12
UTY           3.75620e-05
CTNNAL1       7.86685e-04</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-16" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-16">6.3.5.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb512"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb512-1"><a href="#cb512-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb512-2"><a href="#cb512-2" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_pDC_IFNpos_vs_IFNneg)</span>
<span id="cb512-3"><a href="#cb512-3" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_pDC_IFNpos_vs_IFNneg)</span>
<span id="cb512-4"><a href="#cb512-4" aria-hidden="true" tabindex="-1"></a>res_df_pDC_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_pDC_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb512-5"><a href="#cb512-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb512-6"><a href="#cb512-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_pDC_IFNpos_vs_IFNneg,</span>
<span id="cb512-7"><a href="#cb512-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_pDC_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb512-8"><a href="#cb512-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb512-9"><a href="#cb512-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb512-10"><a href="#cb512-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb512-11"><a href="#cb512-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb512-12"><a href="#cb512-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in pDC'</span>,</span>
<span id="cb512-13"><a href="#cb512-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb512-14"><a href="#cb512-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb512-15"><a href="#cb512-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb512-16"><a href="#cb512-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb512-17"><a href="#cb512-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb512-18"><a href="#cb512-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb512-19"><a href="#cb512-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb512-20"><a href="#cb512-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb512-21"><a href="#cb512-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb512-22"><a href="#cb512-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb512-23"><a href="#cb512-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb512-24"><a href="#cb512-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 421 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-151-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="differential-expression-analysis-for-pmn-ifnpos-vs-ifnneg" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis-for-pmn-ifnpos-vs-ifnneg">6.3.6 Differential Expression Analysis for PMN (IFNpos vs IFNneg)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb514"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb514-1"><a href="#cb514-1" aria-hidden="true" tabindex="-1"></a>coldata_PMN<span class="sc">$</span>ifn_status <span class="ot">&lt;-</span> <span class="fu">relevel</span>(<span class="fu">factor</span>(coldata_PMN<span class="sc">$</span>ifn_status), <span class="at">ref =</span> <span class="st">"IFNneg"</span>)</span>
<span id="cb514-2"><a href="#cb514-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb514-3"><a href="#cb514-3" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> cts_PMN,</span>
<span id="cb514-4"><a href="#cb514-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">colData =</span> coldata_PMN,</span>
<span id="cb514-5"><a href="#cb514-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">design =</span> <span class="sc">~</span> ifn_status)</span>
<span id="cb514-6"><a href="#cb514-6" aria-hidden="true" tabindex="-1"></a>dds_PMN <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_PMN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 230 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb524"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb524-1"><a href="#cb524-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results: IFNpos vs IFNneg</span></span>
<span id="cb524-2"><a href="#cb524-2" aria-hidden="true" tabindex="-1"></a>res_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_PMN, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"ifn_status"</span>, <span class="st">"IFNpos"</span>, <span class="st">"IFNneg"</span>))</span>
<span id="cb524-3"><a href="#cb524-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb524-4"><a href="#cb524-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Order results by adjusted p-value</span></span>
<span id="cb524-5"><a href="#cb524-5" aria-hidden="true" tabindex="-1"></a>res_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_PMN_IFNpos_vs_IFNneg[<span class="fu">order</span>(res_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>padj), ]</span>
<span id="cb524-6"><a href="#cb524-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb524-7"><a href="#cb524-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect summary</span></span>
<span id="cb524-8"><a href="#cb524-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_PMN_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 17241 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 1251, 7.3%
LFC &lt; 0 (down)     : 1056, 6.1%
outliers [1]       : 0, 0%
low counts [2]     : 335, 1.9%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb526"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb526-1"><a href="#cb526-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_PMN_IFNpos_vs_IFNneg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 6 rows and 6 columns
         baseMean log2FoldChange     lfcSE      stat      pvalue        padj
        &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
HERC6     325.014        3.83337  0.306230   12.5179 5.95628e-36 1.00697e-31
TMEM123  4779.615        2.27720  0.187936   12.1169 8.59581e-34 7.26604e-30
IRF7     2198.851        2.39123  0.199073   12.0118 3.07898e-33 1.73511e-29
PARP14   2981.760        2.39380  0.202569   11.8172 3.18192e-32 1.34484e-28
SPATS2L   283.842        4.53304  0.385073   11.7719 5.44737e-32 1.84186e-28
TDRD7     621.561        2.37489  0.203043   11.6965 1.32809e-31 3.74212e-28</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb528"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb528-1"><a href="#cb528-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink log2 fold changes</span></span>
<span id="cb528-2"><a href="#cb528-2" aria-hidden="true" tabindex="-1"></a>resLFC_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds_PMN,</span>
<span id="cb528-3"><a href="#cb528-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">coef =</span> <span class="st">"ifn_status_IFNpos_vs_IFNneg"</span>, </span>
<span id="cb528-4"><a href="#cb528-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">type =</span> <span class="st">"apeglm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb530"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb530-1"><a href="#cb530-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for significantly differentially expressed genes</span></span>
<span id="cb530-2"><a href="#cb530-2" aria-hidden="true" tabindex="-1"></a>res_sig_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">subset</span>(res_PMN_IFNpos_vs_IFNneg, <span class="sc">!</span><span class="fu">is.na</span>(padj) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="exporting-results-for-pmn-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="exporting-results-for-pmn-ifnpos-vs-ifnneg">6.3.6.1 Exporting results for PMN (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb531"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb531-1"><a href="#cb531-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export results</span></span>
<span id="cb531-2"><a href="#cb531-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(res_sig_PMN_IFNpos_vs_IFNneg), </span>
<span id="cb531-3"><a href="#cb531-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"results/DEG_significant_PMN_IFNpos_vs_IFNneg.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-5" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-of-shrunken-log2-fold-changes-ifnpos-vs-ifnneg-5">6.3.6.2 MA-plot of shrunken log2 fold changes (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb532"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb532-1"><a href="#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLFC_PMN_IFNpos_vs_IFNneg,</span>
<span id="cb532-2"><a href="#cb532-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">main =</span> <span class="st">"MA Plot (Shrunken LFC): IFNpos vs IFNneg in PMN"</span>,</span>
<span id="cb532-3"><a href="#cb532-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-156-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-ifnneg" class="level4">
<h4 class="anchored" data-anchor-id="top-10-differentially-expressed-genes-in-pmn-ifnpos-vs-ifnneg">6.3.6.3 Top 10 Differentially Expressed Genes in PMN (IFNpos vs IFNneg)</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb533"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb533-1"><a href="#cb533-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sig_PMN_IFNpos_vs_IFNneg[<span class="fu">order</span>(<span class="fu">abs</span>(res_sig_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ], <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): ifn_status IFNpos vs IFNneg 
Wald test p-value: ifn status IFNpos vs IFNneg 
DataFrame with 10 rows and 6 columns
               baseMean log2FoldChange     lfcSE      stat      pvalue
              &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
OTOF          130.96602       10.44343  1.010050  10.33952 4.66871e-25
IFI27         317.09943        8.32967  0.867810   9.59850 8.11181e-22
RP11-288L9.1    7.14567        6.96089  0.900625   7.72895 1.08434e-14
SIGLEC1        61.42598        6.61065  0.785561   8.41520 3.92236e-17
USP41          70.01522        6.37416  0.653631   9.75192 1.81009e-22
USP18         423.57521        6.27584  0.783702   8.00794 1.16646e-15
CCNA1           6.11776        5.87689  1.592767   3.68974 2.24485e-04
TMEM255A        3.89381        5.74298  1.609778   3.56756 3.60323e-04
RTP4          318.83494        5.34961  0.815624   6.55891 5.42024e-11
IFI44L       3549.05385        5.32201  0.645580   8.24376 1.66882e-16
                    padj
               &lt;numeric&gt;
OTOF         6.07147e-22
IFI27        5.96253e-19
RP11-288L9.1 1.99330e-12
SIGLEC1      1.16336e-14
USP41        1.70008e-19
USP18        2.59476e-13
CCNA1        4.53421e-03
TMEM255A     6.64299e-03
RTP4         5.83660e-09
IFI44L       4.34047e-14</code></pre>
</div>
</div>
</section>
<section id="volcano-plot-17" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-17">6.3.6.4 Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb535"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb535-1"><a href="#cb535-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and clean</span></span>
<span id="cb535-2"><a href="#cb535-2" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resLFC_PMN_IFNpos_vs_IFNneg)</span>
<span id="cb535-3"><a href="#cb535-3" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df_PMN_IFNpos_vs_IFNneg)</span>
<span id="cb535-4"><a href="#cb535-4" aria-hidden="true" tabindex="-1"></a>res_df_PMN_IFNpos_vs_IFNneg <span class="ot">&lt;-</span> res_df_PMN_IFNpos_vs_IFNneg[<span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(res_df_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb535-5"><a href="#cb535-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-6"><a href="#cb535-6" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(res_df_PMN_IFNpos_vs_IFNneg,</span>
<span id="cb535-7"><a href="#cb535-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> res_df_PMN_IFNpos_vs_IFNneg<span class="sc">$</span>gene,</span>
<span id="cb535-8"><a href="#cb535-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">'log2FoldChange'</span>,</span>
<span id="cb535-9"><a href="#cb535-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">'padj'</span>,</span>
<span id="cb535-10"><a href="#cb535-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">pCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb535-11"><a href="#cb535-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">FCcutoff =</span> <span class="dv">0</span>,</span>
<span id="cb535-12"><a href="#cb535-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">'IFNpos_vs_IFNneg in PMN'</span>,</span>
<span id="cb535-13"><a href="#cb535-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb535-14"><a href="#cb535-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">'log2 Fold Change'</span>,</span>
<span id="cb535-15"><a href="#cb535-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">'-log10 Adjusted p-value'</span>,</span>
<span id="cb535-16"><a href="#cb535-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb535-17"><a href="#cb535-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb535-18"><a href="#cb535-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(<span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'grey'</span>, <span class="st">'red'</span>),</span>
<span id="cb535-19"><a href="#cb535-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">'bottom'</span>,</span>
<span id="cb535-20"><a href="#cb535-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendLabSize =</span> <span class="dv">10</span>,</span>
<span id="cb535-21"><a href="#cb535-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendIconSize =</span> <span class="fl">2.5</span>,</span>
<span id="cb535-22"><a href="#cb535-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">drawConnectors =</span> <span class="cn">TRUE</span>,</span>
<span id="cb535-23"><a href="#cb535-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">widthConnectors =</span> <span class="fl">0.5</span>,</span>
<span id="cb535-24"><a href="#cb535-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">axisLabSize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 2297 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-158-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="functional-enrichment-analysis" class="level1">
<h1>7. Functional Enrichment Analysis</h1>
<p>To better understand the biological significance of the differentially expressed genes, functional enrichment analysis was performed using the SummArIzeR R package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb537"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb537-1"><a href="#cb537-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to prepare the gene list data frame for one comparison</span></span>
<span id="cb537-2"><a href="#cb537-2" aria-hidden="true" tabindex="-1"></a>prepare_genelist_df <span class="ot">&lt;-</span> <span class="cf">function</span>(res_object, celltype, groupname) {</span>
<span id="cb537-3"><a href="#cb537-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb537-4"><a href="#cb537-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter for padj &lt; 0.05 and |log2FoldChange| &gt; 1</span></span>
<span id="cb537-5"><a href="#cb537-5" aria-hidden="true" tabindex="-1"></a>  filtered <span class="ot">&lt;-</span> res_object[<span class="sc">!</span><span class="fu">is.na</span>(res_object<span class="sc">$</span>padj) <span class="sc">&amp;</span> res_object<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>    (res_object<span class="sc">$</span>log2FoldChange) <span class="sc">&gt;</span> <span class="dv">1</span>, ]</span>
<span id="cb537-6"><a href="#cb537-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb537-7"><a href="#cb537-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(filtered) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb537-8"><a href="#cb537-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb537-9"><a href="#cb537-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb537-10"><a href="#cb537-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb537-11"><a href="#cb537-11" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb537-12"><a href="#cb537-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes =</span> <span class="fu">rownames</span>(filtered),</span>
<span id="cb537-13"><a href="#cb537-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Group =</span> groupname,</span>
<span id="cb537-14"><a href="#cb537-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">CellType =</span> celltype,</span>
<span id="cb537-15"><a href="#cb537-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">regulation =</span> <span class="fu">ifelse</span>(filtered<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"up-regulated"</span>, <span class="st">"down-regulated"</span>),</span>
<span id="cb537-16"><a href="#cb537-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">log2fold =</span> filtered<span class="sc">$</span>log2FoldChange</span>
<span id="cb537-17"><a href="#cb537-17" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb537-18"><a href="#cb537-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb537-19"><a href="#cb537-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb537-20"><a href="#cb537-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="functional-enrichment-for-classical-monocytes-cmo" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-classical-monocytes-cmo">7.1 Functional Enrichment for Classical Monocytes (cMo)</h2>
<p>This section outlines the complete enrichment analysis pipeline for classical monocytes, including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-cmo-cells" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-cmo-cells">7.1.1 Preparation of Gene Lists for Enrichment Analysis (cMo Cells)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb538"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb538-1"><a href="#cb538-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb538-2"><a href="#cb538-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb538-3"><a href="#cb538-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb538-4"><a href="#cb538-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb539"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb539-1"><a href="#cb539-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb539-2"><a href="#cb539-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_cMo,</span>
<span id="cb539-3"><a href="#cb539-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_cMo,</span>
<span id="cb539-4"><a href="#cb539-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_cMo</span>
<span id="cb539-5"><a href="#cb539-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-cmo-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-cmo-cells">7.1.2 Running SummArIzeR Enrichment Analysis for cMo Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb540"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb540-1"><a href="#cb540-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb540-2"><a href="#cb540-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb540-3"><a href="#cb540-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb540-4"><a href="#cb540-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb540-5"><a href="#cb540-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb540-6"><a href="#cb540-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb540-7"><a href="#cb540-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb540-8"><a href="#cb540-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb542"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb542-1"><a href="#cb542-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
2 Defense Response To Virus (GO:00516… RTP4  5.13e-36 GO_B… cMo_IFNp… up-regula…
3 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
4 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
5 Defense Response To Virus (GO:00516… IFIT5 5.13e-36 GO_B… cMo_IFNp… up-regula…
6 Defense Response To Virus (GO:00516… IFI6  5.13e-36 GO_B… cMo_IFNp… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms">7.1.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb544"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb544-1"><a href="#cb544-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb545"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb545-1"><a href="#cb545-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb545-2"><a href="#cb545-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb545-3"><a href="#cb545-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Cytokine-Mediated Interferon Signaling and Systemic Immune Regulation'</span>,</span>
<span id="cb545-4"><a href="#cb545-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Proteolytic Processing and Pulmonary Surfactant Metabolism'</span></span>
<span id="cb545-5"><a href="#cb545-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb546"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb546-1"><a href="#cb546-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 4 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 1: `condition = "cMo_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 3 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-cmo-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-cmo-cells">7.1.4 Heatmap of Enriched Biological Terms for cMo Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb548"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb548-1"><a href="#cb548-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb548-2"><a href="#cb548-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb548-3"><a href="#cb548-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F,</span>
<span id="cb548-4"><a href="#cb548-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb548-5"><a href="#cb548-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb548-6"><a href="#cb548-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb548-7"><a href="#cb548-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prepare_heatmap(input, plot_colors$default): Column
'Cluster_Annotation' contains NA values. Please make sure all clusters are
annotated.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-166-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-cmo-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-cmo-cells">7.1.5 Bubble Plot of Enriched Biological Terms for cMo Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb550"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb550-1"><a href="#cb550-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb550-2"><a href="#cb550-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb550-3"><a href="#cb550-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F</span>
<span id="cb550-4"><a href="#cb550-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-167-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-enrichment-for-conventional-dendritic-cdc" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-conventional-dendritic-cdc">7.2 Functional Enrichment for Conventional Dendritic (cDC)</h2>
<p>This section outlines the complete enrichment analysis pipeline for conventional dendritic cells, including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-cdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-cdc-cells">7.2.1 Preparation of Gene Lists for Enrichment Analysis (cDC Cells)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb551"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb551-1"><a href="#cb551-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb551-2"><a href="#cb551-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb551-3"><a href="#cb551-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb551-4"><a href="#cb551-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb552"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb552-1"><a href="#cb552-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb552-2"><a href="#cb552-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_cDC,</span>
<span id="cb552-3"><a href="#cb552-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_cDC,</span>
<span id="cb552-4"><a href="#cb552-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_cDC</span>
<span id="cb552-5"><a href="#cb552-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-cdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-cdc-cells">7.2.2 Running SummArIzeR Enrichment Analysis for cDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb553"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb553-1"><a href="#cb553-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb553-2"><a href="#cb553-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb553-3"><a href="#cb553-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb553-4"><a href="#cb553-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb553-5"><a href="#cb553-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb553-6"><a href="#cb553-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb553-7"><a href="#cb553-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb553-8"><a href="#cb553-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb555"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb555-1"><a href="#cb555-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Virus (GO:00516… IFIT… 7.42e-29 GO_B… cDC_IFNp… up-regula…
2 Defense Response To Virus (GO:00516… RTP4  7.42e-29 GO_B… cDC_IFNp… up-regula…
3 Defense Response To Virus (GO:00516… IFIT… 7.42e-29 GO_B… cDC_IFNp… up-regula…
4 Defense Response To Virus (GO:00516… IFIT5 7.42e-29 GO_B… cDC_IFNp… up-regula…
5 Defense Response To Virus (GO:00516… PRF1  7.42e-29 GO_B… cDC_IFNp… up-regula…
6 Defense Response To Virus (GO:00516… IFI6  7.42e-29 GO_B… cDC_IFNp… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-1" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-1">7.2.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb557"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb557-1"><a href="#cb557-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb558"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb558-1"><a href="#cb558-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb558-2"><a href="#cb558-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Cytokine-Mediated Interferon Signaling and Systemic Immune Regulation'</span>,</span>
<span id="cb558-3"><a href="#cb558-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb558-4"><a href="#cb558-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Ephrin Receptor-Mediated Cell Communication and Signal Transduction'</span>,</span>
<span id="cb558-5"><a href="#cb558-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Immunoregulatory Crosstalk Between Lymphoid and Non-Lymphoid Cells'</span></span>
<span id="cb558-6"><a href="#cb558-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb559"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb559-1"><a href="#cb559-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 5 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 1: `condition = "cDC_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 4 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-cdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-cdc-cells">7.2.4 Heatmap of Enriched Biological Terms for cDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb561"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb561-1"><a href="#cb561-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb561-2"><a href="#cb561-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb561-3"><a href="#cb561-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F,</span>
<span id="cb561-4"><a href="#cb561-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb561-5"><a href="#cb561-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb561-6"><a href="#cb561-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb561-7"><a href="#cb561-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-174-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-cdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-cdc-cells">7.2.5 Bubble Plot of Enriched Biological Terms for cDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb562"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb562-1"><a href="#cb562-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb562-2"><a href="#cb562-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb562-3"><a href="#cb562-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F</span>
<span id="cb562-4"><a href="#cb562-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-175-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-enrichment-for-b-cells" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-b-cells">7.3 Functional Enrichment for B Cells</h2>
<p>This section outlines the complete enrichment analysis pipeline for B Cells, including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-b-cells" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-b-cells">7.3.1 Preparation of Gene Lists for Enrichment Analysis (B Cells)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb563"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb563-1"><a href="#cb563-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb563-2"><a href="#cb563-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells, <span class="at">celltype =</span> <span class="st">"B Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb563-3"><a href="#cb563-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"B Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb563-4"><a href="#cb563-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"B Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb564"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb564-1"><a href="#cb564-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb564-2"><a href="#cb564-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_B_Cells,</span>
<span id="cb564-3"><a href="#cb564-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_B_Cells,</span>
<span id="cb564-4"><a href="#cb564-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_B_Cells</span>
<span id="cb564-5"><a href="#cb564-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-b-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-b-cells">7.3.2 Running SummArIzeR Enrichment Analysis for B Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb565"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb565-1"><a href="#cb565-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb565-2"><a href="#cb565-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb565-3"><a href="#cb565-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb565-4"><a href="#cb565-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb565-5"><a href="#cb565-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb565-6"><a href="#cb565-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb565-7"><a href="#cb565-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb565-8"><a href="#cb565-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: GO_Biological_Process_2023 - No genes have
been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: GO_Biological_Process_2023</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: Reactome_2022 - No genes have been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: Reactome_2022</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: BioPlanet_2019 - No genes have been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: BioPlanet_2019</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb576"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb576-1"><a href="#cb576-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Symbiont (GO:01… IFIT… 3.16e-15 GO_B… B Cells_… up-regula…
2 Defense Response To Symbiont (GO:01… IFIT… 3.16e-15 GO_B… B Cells_… up-regula…
3 Defense Response To Symbiont (GO:01… PRF1  3.16e-15 GO_B… B Cells_… up-regula…
4 Defense Response To Symbiont (GO:01… IFI6  3.16e-15 GO_B… B Cells_… up-regula…
5 Defense Response To Symbiont (GO:01… DDX6… 3.16e-15 GO_B… B Cells_… up-regula…
6 Defense Response To Symbiont (GO:01… IFIT1 3.16e-15 GO_B… B Cells_… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-2" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-2">7.3.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb578"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb578-1"><a href="#cb578-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb579"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb579-1"><a href="#cb579-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb579-2"><a href="#cb579-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'B Cell Antigen Receptor Signaling and Activation'</span>,</span>
<span id="cb579-3"><a href="#cb579-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Cytokine-Mediated Immune System Regulation'</span>,</span>
<span id="cb579-4"><a href="#cb579-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb579-5"><a href="#cb579-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Innate and Adaptive Immune Crosstalk Including B Cell Signaling'</span>,</span>
<span id="cb579-6"><a href="#cb579-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Type I Interferon-Induced ISG15-Mediated Antiviral Mechanism'</span>,</span>
<span id="cb579-7"><a href="#cb579-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">'6'</span> <span class="ot">=</span> <span class="st">'Immunoregulatory Communication Between Lymphoid and Non-Lymphoid Cells'</span>,</span>
<span id="cb579-8"><a href="#cb579-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">'7'</span> <span class="ot">=</span> <span class="st">'Interleukin-4 regulation of apoptosis'</span>,</span>
<span id="cb579-9"><a href="#cb579-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">'8'</span> <span class="ot">=</span> <span class="st">'Leptin influence on immune response'</span></span>
<span id="cb579-10"><a href="#cb579-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb580"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb580-1"><a href="#cb580-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 7 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 2: `condition = "B Cells_IFNpos_vs_HC"` `Cluster = 2`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 6 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-b-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-b-cells">7.3.4 Heatmap of Enriched Biological Terms for B Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb582"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb582-1"><a href="#cb582-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb582-2"><a href="#cb582-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb582-3"><a href="#cb582-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T,</span>
<span id="cb582-4"><a href="#cb582-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb582-5"><a href="#cb582-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb582-6"><a href="#cb582-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb582-7"><a href="#cb582-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-182-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-b-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-b-cells">7.3.5 Bubble Plot of Enriched Biological Terms for B Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb583"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb583-1"><a href="#cb583-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb583-2"><a href="#cb583-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb583-3"><a href="#cb583-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T</span>
<span id="cb583-4"><a href="#cb583-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-183-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-enrichment-for-t-cells" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-t-cells">7.4 Functional Enrichment for T Cells</h2>
<p>This section outlines the complete enrichment analysis pipeline for T Cells, including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-t-cells" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-t-cells">7.4.1 Preparation of Gene Lists for Enrichment Analysis (T Cells)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb584"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb584-1"><a href="#cb584-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb584-2"><a href="#cb584-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells, <span class="at">celltype =</span> <span class="st">"T Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb584-3"><a href="#cb584-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"T Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb584-4"><a href="#cb584-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"T Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb585"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb585-1"><a href="#cb585-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb585-2"><a href="#cb585-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_T_Cells,</span>
<span id="cb585-3"><a href="#cb585-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_T_Cells,</span>
<span id="cb585-4"><a href="#cb585-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_T_Cells</span>
<span id="cb585-5"><a href="#cb585-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-t-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-t-cells">7.4.2 Running SummArIzeR Enrichment Analysis for T Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb586"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb586-1"><a href="#cb586-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb586-2"><a href="#cb586-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb586-3"><a href="#cb586-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb586-4"><a href="#cb586-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb586-5"><a href="#cb586-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb586-6"><a href="#cb586-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb586-7"><a href="#cb586-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb586-8"><a href="#cb586-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb588"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb588-1"><a href="#cb588-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Symbiont (GO:01… RTP4  2.26e-19 GO_B… T Cells_… up-regula…
2 Defense Response To Symbiont (GO:01… RSAD2 2.26e-19 GO_B… T Cells_… up-regula…
3 Defense Response To Symbiont (GO:01… STAT1 2.26e-19 GO_B… T Cells_… up-regula…
4 Defense Response To Symbiont (GO:01… MX1   2.26e-19 GO_B… T Cells_… up-regula…
5 Defense Response To Symbiont (GO:01… IFI6  2.26e-19 GO_B… T Cells_… up-regula…
6 Defense Response To Symbiont (GO:01… EIF2… 2.26e-19 GO_B… T Cells_… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-3" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-3">7.4.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb590"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb590-1"><a href="#cb590-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb591"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb591-1"><a href="#cb591-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb591-2"><a href="#cb591-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb591-3"><a href="#cb591-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Regulation of Viral Genome Replication'</span>,</span>
<span id="cb591-4"><a href="#cb591-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Mitotic Cell Cycle and DNA Replication'</span>,</span>
<span id="cb591-5"><a href="#cb591-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Type II interferon signaling (interferon-gamma)'</span></span>
<span id="cb591-6"><a href="#cb591-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb592"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb592-1"><a href="#cb592-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 5 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 1: `condition = "T Cells_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 4 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-t-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-t-cells">7.4.4 Heatmap of Enriched Biological Terms for T Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb594"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb594-1"><a href="#cb594-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb594-2"><a href="#cb594-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb594-3"><a href="#cb594-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F,</span>
<span id="cb594-4"><a href="#cb594-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb594-5"><a href="#cb594-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb594-6"><a href="#cb594-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb594-7"><a href="#cb594-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-190-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-t-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-t-cells">7.4.5 Bubble Plot of Enriched Biological Terms for T Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb595"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb595-1"><a href="#cb595-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb595-2"><a href="#cb595-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb595-3"><a href="#cb595-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F</span>
<span id="cb595-4"><a href="#cb595-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-191-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-enrichment-for-plasmacytoid-dendritic-cells" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-plasmacytoid-dendritic-cells">7.5 Functional Enrichment for plasmacytoid dendritic cells</h2>
<p>This section outlines the complete enrichment analysis pipeline for plasmacytoid dendritic cells (pDC), including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-pdc" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-pdc">7.5.1 Preparation of Gene Lists for Enrichment Analysis (pDC)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb596"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb596-1"><a href="#cb596-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb596-2"><a href="#cb596-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb596-3"><a href="#cb596-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb596-4"><a href="#cb596-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb597"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb597-1"><a href="#cb597-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb597-2"><a href="#cb597-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_pDC,</span>
<span id="cb597-3"><a href="#cb597-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_pDC,</span>
<span id="cb597-4"><a href="#cb597-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_pDC</span>
<span id="cb597-5"><a href="#cb597-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-pdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-pdc-cells">7.5.2 Running SummArIzeR Enrichment Analysis for pDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb598"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb598-1"><a href="#cb598-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb598-2"><a href="#cb598-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb598-3"><a href="#cb598-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb598-4"><a href="#cb598-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb598-5"><a href="#cb598-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb598-6"><a href="#cb598-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb598-7"><a href="#cb598-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb598-8"><a href="#cb598-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb600"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb600-1"><a href="#cb600-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Symbiont (GO:01… IFIT… 1.87e-32 GO_B… pDC_IFNp… up-regula…
2 Defense Response To Symbiont (GO:01… RTP4  1.87e-32 GO_B… pDC_IFNp… up-regula…
3 Defense Response To Symbiont (GO:01… IFIT… 1.87e-32 GO_B… pDC_IFNp… up-regula…
4 Defense Response To Symbiont (GO:01… IFIT5 1.87e-32 GO_B… pDC_IFNp… up-regula…
5 Defense Response To Symbiont (GO:01… IFI6  1.87e-32 GO_B… pDC_IFNp… up-regula…
6 Defense Response To Symbiont (GO:01… DDX6… 1.87e-32 GO_B… pDC_IFNp… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-4" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-4">7.5.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb602"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb602-1"><a href="#cb602-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb603"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb603-1"><a href="#cb603-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb603-2"><a href="#cb603-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Type I Interferon-Induced ISG15-Mediated Antiviral Mechanism'</span>,</span>
<span id="cb603-3"><a href="#cb603-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb603-4"><a href="#cb603-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Innate Immune Defense and Neutrophil-Mediated Pathogen Clearance'</span>,</span>
<span id="cb603-5"><a href="#cb603-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Cellular Response To Cytokine Stimulus'</span></span>
<span id="cb603-6"><a href="#cb603-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb604"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb604-1"><a href="#cb604-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 4 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 4: `condition = "pDC_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 3 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-pdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-pdc-cells">7.5.4 Heatmap of Enriched Biological Terms for pDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb606"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb606-1"><a href="#cb606-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb606-2"><a href="#cb606-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb606-3"><a href="#cb606-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F,</span>
<span id="cb606-4"><a href="#cb606-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb606-5"><a href="#cb606-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb606-6"><a href="#cb606-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb606-7"><a href="#cb606-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-198-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-pdc-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-pdc-cells">7.5.5 Bubble Plot of Enriched Biological Terms for pDC Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb607"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb607-1"><a href="#cb607-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb607-2"><a href="#cb607-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb607-3"><a href="#cb607-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> F</span>
<span id="cb607-4"><a href="#cb607-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-199-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="functional-enrichment-for-polymorphonuclear-cells" class="level2">
<h2 class="anchored" data-anchor-id="functional-enrichment-for-polymorphonuclear-cells">7.6 Functional Enrichment for Polymorphonuclear Cells</h2>
<p>This section outlines the complete enrichment analysis pipeline forpolymorphonuclear cells (PMN), including gene list preparation, term extraction, cluster assignment, and visualization of enriched biological pathways.</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis-pmn" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis-pmn">7.6.1 Preparation of Gene Lists for Enrichment Analysis (PMN)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb608"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb608-1"><a href="#cb608-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare for each comparison</span></span>
<span id="cb608-2"><a href="#cb608-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb608-3"><a href="#cb608-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb608-4"><a href="#cb608-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb609"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb609-1"><a href="#cb609-1" aria-hidden="true" tabindex="-1"></a>genelist_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb609-2"><a href="#cb609-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_PMN,</span>
<span id="cb609-3"><a href="#cb609-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_PMN,</span>
<span id="cb609-4"><a href="#cb609-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_PMN</span>
<span id="cb609-5"><a href="#cb609-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-for-pmn-cells" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-for-pmn-cells">7.6.2 Running SummArIzeR Enrichment Analysis for PMN Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb610"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb610-1"><a href="#cb610-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb610-2"><a href="#cb610-2" aria-hidden="true" tabindex="-1"></a>  genelist_df,</span>
<span id="cb610-3"><a href="#cb610-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb610-4"><a href="#cb610-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb610-5"><a href="#cb610-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb610-6"><a href="#cb610-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">5</span>,</span>
<span id="cb610-7"><a href="#cb610-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb610-8"><a href="#cb610-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb612"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb612-1"><a href="#cb612-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Virus (GO:00516… IFIT… 7.03e-35 GO_B… PMN_IFNp… up-regula…
2 Defense Response To Virus (GO:00516… RTP4  7.03e-35 GO_B… PMN_IFNp… up-regula…
3 Defense Response To Virus (GO:00516… IFIT… 7.03e-35 GO_B… PMN_IFNp… up-regula…
4 Defense Response To Virus (GO:00516… IFIT5 7.03e-35 GO_B… PMN_IFNp… up-regula…
5 Defense Response To Virus (GO:00516… DDX6… 7.03e-35 GO_B… PMN_IFNp… up-regula…
6 Defense Response To Virus (GO:00516… IFIT1 7.03e-35 GO_B… PMN_IFNp… up-regula…</code></pre>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-5" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-5">7.6.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb614"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb614-1"><a href="#cb614-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb615"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb615-1"><a href="#cb615-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb615-2"><a href="#cb615-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Antiviral Immune Response and Replication Control'</span>,</span>
<span id="cb615-3"><a href="#cb615-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Cytokine-Mediated Interferon Signaling and Systemic Immune Regulation'</span>,</span>
<span id="cb615-4"><a href="#cb615-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'T Cell Activation and Interleukin-17 Signaling'</span>,</span>
<span id="cb615-5"><a href="#cb615-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'Influenza Virus Infection and Protein Synthesis Machinery'</span>,</span>
<span id="cb615-6"><a href="#cb615-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Regulation of Apoptotic Execution Phase'</span>,</span>
<span id="cb615-7"><a href="#cb615-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">'6'</span> <span class="ot">=</span> <span class="st">'Interleukin-2 signaling pathway'</span>,</span>
<span id="cb615-8"><a href="#cb615-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">'7'</span> <span class="ot">=</span> <span class="st">'Positive Regulation Of Muscle Contraction'</span></span>
<span id="cb615-9"><a href="#cb615-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb616"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb616-1"><a href="#cb616-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 4 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 1: `condition = "PMN_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 3 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms-for-pmn-cells" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms-for-pmn-cells">7.6.4 Heatmap of Enriched Biological Terms for PMN Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb618"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb618-1"><a href="#cb618-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb618-2"><a href="#cb618-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb618-3"><a href="#cb618-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T,</span>
<span id="cb618-4"><a href="#cb618-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb618-5"><a href="#cb618-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb618-6"><a href="#cb618-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F</span>
<span id="cb618-7"><a href="#cb618-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-206-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bubble-plot-of-enriched-biological-terms-for-pmn-cells" class="level3">
<h3 class="anchored" data-anchor-id="bubble-plot-of-enriched-biological-terms-for-pmn-cells">7.6.5 Bubble Plot of Enriched Biological Terms for PMN Cells</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb619"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb619-1"><a href="#cb619-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb619-2"><a href="#cb619-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb619-3"><a href="#cb619-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T</span>
<span id="cb619-4"><a href="#cb619-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-207-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="combined-enrichment-analysis-across-cell-types" class="level2">
<h2 class="anchored" data-anchor-id="combined-enrichment-analysis-across-cell-types">7.7 Combined Enrichment Analysis Across Cell Types</h2>
<p>In this section, we integrate differentially expressed genes (DEGs) from multiple immune cell types to perform a unified enrichment analysis. By aggregating all DEGs across cell type–specific comparisons, we aim to identify overarching biological processes and signaling pathways that are consistently affected in systemic lupus erythematosus (SLE)</p>
<section id="preparation-of-gene-lists-for-enrichment-analysis" class="level3">
<h3 class="anchored" data-anchor-id="preparation-of-gene-lists-for-enrichment-analysis">7.7.1 Preparation of Gene Lists for Enrichment Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb620"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb620-1"><a href="#cb620-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cMo</span></span>
<span id="cb620-2"><a href="#cb620-2" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-3"><a href="#cb620-3" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-4"><a href="#cb620-4" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_cMo <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cMo_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"cMo"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span>
<span id="cb620-5"><a href="#cb620-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb620-6"><a href="#cb620-6" aria-hidden="true" tabindex="-1"></a><span class="co"># cDC</span></span>
<span id="cb620-7"><a href="#cb620-7" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-8"><a href="#cb620-8" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-9"><a href="#cb620-9" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_cDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_cDC_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"cDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span>
<span id="cb620-10"><a href="#cb620-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb620-11"><a href="#cb620-11" aria-hidden="true" tabindex="-1"></a><span class="co"># B_Cells</span></span>
<span id="cb620-12"><a href="#cb620-12" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells, <span class="at">celltype =</span> <span class="st">"B_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-13"><a href="#cb620-13" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"B_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-14"><a href="#cb620-14" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_B_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_B_Cells_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"B_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span>
<span id="cb620-15"><a href="#cb620-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb620-16"><a href="#cb620-16" aria-hidden="true" tabindex="-1"></a><span class="co"># T_Cells</span></span>
<span id="cb620-17"><a href="#cb620-17" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells, <span class="at">celltype =</span> <span class="st">"T_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-18"><a href="#cb620-18" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"T_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-19"><a href="#cb620-19" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_T_Cells <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_T_Cells_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"T_Cells"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span>
<span id="cb620-20"><a href="#cb620-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb620-21"><a href="#cb620-21" aria-hidden="true" tabindex="-1"></a><span class="co"># pDC</span></span>
<span id="cb620-22"><a href="#cb620-22" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-23"><a href="#cb620-23" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-24"><a href="#cb620-24" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_pDC <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_pDC_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"pDC"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span>
<span id="cb620-25"><a href="#cb620-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb620-26"><a href="#cb620-26" aria-hidden="true" tabindex="-1"></a><span class="co"># PMN</span></span>
<span id="cb620-27"><a href="#cb620-27" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_HC_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_HC"</span>)</span>
<span id="cb620-28"><a href="#cb620-28" aria-hidden="true" tabindex="-1"></a>genelist_IFNneg_vs_HC_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN_IFNneg_vs_HC, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNneg_vs_HC"</span>)</span>
<span id="cb620-29"><a href="#cb620-29" aria-hidden="true" tabindex="-1"></a>genelist_IFNpos_vs_IFNneg_PMN <span class="ot">&lt;-</span> <span class="fu">prepare_genelist_df</span>(res_sig_PMN_IFNpos_vs_IFNneg, <span class="at">celltype =</span> <span class="st">"PMN"</span>, <span class="at">groupname =</span> <span class="st">"IFNpos_vs_IFNneg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb621"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb621-1"><a href="#cb621-1" aria-hidden="true" tabindex="-1"></a>genelist_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb621-2"><a href="#cb621-2" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_cMo,</span>
<span id="cb621-3"><a href="#cb621-3" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_cMo,</span>
<span id="cb621-4"><a href="#cb621-4" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_cMo,</span>
<span id="cb621-5"><a href="#cb621-5" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_cDC,</span>
<span id="cb621-6"><a href="#cb621-6" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_cDC,</span>
<span id="cb621-7"><a href="#cb621-7" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_cDC,</span>
<span id="cb621-8"><a href="#cb621-8" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_B_Cells,</span>
<span id="cb621-9"><a href="#cb621-9" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_B_Cells,</span>
<span id="cb621-10"><a href="#cb621-10" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_B_Cells,</span>
<span id="cb621-11"><a href="#cb621-11" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_T_Cells,</span>
<span id="cb621-12"><a href="#cb621-12" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_T_Cells,</span>
<span id="cb621-13"><a href="#cb621-13" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_T_Cells,</span>
<span id="cb621-14"><a href="#cb621-14" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_pDC,</span>
<span id="cb621-15"><a href="#cb621-15" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_pDC,</span>
<span id="cb621-16"><a href="#cb621-16" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_pDC,</span>
<span id="cb621-17"><a href="#cb621-17" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_HC_PMN,</span>
<span id="cb621-18"><a href="#cb621-18" aria-hidden="true" tabindex="-1"></a>  genelist_IFNneg_vs_HC_PMN,</span>
<span id="cb621-19"><a href="#cb621-19" aria-hidden="true" tabindex="-1"></a>  genelist_IFNpos_vs_IFNneg_PMN</span>
<span id="cb621-20"><a href="#cb621-20" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-summarizer-enrichment-analysis-across-cell-types" class="level3">
<h3 class="anchored" data-anchor-id="running-summarizer-enrichment-analysis-across-cell-types">7.7.2 Running SummArIzeR Enrichment Analysis across Cell Types</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb622"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb622-1"><a href="#cb622-1" aria-hidden="true" tabindex="-1"></a>Termlist_all <span class="ot">&lt;-</span> <span class="fu">extractMultipleTerms</span>(</span>
<span id="cb622-2"><a href="#cb622-2" aria-hidden="true" tabindex="-1"></a>  genelist_all,</span>
<span id="cb622-3"><a href="#cb622-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition_col =</span> <span class="fu">c</span>(<span class="st">"CellType"</span>, <span class="st">"Group"</span>),</span>
<span id="cb622-4"><a href="#cb622-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">categories =</span> <span class="fu">c</span>(<span class="st">"GO_Biological_Process_2023"</span>, <span class="st">"Reactome_2022"</span>, <span class="st">"BioPlanet_2019"</span>),</span>
<span id="cb622-5"><a href="#cb622-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">pval_threshold =</span> <span class="fl">0.2</span>,</span>
<span id="cb622-6"><a href="#cb622-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">n =</span> <span class="dv">10</span>,</span>
<span id="cb622-7"><a href="#cb622-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> <span class="cn">TRUE</span></span>
<span id="cb622-8"><a href="#cb622-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: GO_Biological_Process_2023 - No genes have
been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: GO_Biological_Process_2023</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: Reactome_2022 - No genes have been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: Reactome_2022</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in value[[3L]](cond): Warning: Enrichment analysis failed for
down-regulated genes in category: BioPlanet_2019 - No genes have been given</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readGeneList(listpathSig = subset_input, name = cond_value, category
= cat, : Warning: No enrichment terms found for down-regulated genes in
category: BioPlanet_2019</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2023... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying Reactome_2022... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.
Uploading data to Enrichr... Done.
  Querying BioPlanet_2019... Done.
Parsing results... Done.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb633"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb633-1"><a href="#cb633-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Term                                 Genes adj_pval dbs   condition regulation
  &lt;chr&gt;                                &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     
1 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
2 Defense Response To Virus (GO:00516… RTP4  5.13e-36 GO_B… cMo_IFNp… up-regula…
3 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
4 Defense Response To Virus (GO:00516… IFIT… 5.13e-36 GO_B… cMo_IFNp… up-regula…
5 Defense Response To Virus (GO:00516… IFIT5 5.13e-36 GO_B… cMo_IFNp… up-regula…
6 Defense Response To Virus (GO:00516… IFI6  5.13e-36 GO_B… cMo_IFNp… up-regula…</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb635"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb635-1"><a href="#cb635-1" aria-hidden="true" tabindex="-1"></a><span class="fu">evaluateThreshold</span>(Termlist_all)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-30e4409849e39179307f" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-30e4409849e39179307f">{"x":{"visdat":{"24c22c9c4ec96":["function () ","plotlyVisDat"]},"cur_data":"24c22c9c4ec96","attrs":{"24c22c9c4ec96":{"x":{},"y":{},"color":{},"colors":["#00B2AE","#6247A2","#E63946"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter","mode":"lines","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"title":"Threshold Selection Plot","xaxis":{"domain":[0,1],"automargin":true,"title":"Threshold"},"yaxis":{"domain":[0,1],"automargin":true,"title":"Metric Value"},"legend":{"title":{"text":"Metric"}},"template":"plotly_white","hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[0.050000000000000003,0.060000000000000005,0.070000000000000007,0.080000000000000002,0.089999999999999997,0.10000000000000001,0.11,0.12000000000000001,0.13,0.14000000000000001,0.15000000000000002,0.16,0.16999999999999998,0.17999999999999999,0.19,0.20000000000000001,0.21000000000000002,0.22000000000000003,0.22999999999999998,0.23999999999999999,0.25,0.26000000000000001,0.27000000000000002,0.28000000000000003,0.28999999999999998,0.29999999999999999,0.31,0.32000000000000001,0.33000000000000002,0.33999999999999997,0.34999999999999998,0.35999999999999999,0.37,0.38,0.39000000000000001,0.40000000000000002,0.40999999999999998,0.41999999999999998,0.42999999999999999,0.44,0.45000000000000001,0.46000000000000002,0.46999999999999997,0.47999999999999998,0.48999999999999999,0.5,0.51000000000000001,0.52000000000000002,0.53000000000000003,0.54000000000000004,0.55000000000000004,0.56000000000000005,0.57000000000000006,0.58000000000000007,0.59000000000000008,0.60000000000000009,0.6100000000000001,0.62000000000000011,0.63,0.64000000000000001,0.65000000000000002,0.66000000000000003,0.67000000000000004,0.68000000000000005,0.69000000000000006,0.70000000000000007,0.71000000000000008,0.72000000000000008,0.73000000000000009,0.7400000000000001,0.75000000000000011,0.76000000000000001,0.77000000000000002,0.78000000000000003,0.79000000000000004,0.80000000000000004,0.81000000000000005,0.82000000000000006,0.83000000000000007,0.84000000000000008,0.85000000000000009,0.8600000000000001,0.87000000000000011,0.88000000000000012,0.89000000000000001,0.90000000000000002,0.91000000000000003,0.92000000000000004,0.93000000000000005,0.94000000000000006,0.95000000000000007,0.96000000000000008,0.97000000000000008,0.98000000000000009,0.9900000000000001,1],"y":[15,16,16,16,16,17,17,23,23,23,25,27,31,34,38,41,44,45,46,47,47,47,48,52,55,57,55,55,55,58,61,63,65,67,69,70,72,72,77,77,78,79,79,80,81,81,84,84,85,85,86,86,86,86,87,88,88,90,90,90,91,91,94,95,95,95,95,96,98,98,99,100,102,102,103,103,103,103,103,104,104,108,108,112,115,116,116,116,119,119,119,119,119,119,119,119],"type":"scatter","mode":"lines","name":"Cluster_Count","marker":{"color":"rgba(0,178,174,1)","line":{"color":"rgba(0,178,174,1)"}},"textfont":{"color":"rgba(0,178,174,1)"},"error_y":{"color":"rgba(0,178,174,1)"},"error_x":{"color":"rgba(0,178,174,1)"},"line":{"color":"rgba(0,178,174,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":[0.050000000000000003,0.060000000000000005,0.070000000000000007,0.080000000000000002,0.089999999999999997,0.10000000000000001,0.11,0.12000000000000001,0.13,0.14000000000000001,0.15000000000000002,0.16,0.16999999999999998,0.17999999999999999,0.19,0.20000000000000001,0.21000000000000002,0.22000000000000003,0.22999999999999998,0.23999999999999999,0.25,0.26000000000000001,0.27000000000000002,0.28000000000000003,0.28999999999999998,0.29999999999999999,0.31,0.32000000000000001,0.33000000000000002,0.33999999999999997,0.34999999999999998,0.35999999999999999,0.37,0.38,0.39000000000000001,0.40000000000000002,0.40999999999999998,0.41999999999999998,0.42999999999999999,0.44,0.45000000000000001,0.46000000000000002,0.46999999999999997,0.47999999999999998,0.48999999999999999,0.5,0.51000000000000001,0.52000000000000002,0.53000000000000003,0.54000000000000004,0.55000000000000004,0.56000000000000005,0.57000000000000006,0.58000000000000007,0.59000000000000008,0.60000000000000009,0.6100000000000001,0.62000000000000011,0.63,0.64000000000000001,0.65000000000000002,0.66000000000000003,0.67000000000000004,0.68000000000000005,0.69000000000000006,0.70000000000000007,0.71000000000000008,0.72000000000000008,0.73000000000000009,0.7400000000000001,0.75000000000000011,0.76000000000000001,0.77000000000000002,0.78000000000000003,0.79000000000000004,0.80000000000000004,0.81000000000000005,0.82000000000000006,0.83000000000000007,0.84000000000000008,0.85000000000000009,0.8600000000000001,0.87000000000000011,0.88000000000000012,0.89000000000000001,0.90000000000000002,0.91000000000000003,0.92000000000000004,0.93000000000000005,0.94000000000000006,0.95000000000000007,0.96000000000000008,0.97000000000000008,0.98000000000000009,0.9900000000000001,1],"y":[12,13,13,15,15,16,16,19,19,20,20,22,27,31,33,38,41,42,43,43,43,43,44,48,49,52,52,54,55,58,59,61,65,67,69,70,72,72,77,77,78,79,79,80,81,81,84,84,85,85,86,86,86,86,87,88,88,90,90,90,91,91,92,93,93,93,93,95,97,97,98,99,101,102,103,103,103,103,103,104,104,108,108,112,115,116,116,116,119,119,119,119,119,119,119,119],"type":"scatter","mode":"lines","name":"Connected_Components","marker":{"color":"rgba(98,71,162,1)","line":{"color":"rgba(98,71,162,1)"}},"textfont":{"color":"rgba(98,71,162,1)"},"error_y":{"color":"rgba(98,71,162,1)"},"error_x":{"color":"rgba(98,71,162,1)"},"line":{"color":"rgba(98,71,162,1)"},"xaxis":"x","yaxis":"y","frame":null},{"x":[0.050000000000000003,0.060000000000000005,0.070000000000000007,0.080000000000000002,0.089999999999999997,0.10000000000000001,0.11,0.12000000000000001,0.13,0.14000000000000001,0.15000000000000002,0.16,0.16999999999999998,0.17999999999999999,0.19,0.20000000000000001,0.21000000000000002,0.22000000000000003,0.22999999999999998,0.23999999999999999,0.25,0.26000000000000001,0.27000000000000002,0.28000000000000003,0.28999999999999998,0.29999999999999999,0.31,0.32000000000000001,0.33000000000000002,0.33999999999999997,0.34999999999999998,0.35999999999999999,0.37,0.38,0.39000000000000001,0.40000000000000002,0.40999999999999998,0.41999999999999998,0.42999999999999999,0.44,0.45000000000000001,0.46000000000000002,0.46999999999999997,0.47999999999999998,0.48999999999999999,0.5,0.51000000000000001,0.52000000000000002,0.53000000000000003,0.54000000000000004,0.55000000000000004,0.56000000000000005,0.57000000000000006,0.58000000000000007,0.59000000000000008,0.60000000000000009,0.6100000000000001,0.62000000000000011,0.63,0.64000000000000001,0.65000000000000002,0.66000000000000003,0.67000000000000004,0.68000000000000005,0.69000000000000006,0.70000000000000007,0.71000000000000008,0.72000000000000008,0.73000000000000009,0.7400000000000001,0.75000000000000011,0.76000000000000001,0.77000000000000002,0.78000000000000003,0.79000000000000004,0.80000000000000004,0.81000000000000005,0.82000000000000006,0.83000000000000007,0.84000000000000008,0.85000000000000009,0.8600000000000001,0.87000000000000011,0.88000000000000012,0.89000000000000001,0.90000000000000002,0.91000000000000003,0.92000000000000004,0.93000000000000005,0.94000000000000006,0.95000000000000007,0.96000000000000008,0.97000000000000008,0.98000000000000009,0.9900000000000001,1],"y":[0.63237523503028603,0.63740498693924208,0.64874139049129298,0.65233870849346076,0.65441311095504273,0.65502077585862373,0.65561070611634609,0.65344652189763708,0.65289353876029166,0.65199983476819923,0.64934958566800449,0.64660536042943828,0.64242003692117788,0.63850969019168946,0.6374910063738507,0.63236161693424486,0.62856012036779951,0.62006710447228119,0.61464299279322987,0.61149946413532796,0.60804858537731943,0.60307027235190214,0.59334061955690576,0.5850367155992835,0.57653499755300675,0.57263379258948588,0.56535100687108031,0.56447461573480429,0.56271755597144901,0.55214780413533693,0.55047590305839178,0.53629861503717158,0.52605946306970697,0.51478054786365501,0.51049314392308054,0.50247165449426956,0.49452596876649568,0.49223582067905353,0.47722001406345765,0.47234112050239552,0.46714142755224863,0.46875554104417561,0.46327626218139201,0.45734482028867535,0.45419172083564013,0.45097888869355762,0.42715450738664629,0.42384207117066824,0.40936009443599558,0.40107683293634677,0.3982512927089375,0.40210741695643903,0.39806317731143182,0.38482522562743532,0.36774472919867768,0.36294334438049103,0.36294334438049103,0.35133743259006017,0.35901592046446884,0.36060674922636576,0.35425183446809083,0.35425183446809083,0.45066091514447609,0.4438575764556088,0.4438575764556088,0.43783945512046596,0.43783945512046596,0.4300276307775438,0.40756426750848213,0.40756426750848213,0.49409319954399128,0.48561875268274751,0.46798312047843482,0.45504306671863132,0.44456692694735211,0.44456692694735211,0.4244428112585783,0.4244428112585783,0.41268870998855661,0.39997957699006476,0.39997957699006476,0.52227211038214449,0.52227211038214449,0.45592119112583807,0.37425930831149734,0.34942536038240063,0.34942536038240063,0.34942536038240063,0.26190476190476192,0.26190476190476192,0.26190476190476192,0.26190476190476192,0.26190476190476192,0.26190476190476192,0.26190476190476192,0.26190476190476192],"type":"scatter","mode":"lines","name":"Modularity","marker":{"color":"rgba(230,57,70,1)","line":{"color":"rgba(230,57,70,1)"}},"textfont":{"color":"rgba(230,57,70,1)"},"error_y":{"color":"rgba(230,57,70,1)"},"error_x":{"color":"rgba(230,57,70,1)"},"line":{"color":"rgba(230,57,70,1)"},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="assignment-of-biological-clusters-to-enriched-terms-6" class="level3">
<h3 class="anchored" data-anchor-id="assignment-of-biological-clusters-to-enriched-terms-6">7.7.3 Assignment of Biological Clusters to Enriched Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb636"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb636-1"><a href="#cb636-1" aria-hidden="true" tabindex="-1"></a>Genelist_test_cluster <span class="ot">&lt;-</span> <span class="fu">returnIgraphCluster</span>(Termlist_all, <span class="at">ts =</span> <span class="fl">0.3</span>)</span>
<span id="cb636-2"><a href="#cb636-2" aria-hidden="true" tabindex="-1"></a><span class="fu">generateGPTPrompt</span>(Genelist_test_cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Cluster 1:
Please find a summary term for the following terms: Immunoregulatory Interactions Between A Lymphoid And A non-Lymphoid Cell R-HSA-198933, Adaptive Immune System R-HSA-1280218, Adaptive immune system, Immunoregulatory interactions between a lymphoid and a non-lymphoid cell

Cluster 2:
Please find a summary term for the following terms: Defense Response To Virus (GO:0051607), Defense Response To Symbiont (GO:0140546), Negative Regulation Of Viral Process (GO:0048525), Negative Regulation Of Viral Genome Replication (GO:0045071), Regulation Of Viral Genome Replication (GO:0045069), Interferon Signaling R-HSA-913531, Interferon Alpha/Beta Signaling R-HSA-909733, Cytokine Signaling In Immune System R-HSA-1280215, Interferon Gamma Signaling R-HSA-877300, Immune System R-HSA-168256, Interferon signaling, Interferon alpha/beta signaling, Immune system signaling by interferons, interleukins, prolactin, and growth hormones, Immune system, Interferon-gamma signaling pathway

Cluster 3:
Please find a summary term for the following terms: Cell Cycle R-HSA-1640170, Cell Cycle, Mitotic R-HSA-69278, Cell Cycle Checkpoints R-HSA-69620, Resolution Of Sister Chromatid Cohesion R-HSA-2500257, Mitotic Prometaphase R-HSA-68877, Mitotic Spindle Checkpoint R-HSA-69618, Cell cycle, DNA replication, M phase pathway, Polo-like kinase 1 (PLK1) pathway, E2F transcription factor network, Mitotic G1-G1/S phases

Cluster 4:
Please find a summary term for the following terms: Alpha-Beta T Cell Activation (GO:0046631), CTL mediated immune response against target cells, Interleukin-12-mediated signaling events, PD-1 signaling, Generation of second messenger molecules, T helper cell surface molecules, Interleukin-17 signaling pathway, NO2-dependent IL-12 pathway in NK cells

Cluster 5:
Please find a summary term for the following terms: Innate immune system, Complement cascade, Signaling by the B cell receptor (BCR), Antigen-activated B-cell receptor generation of second messengers

Cluster 6:
Please find a summary term for the following terms: Mitotic Sister Chromatid Segregation (GO:0000070), Microtubule Cytoskeleton Organization Involved In Mitosis (GO:1902850)

Cluster 7:
Please find a summary term for the following terms: Innate Immune System R-HSA-168249, Neutrophil Degranulation R-HSA-6798695

Cluster 8:
Please find a summary term for the following terms: B Cell Receptor Signaling Pathway (GO:0050853), Antigen Receptor-Mediated Signaling Pathway (GO:0050851), Retina Homeostasis (GO:0001895)

Cluster 9:
Please find a summary term for the following terms: Antiviral Mechanism By IFN-stimulated Genes R-HSA-1169410, ISG15 Antiviral Mechanism R-HSA-1169408, Antiviral mechanism by interferon-stimulated genes

Cluster 10:
Please find a summary term for the following terms: DDX58/IFIH1-mediated Induction Of Interferon-Alpha/Beta R-HSA-168928, Negative Regulators Of DDX58/IFIH1 Signaling R-HSA-936440, RIG-I-like receptor signaling pathway, RIG-I/MDA5-mediated induction of interferon-alpha/beta pathways, Negative regulators of RIG-I/MDA5 signaling

Cluster 11:
Please find a summary term for the following terms: Interleukin-27-Mediated Signaling Pathway (GO:0070106), Regulation Of Ribonuclease Activity (GO:0060700), OAS Antiviral Response R-HSA-8983711

Cluster 12:
Please find a summary term for the following terms: Macromolecule Biosynthetic Process (GO:0009059), Cytoplasmic Translation (GO:0002181), Eukaryotic Translation Elongation R-HSA-156842, Response Of EIF2AK4 (GCN2) To Amino Acid Deficiency R-HSA-9633012, Peptide Chain Elongation R-HSA-156902, Selenocysteine Synthesis R-HSA-2408557, Eukaryotic Translation Termination R-HSA-72764, Viral mRNA Translation R-HSA-192823, Nonsense Mediated Decay (NMD) Independent Of Exon Junction Complex (EJC) R-HSA-975956, Formation Of A Pool Of Free 40S Subunits R-HSA-72689, Influenza Viral RNA Transcription And Replication R-HSA-168273, SRP-dependent Cotranslational Protein Targeting To Membrane R-HSA-1799339, L13a-mediated Translational Silencing Of Ceruloplasmin Expression R-HSA-156827, GTP Hydrolysis And Joining Of 60S Ribosomal Subunit R-HSA-72706, Influenza viral RNA transcription and replication, Cytoplasmic ribosomal proteins, Influenza infection, Translation

Cluster 13:
Please find a summary term for the following terms: Cellular Response To Cytokine Stimulus (GO:0071345), Cytokine-Mediated Signaling Pathway (GO:0019221)

Cluster 14:
Please find a summary term for the following terms: Ephrin Receptor Signaling Pathway (GO:0048013), Ephrin Signaling R-HSA-3928664

Cluster 15:
Please find a summary term for the following terms: Phagosome, Leishmaniasis

Cluster 16:
Please find a summary term for the following terms: TNFs Bind Their Physiological Receptors R-HSA-5669034, TNFR2 Non-Canonical NF-kB Pathway R-HSA-5668541

Cluster 17:
Please find a summary term for the following terms: Hydrogen Peroxide Catabolic Process (GO:0042744), One-Carbon Compound Transport (GO:0019755)

Cluster 18:
Please find a summary term for the following terms: Positive Regulation Of Type I Interferon Production (GO:0032481), Positive Regulation Of Interferon-Beta Production (GO:0032728)

Cluster 19:
Please find a summary term for the following terms: Proteolysis (GO:0006508), Surfactant Metabolism R-HSA-5683826

Cluster 20:
Please find a summary term for the following terms: Negative Regulation Of Execution Phase Of Apoptosis (GO:1900118), Regulation Of Execution Phase Of Apoptosis (GO:1900117)

Cluster 21:
Please find a summary term for the following terms: Metallothioneins Bind Metals R-HSA-5661231, Response To Metal Ions R-HSA-5660526

Cluster 22:
Please find a summary term for the following terms: Antiviral Innate Immune Response (GO:0140374)

Cluster 23:
Please find a summary term for the following terms: BDNF signaling pathway

Cluster 24:
Please find a summary term for the following terms: BRCA1, BRCA2 and ATR roles in cancer susceptibility

Cluster 25:
Please find a summary term for the following terms: Cardiac Conduction (GO:0061337)

Cluster 26:
Please find a summary term for the following terms: Cellular Response To Type I Interferon (GO:0071357)

Cluster 27:
Please find a summary term for the following terms: Chemokine Receptors Bind Chemokines R-HSA-380108

Cluster 28:
Please find a summary term for the following terms: Extracellular Matrix Organization R-HSA-1474244

Cluster 29:
Please find a summary term for the following terms: FSH regulation of apoptosis

Cluster 30:
Please find a summary term for the following terms: Hematopoietic cell lineage

Cluster 31:
Please find a summary term for the following terms: Inflammatory Response (GO:0006954)

Cluster 32:
Please find a summary term for the following terms: Interleukin-2 signaling pathway

Cluster 33:
Please find a summary term for the following terms: Interleukin-4 regulation of apoptosis

Cluster 34:
Please find a summary term for the following terms: Interleukin-4 signaling pathway

Cluster 35:
Please find a summary term for the following terms: Leptin influence on immune response

Cluster 36:
Please find a summary term for the following terms: Metabolism R-HSA-1430728

Cluster 37:
Please find a summary term for the following terms: MyD88 Deficiency (TLR2/4) R-HSA-5602498

Cluster 38:
Please find a summary term for the following terms: Myelination (GO:0042552)

Cluster 39:
Please find a summary term for the following terms: NGF-stimulated Transcription R-HSA-9031628

Cluster 40:
Please find a summary term for the following terms: Neddylation R-HSA-8951664

Cluster 41:
Please find a summary term for the following terms: Negative Regulation Of Innate Immune Response (GO:0045824)

Cluster 42:
Please find a summary term for the following terms: Nucleotide-binding domain, leucine rich repeat containing receptor (NLR) signaling pathways

Cluster 43:
Please find a summary term for the following terms: Positive Regulation Of Cytokine Production (GO:0001819)

Cluster 44:
Please find a summary term for the following terms: Positive Regulation Of Muscle Contraction (GO:0045933)

Cluster 45:
Please find a summary term for the following terms: Positive Regulation Of Tumor Necrosis Factor Production (GO:0032760)

Cluster 46:
Please find a summary term for the following terms: Prolactin regulation of apoptosis

Cluster 47:
Please find a summary term for the following terms: Regulation Of IFNA/IFNB Signaling R-HSA-912694

Cluster 48:
Please find a summary term for the following terms: Regulation Of Natural Killer Cell Mediated Cytotoxicity (GO:0042269)

Cluster 49:
Please find a summary term for the following terms: Response To Cytokine (GO:0034097)

Cluster 50:
Please find a summary term for the following terms: Response To Interferon-Alpha (GO:0035455)

Cluster 51:
Please find a summary term for the following terms: Response To Interferon-Beta (GO:0035456)

Cluster 52:
Please find a summary term for the following terms: Response To Type II Interferon (GO:0034341)

Cluster 53:
Please find a summary term for the following terms: Signal Transduction R-HSA-162582

Cluster 54:
Please find a summary term for the following terms: Supramolecular Fiber Organization (GO:0097435)

Cluster 55:
Please find a summary term for the following terms: T cell receptor regulation of apoptosis

Cluster 56:
Please find a summary term for the following terms: TNF-alpha effects on cytokine activity, cell motility, and apoptosis

Cluster 57:
Please find a summary term for the following terms: Type II interferon signaling (interferon-gamma)

Please summarize the following clusters of terms with a single term for each cluster. 
 Then, assign the cluster numbers to these summary terms in an R vector, called `cluster_summary`, formatted as:
 '1' = 'Summary Term 1', '2' = 'Summary Term 2', ..., 'n' = 'Summary Term n'.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb638"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb638-1"><a href="#cb638-1" aria-hidden="true" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb638-2"><a href="#cb638-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">'1'</span> <span class="ot">=</span> <span class="st">'Adaptive Immune Interactions'</span>,</span>
<span id="cb638-3"><a href="#cb638-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">'2'</span> <span class="ot">=</span> <span class="st">'Interferon and Antiviral Immune Signaling'</span>,</span>
<span id="cb638-4"><a href="#cb638-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">'3'</span> <span class="ot">=</span> <span class="st">'Cell Cycle and Mitosis Regulation'</span>,</span>
<span id="cb638-5"><a href="#cb638-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">'4'</span> <span class="ot">=</span> <span class="st">'T Cell-Mediated Immune Activation'</span>,</span>
<span id="cb638-6"><a href="#cb638-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">'5'</span> <span class="ot">=</span> <span class="st">'Innate and B Cell Receptor-Mediated Immunity'</span>,</span>
<span id="cb638-7"><a href="#cb638-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">'6'</span> <span class="ot">=</span> <span class="st">'Mitosis and Chromosome Segregation'</span>,</span>
<span id="cb638-8"><a href="#cb638-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">'7'</span> <span class="ot">=</span> <span class="st">'Innate Immunity and Neutrophil Activity'</span>,</span>
<span id="cb638-9"><a href="#cb638-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">'8'</span> <span class="ot">=</span> <span class="st">'B Cell Signaling and Retina Function'</span>,</span>
<span id="cb638-10"><a href="#cb638-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">'9'</span> <span class="ot">=</span> <span class="st">'IFN-Stimulated Antiviral Defense'</span>,</span>
<span id="cb638-11"><a href="#cb638-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">'10'</span> <span class="ot">=</span> <span class="st">'RIG-I/MDA5-Mediated Antiviral Signaling'</span>,</span>
<span id="cb638-12"><a href="#cb638-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">'11'</span> <span class="ot">=</span> <span class="st">'Cytokine Response and OAS Pathway'</span>,</span>
<span id="cb638-13"><a href="#cb638-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">'12'</span> <span class="ot">=</span> <span class="st">'Protein Translation and Viral Replication'</span>,</span>
<span id="cb638-14"><a href="#cb638-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">'13'</span> <span class="ot">=</span> <span class="st">'Cytokine Response Signaling'</span>,</span>
<span id="cb638-15"><a href="#cb638-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">'14'</span> <span class="ot">=</span> <span class="st">'Ephrin Receptor Signaling'</span>,</span>
<span id="cb638-16"><a href="#cb638-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">'15'</span> <span class="ot">=</span> <span class="st">'Phagocytosis and Parasitic Infection'</span>,</span>
<span id="cb638-17"><a href="#cb638-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">'16'</span> <span class="ot">=</span> <span class="st">'TNF Receptor Signaling'</span>,</span>
<span id="cb638-18"><a href="#cb638-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">'17'</span> <span class="ot">=</span> <span class="st">'Oxidative Stress and One-Carbon Metabolism'</span>,</span>
<span id="cb638-19"><a href="#cb638-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">'18'</span> <span class="ot">=</span> <span class="st">'Positive Regulation of Type I IFN Production'</span>,</span>
<span id="cb638-20"><a href="#cb638-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">'19'</span> <span class="ot">=</span> <span class="st">'Protein Degradation and Surfactant Metabolism'</span>,</span>
<span id="cb638-21"><a href="#cb638-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">'20'</span> <span class="ot">=</span> <span class="st">'Apoptosis Execution Phase Regulation'</span>,</span>
<span id="cb638-22"><a href="#cb638-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">'21'</span> <span class="ot">=</span> <span class="st">'Metal Ion Homeostasis'</span>,</span>
<span id="cb638-23"><a href="#cb638-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">'22'</span> <span class="ot">=</span> <span class="st">'Innate Antiviral Immunity'</span>,</span>
<span id="cb638-24"><a href="#cb638-24" aria-hidden="true" tabindex="-1"></a>  <span class="st">'23'</span> <span class="ot">=</span> <span class="st">'BDNF Signaling'</span>,</span>
<span id="cb638-25"><a href="#cb638-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">'24'</span> <span class="ot">=</span> <span class="st">'BRCA-Mediated DNA Damage Response'</span>,</span>
<span id="cb638-26"><a href="#cb638-26" aria-hidden="true" tabindex="-1"></a>  <span class="st">'25'</span> <span class="ot">=</span> <span class="st">'Cardiac Electrical Conduction'</span>,</span>
<span id="cb638-27"><a href="#cb638-27" aria-hidden="true" tabindex="-1"></a>  <span class="st">'26'</span> <span class="ot">=</span> <span class="st">'Cellular Response to Type I Interferon'</span>,</span>
<span id="cb638-28"><a href="#cb638-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">'27'</span> <span class="ot">=</span> <span class="st">'Chemokine Receptor Signaling'</span>,</span>
<span id="cb638-29"><a href="#cb638-29" aria-hidden="true" tabindex="-1"></a>  <span class="st">'28'</span> <span class="ot">=</span> <span class="st">'Extracellular Matrix Remodeling'</span>,</span>
<span id="cb638-30"><a href="#cb638-30" aria-hidden="true" tabindex="-1"></a>  <span class="st">'29'</span> <span class="ot">=</span> <span class="st">'FSH-Regulated Apoptosis'</span>,</span>
<span id="cb638-31"><a href="#cb638-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">'30'</span> <span class="ot">=</span> <span class="st">'Hematopoietic Cell Development'</span>,</span>
<span id="cb638-32"><a href="#cb638-32" aria-hidden="true" tabindex="-1"></a>  <span class="st">'31'</span> <span class="ot">=</span> <span class="st">'Inflammatory Response'</span>,</span>
<span id="cb638-33"><a href="#cb638-33" aria-hidden="true" tabindex="-1"></a>  <span class="st">'32'</span> <span class="ot">=</span> <span class="st">'Interleukin-2 Signaling'</span>,</span>
<span id="cb638-34"><a href="#cb638-34" aria-hidden="true" tabindex="-1"></a>  <span class="st">'33'</span> <span class="ot">=</span> <span class="st">'IL-4-Related Apoptosis Regulation'</span>,</span>
<span id="cb638-35"><a href="#cb638-35" aria-hidden="true" tabindex="-1"></a>  <span class="st">'34'</span> <span class="ot">=</span> <span class="st">'Interleukin-4 Signaling'</span>,</span>
<span id="cb638-36"><a href="#cb638-36" aria-hidden="true" tabindex="-1"></a>  <span class="st">'35'</span> <span class="ot">=</span> <span class="st">'Leptin-Immune Interaction'</span>,</span>
<span id="cb638-37"><a href="#cb638-37" aria-hidden="true" tabindex="-1"></a>  <span class="st">'36'</span> <span class="ot">=</span> <span class="st">'General Metabolism'</span>,</span>
<span id="cb638-38"><a href="#cb638-38" aria-hidden="true" tabindex="-1"></a>  <span class="st">'37'</span> <span class="ot">=</span> <span class="st">'MyD88/TLR Signaling Deficiency'</span>,</span>
<span id="cb638-39"><a href="#cb638-39" aria-hidden="true" tabindex="-1"></a>  <span class="st">'38'</span> <span class="ot">=</span> <span class="st">'Myelination'</span>,</span>
<span id="cb638-40"><a href="#cb638-40" aria-hidden="true" tabindex="-1"></a>  <span class="st">'39'</span> <span class="ot">=</span> <span class="st">'NGF-Driven Transcription'</span>,</span>
<span id="cb638-41"><a href="#cb638-41" aria-hidden="true" tabindex="-1"></a>  <span class="st">'40'</span> <span class="ot">=</span> <span class="st">'Neddylation Pathway'</span>,</span>
<span id="cb638-42"><a href="#cb638-42" aria-hidden="true" tabindex="-1"></a>  <span class="st">'41'</span> <span class="ot">=</span> <span class="st">'Negative Regulation of Innate Immunity'</span>,</span>
<span id="cb638-43"><a href="#cb638-43" aria-hidden="true" tabindex="-1"></a>  <span class="st">'42'</span> <span class="ot">=</span> <span class="st">'NLR Signaling Pathways'</span>,</span>
<span id="cb638-44"><a href="#cb638-44" aria-hidden="true" tabindex="-1"></a>  <span class="st">'43'</span> <span class="ot">=</span> <span class="st">'Positive Cytokine Production Regulation'</span>,</span>
<span id="cb638-45"><a href="#cb638-45" aria-hidden="true" tabindex="-1"></a>  <span class="st">'44'</span> <span class="ot">=</span> <span class="st">'Positive Regulation of Muscle Contraction'</span>,</span>
<span id="cb638-46"><a href="#cb638-46" aria-hidden="true" tabindex="-1"></a>  <span class="st">'45'</span> <span class="ot">=</span> <span class="st">'TNF Production Regulation'</span>,</span>
<span id="cb638-47"><a href="#cb638-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">'46'</span> <span class="ot">=</span> <span class="st">'Prolactin and Apoptosis Regulation'</span>,</span>
<span id="cb638-48"><a href="#cb638-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">'47'</span> <span class="ot">=</span> <span class="st">'Type I IFN Signaling Regulation'</span>,</span>
<span id="cb638-49"><a href="#cb638-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">'48'</span> <span class="ot">=</span> <span class="st">'NK Cell Cytotoxicity Regulation'</span>,</span>
<span id="cb638-50"><a href="#cb638-50" aria-hidden="true" tabindex="-1"></a>  <span class="st">'49'</span> <span class="ot">=</span> <span class="st">'Cytokine Response'</span>,</span>
<span id="cb638-51"><a href="#cb638-51" aria-hidden="true" tabindex="-1"></a>  <span class="st">'50'</span> <span class="ot">=</span> <span class="st">'Interferon-Alpha Response'</span>,</span>
<span id="cb638-52"><a href="#cb638-52" aria-hidden="true" tabindex="-1"></a>  <span class="st">'51'</span> <span class="ot">=</span> <span class="st">'Interferon-Beta Response'</span>,</span>
<span id="cb638-53"><a href="#cb638-53" aria-hidden="true" tabindex="-1"></a>  <span class="st">'52'</span> <span class="ot">=</span> <span class="st">'Type II Interferon Response'</span>,</span>
<span id="cb638-54"><a href="#cb638-54" aria-hidden="true" tabindex="-1"></a>  <span class="st">'53'</span> <span class="ot">=</span> <span class="st">'Signal Transduction Pathways'</span>,</span>
<span id="cb638-55"><a href="#cb638-55" aria-hidden="true" tabindex="-1"></a>  <span class="st">'54'</span> <span class="ot">=</span> <span class="st">'Supramolecular Fiber Organization'</span>,</span>
<span id="cb638-56"><a href="#cb638-56" aria-hidden="true" tabindex="-1"></a>  <span class="st">'55'</span> <span class="ot">=</span> <span class="st">'T Cell Receptor and Apoptosis Regulation'</span>,</span>
<span id="cb638-57"><a href="#cb638-57" aria-hidden="true" tabindex="-1"></a>  <span class="st">'56'</span> <span class="ot">=</span> <span class="st">'TNF-Alpha Mediated Inflammation and Apoptosis'</span>,</span>
<span id="cb638-58"><a href="#cb638-58" aria-hidden="true" tabindex="-1"></a>  <span class="st">'57'</span> <span class="ot">=</span> <span class="st">'Type II Interferon (IFN-γ) Signaling'</span></span>
<span id="cb638-59"><a href="#cb638-59" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb639"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb639-1"><a href="#cb639-1" aria-hidden="true" tabindex="-1"></a>Annotation_list <span class="ot">&lt;-</span> <span class="fu">annotateClusters</span>(Genelist_test_cluster, cluster_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 44 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `pval_pooled = poolPValues(adj_pval, method = method, weights =
  weights, min_pval = min_pval)`.
ℹ In group 3: `condition = "B_Cells_IFNpos_vs_HC"` `Cluster = 1`.
Caused by warning in `poolPValues()`:
! Pooled p-value capped to minimum threshold: 1.0e-10
ℹ Run `dplyr::last_dplyr_warnings()` to see the 43 remaining warnings.</code></pre>
</div>
</div>
</section>
<section id="heatmap-of-enriched-biological-terms" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-enriched-biological-terms">7.7.4 Heatmap of Enriched Biological Terms</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb641"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb641-1"><a href="#cb641-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(</span>
<span id="cb641-2"><a href="#cb641-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb641-3"><a href="#cb641-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T,</span>
<span id="cb641-4"><a href="#cb641-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">rot =</span> <span class="dv">90</span>,</span>
<span id="cb641-5"><a href="#cb641-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_bar =</span> F,</span>
<span id="cb641-6"><a href="#cb641-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">column_names_centered =</span> F,</span>
<span id="cb641-7"><a href="#cb641-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> T,</span>
<span id="cb641-8"><a href="#cb641-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_columns =</span> T</span>
<span id="cb641-9"><a href="#cb641-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-215-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb642"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb642-1"><a href="#cb642-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubbleplot</span>(</span>
<span id="cb642-2"><a href="#cb642-2" aria-hidden="true" tabindex="-1"></a>  Annotation_list,</span>
<span id="cb642-3"><a href="#cb642-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">split_by_reg =</span> T</span>
<span id="cb642-4"><a href="#cb642-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-216-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="signature-scores-across-cell-types" class="level1">
<h1>8. Signature Scores Across Cell Types</h1>
<p>In this section, we compute and visualize custom gene signature scores derived from differentially expressed genes across multiple immune cell types. These signature scores are calculated using the hacksig package and allow us to assess how distinct the transcriptional patterns of INFpos patients are relative to INFneg and HC samples within each cell type.</p>
<section id="ifnpos-vs-healthy-controls" class="level2">
<h2 class="anchored" data-anchor-id="ifnpos-vs-healthy-controls">8.1 IFNpos vs Healthy Controls</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb643"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb643-1"><a href="#cb643-1" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cMo"</span>, <span class="st">"cDC"</span>, <span class="st">"B cells"</span>, <span class="st">"T cells"</span>, <span class="st">"pDC"</span>, <span class="st">"PMN"</span>)</span>
<span id="cb643-2"><a href="#cb643-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb643-3"><a href="#cb643-3" aria-hidden="true" tabindex="-1"></a>res_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb643-4"><a href="#cb643-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cMo     =</span> res_sig_cMo,</span>
<span id="cb643-5"><a href="#cb643-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cDC     =</span> res_sig_cDC,</span>
<span id="cb643-6"><a href="#cb643-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">B cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_B_Cells,</span>
<span id="cb643-7"><a href="#cb643-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">T cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_T_Cells,</span>
<span id="cb643-8"><a href="#cb643-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pDC     =</span> res_sig_pDC,</span>
<span id="cb643-9"><a href="#cb643-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">PMN     =</span> res_sig_PMN</span>
<span id="cb643-10"><a href="#cb643-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb644"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb644-1"><a href="#cb644-1" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb644-2"><a href="#cb644-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb644-3"><a href="#cb644-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb644-4"><a href="#cb644-4" aria-hidden="true" tabindex="-1"></a>  sig_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_list[[ct]])</span>
<span id="cb644-5"><a href="#cb644-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-6"><a href="#cb644-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(sig_genes) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">next</span></span>
<span id="cb644-7"><a href="#cb644-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-8"><a href="#cb644-8" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> vsd_clean[, <span class="fu">colData</span>(vsd_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> ct]</span>
<span id="cb644-9"><a href="#cb644-9" aria-hidden="true" tabindex="-1"></a>  expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_sub)</span>
<span id="cb644-10"><a href="#cb644-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-11"><a href="#cb644-11" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> <span class="fu">hack_sig</span>(expr_mat, <span class="at">signatures =</span> <span class="fu">list</span>(<span class="at">sig =</span> sig_genes), <span class="at">method =</span> <span class="st">"singscore"</span>)</span>
<span id="cb644-12"><a href="#cb644-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-13"><a href="#cb644-13" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> score_df <span class="sc">%&gt;%</span></span>
<span id="cb644-14"><a href="#cb644-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb644-15"><a href="#cb644-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">ifn_status =</span> <span class="fu">case_when</span>(</span>
<span id="cb644-16"><a href="#cb644-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNpos"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNpos"</span>,</span>
<span id="cb644-17"><a href="#cb644-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNneg"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNneg"</span>,</span>
<span id="cb644-18"><a href="#cb644-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"HC"</span>, sample_id) <span class="sc">~</span> <span class="st">"HC"</span></span>
<span id="cb644-19"><a href="#cb644-19" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb644-20"><a href="#cb644-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb644-21"><a href="#cb644-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-22"><a href="#cb644-22" aria-hidden="true" tabindex="-1"></a>  score_column <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(score_df), <span class="st">"sample_id"</span>)[<span class="dv">1</span>]</span>
<span id="cb644-23"><a href="#cb644-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb644-24"><a href="#cb644-24" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(score_df, <span class="fu">aes</span>(<span class="at">x =</span> ifn_status, <span class="at">y =</span> .data[[score_column]], <span class="at">fill =</span> ifn_status)) <span class="sc">+</span></span>
<span id="cb644-25"><a href="#cb644-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb644-26"><a href="#cb644-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb644-27"><a href="#cb644-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(ct),</span>
<span id="cb644-28"><a href="#cb644-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb644-29"><a href="#cb644-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Signature Score"</span></span>
<span id="cb644-30"><a href="#cb644-30" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb644-31"><a href="#cb644-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">50</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb644-32"><a href="#cb644-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb644-33"><a href="#cb644-33" aria-hidden="true" tabindex="-1"></a>  plots[[ct]] <span class="ot">&lt;-</span> p</span>
<span id="cb644-34"><a href="#cb644-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb645"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb645-1"><a href="#cb645-1" aria-hidden="true" tabindex="-1"></a>final_plot <span class="ot">&lt;-</span> (plots<span class="sc">$</span>cMo <span class="sc">|</span> plots<span class="sc">$</span>cDC <span class="sc">|</span> plots[[<span class="st">"B cells"</span>]]) <span class="sc">/</span></span>
<span id="cb645-2"><a href="#cb645-2" aria-hidden="true" tabindex="-1"></a>              (plots[[<span class="st">"T cells"</span>]] <span class="sc">|</span> plots<span class="sc">$</span>pDC <span class="sc">|</span> plots<span class="sc">$</span>PMN)</span>
<span id="cb645-3"><a href="#cb645-3" aria-hidden="true" tabindex="-1"></a>final_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-219-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ifnneg-vs-healthy-controls" class="level2">
<h2 class="anchored" data-anchor-id="ifnneg-vs-healthy-controls">8.1 IFNneg vs Healthy Controls</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb646"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb646-1"><a href="#cb646-1" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cMo"</span>, <span class="st">"cDC"</span>, <span class="st">"B cells"</span>, <span class="st">"T cells"</span>, <span class="st">"pDC"</span>, <span class="st">"PMN"</span>)</span>
<span id="cb646-2"><a href="#cb646-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb646-3"><a href="#cb646-3" aria-hidden="true" tabindex="-1"></a>res_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb646-4"><a href="#cb646-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cMo     =</span> res_sig_cMo_IFNneg_vs_HC,</span>
<span id="cb646-5"><a href="#cb646-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cDC     =</span> res_sig_cDC_IFNneg_vs_HC,</span>
<span id="cb646-6"><a href="#cb646-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">B cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_B_Cells_IFNneg_vs_HC,</span>
<span id="cb646-7"><a href="#cb646-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">T cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_T_Cells_IFNneg_vs_HC,</span>
<span id="cb646-8"><a href="#cb646-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pDC     =</span> res_sig_pDC_IFNneg_vs_HC,</span>
<span id="cb646-9"><a href="#cb646-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">PMN     =</span> res_sig_PMN_IFNneg_vs_HC</span>
<span id="cb646-10"><a href="#cb646-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb647"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb647-1"><a href="#cb647-1" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb647-2"><a href="#cb647-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb647-3"><a href="#cb647-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb647-4"><a href="#cb647-4" aria-hidden="true" tabindex="-1"></a>  sig_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_list[[ct]])</span>
<span id="cb647-5"><a href="#cb647-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-6"><a href="#cb647-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(sig_genes) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">next</span></span>
<span id="cb647-7"><a href="#cb647-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-8"><a href="#cb647-8" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> vsd_clean[, <span class="fu">colData</span>(vsd_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> ct]</span>
<span id="cb647-9"><a href="#cb647-9" aria-hidden="true" tabindex="-1"></a>  expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_sub)</span>
<span id="cb647-10"><a href="#cb647-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-11"><a href="#cb647-11" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> <span class="fu">hack_sig</span>(expr_mat, <span class="at">signatures =</span> <span class="fu">list</span>(<span class="at">sig =</span> sig_genes), <span class="at">method =</span> <span class="st">"singscore"</span>)</span>
<span id="cb647-12"><a href="#cb647-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-13"><a href="#cb647-13" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> score_df <span class="sc">%&gt;%</span></span>
<span id="cb647-14"><a href="#cb647-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb647-15"><a href="#cb647-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">ifn_status =</span> <span class="fu">case_when</span>(</span>
<span id="cb647-16"><a href="#cb647-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNpos"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNpos"</span>,</span>
<span id="cb647-17"><a href="#cb647-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNneg"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNneg"</span>,</span>
<span id="cb647-18"><a href="#cb647-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"HC"</span>, sample_id) <span class="sc">~</span> <span class="st">"HC"</span></span>
<span id="cb647-19"><a href="#cb647-19" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb647-20"><a href="#cb647-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb647-21"><a href="#cb647-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-22"><a href="#cb647-22" aria-hidden="true" tabindex="-1"></a>  score_column <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(score_df), <span class="st">"sample_id"</span>)[<span class="dv">1</span>]</span>
<span id="cb647-23"><a href="#cb647-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb647-24"><a href="#cb647-24" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(score_df, <span class="fu">aes</span>(<span class="at">x =</span> ifn_status, <span class="at">y =</span> .data[[score_column]], <span class="at">fill =</span> ifn_status)) <span class="sc">+</span></span>
<span id="cb647-25"><a href="#cb647-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb647-26"><a href="#cb647-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb647-27"><a href="#cb647-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(ct),</span>
<span id="cb647-28"><a href="#cb647-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb647-29"><a href="#cb647-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Signature Score"</span></span>
<span id="cb647-30"><a href="#cb647-30" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb647-31"><a href="#cb647-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">50</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb647-32"><a href="#cb647-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb647-33"><a href="#cb647-33" aria-hidden="true" tabindex="-1"></a>  plots[[ct]] <span class="ot">&lt;-</span> p</span>
<span id="cb647-34"><a href="#cb647-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb648"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb648-1"><a href="#cb648-1" aria-hidden="true" tabindex="-1"></a>final_plot <span class="ot">&lt;-</span> (plots<span class="sc">$</span>cMo <span class="sc">|</span> plots<span class="sc">$</span>cDC <span class="sc">|</span> plots[[<span class="st">"B cells"</span>]]) <span class="sc">/</span></span>
<span id="cb648-2"><a href="#cb648-2" aria-hidden="true" tabindex="-1"></a>              (plots[[<span class="st">"T cells"</span>]] <span class="sc">|</span> plots<span class="sc">$</span>pDC <span class="sc">|</span> plots<span class="sc">$</span>PMN)</span>
<span id="cb648-3"><a href="#cb648-3" aria-hidden="true" tabindex="-1"></a>final_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-222-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ifnpos-vs-ifnneg" class="level2">
<h2 class="anchored" data-anchor-id="ifnpos-vs-ifnneg">8.1 IFNpos vs IFNneg</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb649"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb649-1"><a href="#cb649-1" aria-hidden="true" tabindex="-1"></a>cell_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"cMo"</span>, <span class="st">"cDC"</span>, <span class="st">"B cells"</span>, <span class="st">"T cells"</span>, <span class="st">"pDC"</span>, <span class="st">"PMN"</span>)</span>
<span id="cb649-2"><a href="#cb649-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb649-3"><a href="#cb649-3" aria-hidden="true" tabindex="-1"></a>res_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb649-4"><a href="#cb649-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cMo     =</span> res_sig_cMo_IFNpos_vs_IFNneg,</span>
<span id="cb649-5"><a href="#cb649-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cDC     =</span> res_sig_cDC_IFNpos_vs_IFNneg,</span>
<span id="cb649-6"><a href="#cb649-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">B cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_B_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb649-7"><a href="#cb649-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">T cells</span><span class="st">`</span> <span class="ot">=</span> res_sig_T_Cells_IFNpos_vs_IFNneg,</span>
<span id="cb649-8"><a href="#cb649-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">pDC     =</span> res_sig_pDC_IFNpos_vs_IFNneg,</span>
<span id="cb649-9"><a href="#cb649-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">PMN     =</span> res_sig_PMN_IFNpos_vs_IFNneg</span>
<span id="cb649-10"><a href="#cb649-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb650"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb650-1"><a href="#cb650-1" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb650-2"><a href="#cb650-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb650-3"><a href="#cb650-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ct <span class="cf">in</span> cell_types) {</span>
<span id="cb650-4"><a href="#cb650-4" aria-hidden="true" tabindex="-1"></a>  sig_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_list[[ct]])</span>
<span id="cb650-5"><a href="#cb650-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-6"><a href="#cb650-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(sig_genes) <span class="sc">==</span> <span class="dv">0</span>) <span class="cf">next</span></span>
<span id="cb650-7"><a href="#cb650-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-8"><a href="#cb650-8" aria-hidden="true" tabindex="-1"></a>  vsd_sub <span class="ot">&lt;-</span> vsd_clean[, <span class="fu">colData</span>(vsd_clean)<span class="sc">$</span>cell_type <span class="sc">==</span> ct]</span>
<span id="cb650-9"><a href="#cb650-9" aria-hidden="true" tabindex="-1"></a>  expr_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_sub)</span>
<span id="cb650-10"><a href="#cb650-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-11"><a href="#cb650-11" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> <span class="fu">hack_sig</span>(expr_mat, <span class="at">signatures =</span> <span class="fu">list</span>(<span class="at">sig =</span> sig_genes), <span class="at">method =</span> <span class="st">"singscore"</span>)</span>
<span id="cb650-12"><a href="#cb650-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-13"><a href="#cb650-13" aria-hidden="true" tabindex="-1"></a>  score_df <span class="ot">&lt;-</span> score_df <span class="sc">%&gt;%</span></span>
<span id="cb650-14"><a href="#cb650-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb650-15"><a href="#cb650-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">ifn_status =</span> <span class="fu">case_when</span>(</span>
<span id="cb650-16"><a href="#cb650-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNpos"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNpos"</span>,</span>
<span id="cb650-17"><a href="#cb650-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"IFNneg"</span>, sample_id) <span class="sc">~</span> <span class="st">"IFNneg"</span>,</span>
<span id="cb650-18"><a href="#cb650-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"HC"</span>, sample_id) <span class="sc">~</span> <span class="st">"HC"</span></span>
<span id="cb650-19"><a href="#cb650-19" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb650-20"><a href="#cb650-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb650-21"><a href="#cb650-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-22"><a href="#cb650-22" aria-hidden="true" tabindex="-1"></a>  score_column <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(score_df), <span class="st">"sample_id"</span>)[<span class="dv">1</span>]</span>
<span id="cb650-23"><a href="#cb650-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb650-24"><a href="#cb650-24" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(score_df, <span class="fu">aes</span>(<span class="at">x =</span> ifn_status, <span class="at">y =</span> .data[[score_column]], <span class="at">fill =</span> ifn_status)) <span class="sc">+</span></span>
<span id="cb650-25"><a href="#cb650-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb650-26"><a href="#cb650-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb650-27"><a href="#cb650-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(ct),</span>
<span id="cb650-28"><a href="#cb650-28" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb650-29"><a href="#cb650-29" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Signature Score"</span></span>
<span id="cb650-30"><a href="#cb650-30" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb650-31"><a href="#cb650-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">50</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb650-32"><a href="#cb650-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb650-33"><a href="#cb650-33" aria-hidden="true" tabindex="-1"></a>  plots[[ct]] <span class="ot">&lt;-</span> p</span>
<span id="cb650-34"><a href="#cb650-34" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb651"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb651-1"><a href="#cb651-1" aria-hidden="true" tabindex="-1"></a>final_plot <span class="ot">&lt;-</span> (plots<span class="sc">$</span>cMo <span class="sc">|</span> plots<span class="sc">$</span>cDC <span class="sc">|</span> plots[[<span class="st">"B cells"</span>]]) <span class="sc">/</span></span>
<span id="cb651-2"><a href="#cb651-2" aria-hidden="true" tabindex="-1"></a>              (plots[[<span class="st">"T cells"</span>]] <span class="sc">|</span> plots<span class="sc">$</span>pDC <span class="sc">|</span> plots<span class="sc">$</span>PMN)</span>
<span id="cb651-3"><a href="#cb651-3" aria-hidden="true" tabindex="-1"></a>final_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="SLE_Transcriptome_Analysis_files/figure-html/unnamed-chunk-225-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="references" class="level1">
<h1>9. References</h1>
<p>Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for sequence count data: removing the noise and preserving large differences. Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/bty895" class="uri">https://doi.org/10.1093/bioinformatics/bty895</a></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>